{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aa01eedb-fcbd-45b5-a642-c308b717a25c",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Aggregate all the various counts and metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "080d9351-837f-4e60-b199-de54a8e67b66",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T18:48:19.768974Z",
     "iopub.status.busy": "2023-04-08T18:48:19.768365Z",
     "iopub.status.idle": "2023-04-08T18:48:20.363478Z",
     "shell.execute_reply": "2023-04-08T18:48:20.362269Z",
     "shell.execute_reply.started": "2023-04-08T18:48:19.768928Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "import numpy\n",
    "\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "442943df-4fd0-455e-ab33-1c0c16a61496",
   "metadata": {},
   "source": [
    "Get variables from `snakemake`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "301ac539-6b0b-4d37-b08b-43a059f521e8",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T18:48:44.985083Z",
     "iopub.status.busy": "2023-04-08T18:48:44.984341Z",
     "iopub.status.idle": "2023-04-08T18:48:44.991715Z",
     "shell.execute_reply": "2023-04-08T18:48:44.990476Z",
     "shell.execute_reply.started": "2023-04-08T18:48:44.985026Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "counts_and_coverage_files = snakemake.input.counts_and_coverage\n",
    "mito_ref_info_csv = snakemake.input.mito_ref_info\n",
    "metadata_csv = snakemake.input.metadata\n",
    "read_counts_csv = snakemake.input.read_counts\n",
    "sars2_ref_id = snakemake.params.sars2_ref_id\n",
    "mito_genomes_to_keep = snakemake.params.mito_genomes_to_keep\n",
    "mito_composition_filters = snakemake.params.mito_composition_filters\n",
    "metagenomic_descriptions = snakemake.params.metagenomic_descriptions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bfc352d8-105f-4f1e-9412-b376809d0b25",
   "metadata": {},
   "source": [
    "Read information on mitochondrial reference genomes, and get species name as just first two words:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9936a779-9b99-4b35-b2b1-cf41bdd23edc",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T18:48:26.077066Z",
     "iopub.status.busy": "2023-04-08T18:48:26.076351Z",
     "iopub.status.idle": "2023-04-08T18:48:26.099800Z",
     "shell.execute_reply": "2023-04-08T18:48:26.099038Z",
     "shell.execute_reply.started": "2023-04-08T18:48:26.077013Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>species</th>\n",
       "      <th>common_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KX964606.1</td>\n",
       "      <td>Erinaceus amurensis</td>\n",
       "      <td>Amur hedgehog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_012095.1</td>\n",
       "      <td>Sus scrofa domesticus</td>\n",
       "      <td>pig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>Rattus norvegicus strain Wistar</td>\n",
       "      <td>brown rat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_001913.1</td>\n",
       "      <td>Oryctolagus cuniculus</td>\n",
       "      <td>rabbit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_002008.4</td>\n",
       "      <td>Canis lupus familiaris</td>\n",
       "      <td>dog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4166</th>\n",
       "      <td>NC_029381.1</td>\n",
       "      <td>Tlacuatzin canescens</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4167</th>\n",
       "      <td>NC_026049.1</td>\n",
       "      <td>Pelusios castaneus voucher AHNU26080283</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4168</th>\n",
       "      <td>NC_044441.1</td>\n",
       "      <td>Aphyosemion cognatum voucher AMNH-I-251785</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4169</th>\n",
       "      <td>NC_051037.1</td>\n",
       "      <td>Cinclus mexicanus</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4170</th>\n",
       "      <td>NC_060309.1</td>\n",
       "      <td>Nyctalus aviator</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4171 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               id                                     species    common_name\n",
       "0      KX964606.1                         Erinaceus amurensis  Amur hedgehog\n",
       "1     NC_012095.1                       Sus scrofa domesticus            pig\n",
       "2     AC_000022.2             Rattus norvegicus strain Wistar      brown rat\n",
       "3     NC_001913.1                       Oryctolagus cuniculus         rabbit\n",
       "4     NC_002008.4                      Canis lupus familiaris            dog\n",
       "...           ...                                         ...            ...\n",
       "4166  NC_029381.1                        Tlacuatzin canescens            NaN\n",
       "4167  NC_026049.1     Pelusios castaneus voucher AHNU26080283            NaN\n",
       "4168  NC_044441.1  Aphyosemion cognatum voucher AMNH-I-251785            NaN\n",
       "4169  NC_051037.1                           Cinclus mexicanus            NaN\n",
       "4170  NC_060309.1                            Nyctalus aviator            NaN\n",
       "\n",
       "[4171 rows x 3 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mito_ref_info = pd.read_csv(mito_ref_info_csv)\n",
    "\n",
    "mito_ref_info"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28989fa2-3ee3-4f93-9c2d-1cfdd8b50765",
   "metadata": {},
   "source": [
    "Read metadata on samples.\n",
    "For sample titles that have multiple sample names, append the name to the title:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "66dd0295-14a9-4004-a670-a7c609ba2459",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T18:48:28.896465Z",
     "iopub.status.busy": "2023-04-08T18:48:28.895874Z",
     "iopub.status.idle": "2023-04-08T18:48:28.918904Z",
     "shell.execute_reply": "2023-04-08T18:48:28.918054Z",
     "shell.execute_reply.started": "2023-04-08T18:48:28.896421Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "metadata_all = pd.read_csv(metadata_csv)\n",
    "\n",
    "# downsize to most relevant metadata to keep\n",
    "metadata = (\n",
    "    metadata_all\n",
    "    # make sample titles unique if they correspond to multiple names\n",
    "    .assign(\n",
    "        n_sample_names=lambda x: (\n",
    "            x.groupby(\"Sample title\")[\"Sample name\"].transform(\"nunique\")\n",
    "        ),\n",
    "        sample=lambda x: x[\"Sample title\"].where(\n",
    "            x[\"n_sample_names\"] == 1,\n",
    "            x[\"Sample title\"] + \"_\" + x[\"Sample name\"],\n",
    "        )\n",
    "    )\n",
    "    # rename and organize\n",
    "    .rename(columns={\"Public description\": \"description\"})\n",
    "    [\n",
    "        [\n",
    "            \"Run accession\",\n",
    "            \"sample\",\n",
    "            \"Sample name\",\n",
    "            \"Collection date\",\n",
    "            \"description\",\n",
    "            \"Isolation source\",\n",
    "        ]\n",
    "    ]\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bf5e0344-11fa-406b-99b1-5d9d10e238f1",
   "metadata": {},
   "source": [
    "Tally all the alignment counts:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "be252664-0415-4f43-b4f1-25d6c642bf5f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T18:59:18.795523Z",
     "iopub.status.busy": "2023-04-08T18:59:18.795144Z",
     "iopub.status.idle": "2023-04-08T18:59:21.340931Z",
     "shell.execute_reply": "2023-04-08T18:59:21.339784Z",
     "shell.execute_reply.started": "2023-04-08T18:59:18.795494Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run accession</th>\n",
       "      <th>alignment_reference</th>\n",
       "      <th>aligned_reads</th>\n",
       "      <th>covered_bases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>NC_045512v2</td>\n",
       "      <td>3</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>NC_021950.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>NC_046471.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>NC_029247.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>NC_007783.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643763</th>\n",
       "      <td>CRR711155</td>\n",
       "      <td>NC_026049.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643764</th>\n",
       "      <td>CRR711155</td>\n",
       "      <td>NC_044441.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643765</th>\n",
       "      <td>CRR711155</td>\n",
       "      <td>NC_051037.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643766</th>\n",
       "      <td>CRR711155</td>\n",
       "      <td>NC_060309.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643767</th>\n",
       "      <td>CRR711155</td>\n",
       "      <td>KX964606.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1643768 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Run accession alignment_reference  aligned_reads  covered_bases\n",
       "0           CRR710761         NC_045512v2              3            129\n",
       "1           CRR710761         NC_021950.1              0              0\n",
       "2           CRR710761         NC_046471.1              0              0\n",
       "3           CRR710761         NC_029247.1              0              0\n",
       "4           CRR710761         NC_007783.2              0              0\n",
       "...               ...                 ...            ...            ...\n",
       "1643763     CRR711155         NC_026049.1              0              0\n",
       "1643764     CRR711155         NC_044441.1              0              0\n",
       "1643765     CRR711155         NC_051037.1              0              0\n",
       "1643766     CRR711155         NC_060309.1              0              0\n",
       "1643767     CRR711155          KX964606.1              0              0\n",
       "\n",
       "[1643768 rows x 4 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_and_coverage = pd.concat(\n",
    "    [\n",
    "        (\n",
    "            pd.read_csv(\n",
    "                f,\n",
    "                sep=\"\\t\",\n",
    "                names=[\"alignment_reference\", \"aligned_reads\", \"covered_bases\"],\n",
    "                header=0,\n",
    "            )\n",
    "            .assign(**{\"Run accession\": os.path.splitext(os.path.basename(f))[0]})\n",
    "        )\n",
    "        for f in counts_and_coverage_files\n",
    "    ],\n",
    "    ignore_index=True,\n",
    ")[[\"Run accession\", \"alignment_reference\", \"aligned_reads\", \"covered_bases\"]]\n",
    "\n",
    "# make sure all alignment references are SARS2 or a known mitochondrial genome\n",
    "assert set(counts_and_coverage[\"alignment_reference\"]).issubset(\n",
    "    set(mito_ref_info[\"id\"]).union([sars2_ref_id])\n",
    ")\n",
    "\n",
    "assert (\n",
    "    len(counts_and_coverage)\n",
    "    == len(counts_and_coverage.drop_duplicates())\n",
    "    == counts_and_coverage[\"alignment_reference\"].nunique() * counts_and_coverage[\"Run accession\"].nunique()\n",
    ")\n",
    "\n",
    "counts_and_coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "106771ea-1d7b-47fd-8aea-3502db94cef9",
   "metadata": {},
   "source": [
    "We expect some correlation between aligned reads and covered bases, check that is true among references with reasonable coverage:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "7bba4480-39da-42d2-90e9-d23654447082",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T19:10:00.127632Z",
     "iopub.status.busy": "2023-04-08T19:10:00.127176Z",
     "iopub.status.idle": "2023-04-08T19:10:01.665438Z",
     "shell.execute_reply": "2023-04-08T19:10:01.664388Z",
     "shell.execute_reply.started": "2023-04-08T19:10:00.127595Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alignment_reference</th>\n",
       "      <th>correlation</th>\n",
       "      <th>total_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_065160.1</td>\n",
       "      <td>0.712755</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_026307.1</td>\n",
       "      <td>0.777050</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_037513.1</td>\n",
       "      <td>0.799770</td>\n",
       "      <td>2920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_015238.2</td>\n",
       "      <td>0.816475</td>\n",
       "      <td>790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_009728.1</td>\n",
       "      <td>0.860335</td>\n",
       "      <td>371176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>NC_003128.3</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>NC_008066.1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>157882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>NC_023962.1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>NC_007439.1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>NC_008140.1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>308</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    alignment_reference  correlation  total_reads\n",
       "0           NC_065160.1     0.712755          164\n",
       "1           NC_026307.1     0.777050          108\n",
       "2           NC_037513.1     0.799770         2920\n",
       "3           NC_015238.2     0.816475          790\n",
       "4           NC_009728.1     0.860335       371176\n",
       "..                  ...          ...          ...\n",
       "165         NC_003128.3     0.999999          145\n",
       "166         NC_008066.1     1.000000       157882\n",
       "167         NC_023962.1     1.000000          194\n",
       "168         NC_007439.1     1.000000          139\n",
       "169         NC_008140.1     1.000000          308\n",
       "\n",
       "[170 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "read_vs_coverage_corr = (\n",
    "    counts_and_coverage\n",
    "    .groupby(\"alignment_reference\")\n",
    "    [[\"aligned_reads\", \"covered_bases\"]]\n",
    "    .corr(method=\"spearman\")\n",
    "    .reset_index()\n",
    "    .query(\"level_1 == 'aligned_reads'\")\n",
    "    .drop(columns=[\"level_1\", \"aligned_reads\"])\n",
    "    .rename(columns={\"covered_bases\": \"correlation\"})\n",
    "    .query(\"correlation.notnull()\")\n",
    "    .sort_values(\"correlation\")\n",
    "    .merge(\n",
    "        counts_and_coverage\n",
    "        .groupby(\"alignment_reference\", as_index=False)\n",
    "        .aggregate(total_reads=pd.NamedAgg(\"aligned_reads\", \"sum\"))\n",
    "    )\n",
    "    .query(\"total_reads > 100\")\n",
    "    .reset_index(drop=True)\n",
    ")\n",
    "\n",
    "display(read_vs_coverage_corr)\n",
    "\n",
    "assert read_vs_coverage_corr[\"correlation\"].min() > 0.65, read_vs_coverage_corr"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6955a97c-9227-4993-a247-1ccf487fd55b",
   "metadata": {},
   "source": [
    "Get the read and alignment counts:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "b1e5d096-cfe9-4bb1-b255-c178daa5a605",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T18:58:40.154643Z",
     "iopub.status.busy": "2023-04-08T18:58:40.153996Z",
     "iopub.status.idle": "2023-04-08T18:58:40.166595Z",
     "shell.execute_reply": "2023-04-08T18:58:40.165733Z",
     "shell.execute_reply.started": "2023-04-08T18:58:40.154599Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "read_counts = pd.read_csv(read_counts_csv)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6748fae6-e8a7-4b99-85c2-980abbe1ac04",
   "metadata": {},
   "source": [
    "## SARS-CoV-2 read counts\n",
    "Now get the SARS-CoV-2 read counts as a percentage of all preprocessed reads.\n",
    "\n",
    "We first do this by run:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "a8807c49-2033-43ec-a14a-d68ce74ab27a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T19:00:36.400669Z",
     "iopub.status.busy": "2023-04-08T19:00:36.400373Z",
     "iopub.status.idle": "2023-04-08T19:00:36.483455Z",
     "shell.execute_reply": "2023-04-08T19:00:36.482515Z",
     "shell.execute_reply.started": "2023-04-08T19:00:36.400646Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run accession</th>\n",
       "      <th>SARS2_aligned_reads</th>\n",
       "      <th>SARS2_covered_bases</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>preprocessed_reads</th>\n",
       "      <th>percent_preprocessed_reads_aligning_to_SARS2</th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>description</th>\n",
       "      <th>Isolation source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>3</td>\n",
       "      <td>129</td>\n",
       "      <td>106545428</td>\n",
       "      <td>90854684</td>\n",
       "      <td>3.301976e-06</td>\n",
       "      <td>Q37</td>\n",
       "      <td>Env_0552</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRR710762</td>\n",
       "      <td>1</td>\n",
       "      <td>43</td>\n",
       "      <td>213214887</td>\n",
       "      <td>206952411</td>\n",
       "      <td>4.832029e-07</td>\n",
       "      <td>Q61</td>\n",
       "      <td>Env_0576</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRR710763</td>\n",
       "      <td>3</td>\n",
       "      <td>129</td>\n",
       "      <td>120407578</td>\n",
       "      <td>120404558</td>\n",
       "      <td>2.491600e-06</td>\n",
       "      <td>Q64</td>\n",
       "      <td>Env_0579</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRR710764</td>\n",
       "      <td>2</td>\n",
       "      <td>86</td>\n",
       "      <td>110003167</td>\n",
       "      <td>87071974</td>\n",
       "      <td>2.296950e-06</td>\n",
       "      <td>Q68</td>\n",
       "      <td>Env_0583</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRR710765</td>\n",
       "      <td>2</td>\n",
       "      <td>86</td>\n",
       "      <td>106780574</td>\n",
       "      <td>106777962</td>\n",
       "      <td>1.873046e-06</td>\n",
       "      <td>Q69</td>\n",
       "      <td>Env_0584</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389</th>\n",
       "      <td>CRR711142</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>123643938</td>\n",
       "      <td>123541212</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>CCDC-760</td>\n",
       "      <td>Env_0909</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>390</th>\n",
       "      <td>CRR711152</td>\n",
       "      <td>594725</td>\n",
       "      <td>29870</td>\n",
       "      <td>3342164</td>\n",
       "      <td>3202808</td>\n",
       "      <td>1.856886e+01</td>\n",
       "      <td>F13-20</td>\n",
       "      <td>Env_0313_seq04</td>\n",
       "      <td>2020-01-20</td>\n",
       "      <td>SARS-CoV-2 Amplicon based SARS-CoV-2 whole gen...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391</th>\n",
       "      <td>CRR711153</td>\n",
       "      <td>1210978</td>\n",
       "      <td>29869</td>\n",
       "      <td>5222964</td>\n",
       "      <td>5016616</td>\n",
       "      <td>2.413934e+01</td>\n",
       "      <td>F13-21</td>\n",
       "      <td>Env_0313_seq05</td>\n",
       "      <td>2020-01-21</td>\n",
       "      <td>SARS-CoV-2 Amplicon based SARS-CoV-2 whole gen...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>CRR711154</td>\n",
       "      <td>4156951</td>\n",
       "      <td>29870</td>\n",
       "      <td>4419724</td>\n",
       "      <td>4387884</td>\n",
       "      <td>9.473703e+01</td>\n",
       "      <td>B5-P3</td>\n",
       "      <td>Env_0126_seq06</td>\n",
       "      <td>2020-02-29</td>\n",
       "      <td>SARS-CoV-2 Amplicon based SARS-CoV-2 whole gen...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>CRR711155</td>\n",
       "      <td>3789607</td>\n",
       "      <td>29870</td>\n",
       "      <td>3983946</td>\n",
       "      <td>3965246</td>\n",
       "      <td>9.557054e+01</td>\n",
       "      <td>F54-P3</td>\n",
       "      <td>Env_0354_seq07</td>\n",
       "      <td>2020-02-29</td>\n",
       "      <td>SARS-CoV-2 Amplicon based SARS-CoV-2 whole gen...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>394 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Run accession  SARS2_aligned_reads  SARS2_covered_bases  total_reads  \\\n",
       "0       CRR710761                    3                  129    106545428   \n",
       "1       CRR710762                    1                   43    213214887   \n",
       "2       CRR710763                    3                  129    120407578   \n",
       "3       CRR710764                    2                   86    110003167   \n",
       "4       CRR710765                    2                   86    106780574   \n",
       "..            ...                  ...                  ...          ...   \n",
       "389     CRR711142                    0                    0    123643938   \n",
       "390     CRR711152               594725                29870      3342164   \n",
       "391     CRR711153              1210978                29869      5222964   \n",
       "392     CRR711154              4156951                29870      4419724   \n",
       "393     CRR711155              3789607                29870      3983946   \n",
       "\n",
       "     preprocessed_reads  percent_preprocessed_reads_aligning_to_SARS2  \\\n",
       "0              90854684                                  3.301976e-06   \n",
       "1             206952411                                  4.832029e-07   \n",
       "2             120404558                                  2.491600e-06   \n",
       "3              87071974                                  2.296950e-06   \n",
       "4             106777962                                  1.873046e-06   \n",
       "..                  ...                                           ...   \n",
       "389           123541212                                  0.000000e+00   \n",
       "390             3202808                                  1.856886e+01   \n",
       "391             5016616                                  2.413934e+01   \n",
       "392             4387884                                  9.473703e+01   \n",
       "393             3965246                                  9.557054e+01   \n",
       "\n",
       "       sample     Sample name Collection date  \\\n",
       "0         Q37        Env_0552      2020-01-12   \n",
       "1         Q61        Env_0576      2020-01-12   \n",
       "2         Q64        Env_0579      2020-01-12   \n",
       "3         Q68        Env_0583      2020-01-12   \n",
       "4         Q69        Env_0584      2020-01-12   \n",
       "..        ...             ...             ...   \n",
       "389  CCDC-760        Env_0909      2020-03-02   \n",
       "390    F13-20  Env_0313_seq04      2020-01-20   \n",
       "391    F13-21  Env_0313_seq05      2020-01-21   \n",
       "392     B5-P3  Env_0126_seq06      2020-02-29   \n",
       "393    F54-P3  Env_0354_seq07      2020-02-29   \n",
       "\n",
       "                                           description  Isolation source  \n",
       "0    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "1    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "2    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "3    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "4    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "..                                                 ...               ...  \n",
       "389  RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "390  SARS-CoV-2 Amplicon based SARS-CoV-2 whole gen...  West Wine of HSM  \n",
       "391  SARS-CoV-2 Amplicon based SARS-CoV-2 whole gen...  West Wine of HSM  \n",
       "392  SARS-CoV-2 Amplicon based SARS-CoV-2 whole gen...  West Wine of HSM  \n",
       "393  SARS-CoV-2 Amplicon based SARS-CoV-2 whole gen...  West Wine of HSM  \n",
       "\n",
       "[394 rows x 11 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sars2_aligned_by_run = (\n",
    "    counts_and_coverage\n",
    "    .query(\"alignment_reference == @sars2_ref_id\")\n",
    "    .drop(columns=\"alignment_reference\")\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"aligned_reads\": \"SARS2_aligned_reads\",\n",
    "            \"covered_bases\": \"SARS2_covered_bases\",\n",
    "        },\n",
    "    )\n",
    "    .merge(read_counts, validate=\"one_to_one\", how=\"outer\")\n",
    "    .assign(\n",
    "        percent_preprocessed_reads_aligning_to_SARS2=(\n",
    "            lambda x: x[\"SARS2_aligned_reads\"] / x[\"preprocessed_reads\"] * 100\n",
    "        ),\n",
    "    )\n",
    "    .merge(\n",
    "        metadata,\n",
    "        on=\"Run accession\",\n",
    "        validate=\"one_to_one\",\n",
    "        how=\"outer\",\n",
    "    )\n",
    ")\n",
    "\n",
    "sars2_aligned_by_run.to_csv(\n",
    "    snakemake.output[\"sars2_aligned_by_run\"], index=False, float_format=\"%.5g\",\n",
    ")\n",
    "\n",
    "sars2_aligned_by_run"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d6385ce-0467-40cc-a449-aad91e9aa762",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-04T12:31:56.085557Z",
     "iopub.status.busy": "2023-04-04T12:31:56.085007Z",
     "iopub.status.idle": "2023-04-04T12:31:56.093519Z",
     "shell.execute_reply": "2023-04-04T12:31:56.092581Z",
     "shell.execute_reply.started": "2023-04-04T12:31:56.085514Z"
    },
    "tags": []
   },
   "source": [
    "Now we do this by sample (aggregating runs for each sample) for **just the metagenomic samples**."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f9d7b1f6-50a1-4151-a843-c89e86cc2a84",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T19:01:42.857396Z",
     "iopub.status.busy": "2023-04-08T19:01:42.856848Z",
     "iopub.status.idle": "2023-04-08T19:01:42.910983Z",
     "shell.execute_reply": "2023-04-08T19:01:42.910036Z",
     "shell.execute_reply.started": "2023-04-08T19:01:42.857352Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>description</th>\n",
       "      <th>Isolation source</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>preprocessed_reads</th>\n",
       "      <th>SARS2_aligned_reads</th>\n",
       "      <th>percent_preprocessed_reads_aligning_to_SARS2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>06-29-abv1</td>\n",
       "      <td>Env_0620</td>\n",
       "      <td>2020-01-23</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>1079014964</td>\n",
       "      <td>851849944</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>HJ200033-20200112-1</td>\n",
       "      <td>Env_0548</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>116117847</td>\n",
       "      <td>116115128</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>HJ200034-20200112-1</td>\n",
       "      <td>Env_0549</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>148630458</td>\n",
       "      <td>148628067</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>HJ200035-20200112-1</td>\n",
       "      <td>Env_0550</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>114656800</td>\n",
       "      <td>71848972</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>HJ200036-20200112-1</td>\n",
       "      <td>Env_0551</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>229948912</td>\n",
       "      <td>223587184</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>F98</td>\n",
       "      <td>Env_0398</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>401129274</td>\n",
       "      <td>399741992</td>\n",
       "      <td>330</td>\n",
       "      <td>0.000083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>A61</td>\n",
       "      <td>Env_0061</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>449907818</td>\n",
       "      <td>447903414</td>\n",
       "      <td>333</td>\n",
       "      <td>0.000074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>B17</td>\n",
       "      <td>Env_0138</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>213123774</td>\n",
       "      <td>212249334</td>\n",
       "      <td>405</td>\n",
       "      <td>0.000191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>B5</td>\n",
       "      <td>Env_0126</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>229464470</td>\n",
       "      <td>228684372</td>\n",
       "      <td>1458</td>\n",
       "      <td>0.000638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>F100</td>\n",
       "      <td>Env_0400</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>260045208</td>\n",
       "      <td>258891974</td>\n",
       "      <td>7182</td>\n",
       "      <td>0.002774</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>169 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sample Sample name Collection date  \\\n",
       "0             06-29-abv1    Env_0620      2020-01-23   \n",
       "102  HJ200033-20200112-1    Env_0548      2020-01-12   \n",
       "103  HJ200034-20200112-1    Env_0549      2020-01-12   \n",
       "104  HJ200035-20200112-1    Env_0550      2020-01-12   \n",
       "105  HJ200036-20200112-1    Env_0551      2020-01-12   \n",
       "..                   ...         ...             ...   \n",
       "69                   F98    Env_0398      2020-01-01   \n",
       "40                   A61    Env_0061      2020-01-01   \n",
       "46                   B17    Env_0138      2020-01-01   \n",
       "47                    B5    Env_0126      2020-01-01   \n",
       "67                  F100    Env_0400      2020-01-01   \n",
       "\n",
       "                                           description  Isolation source  \\\n",
       "0    RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "102  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "103  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "104  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "105  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "..                                                 ...               ...   \n",
       "69   RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "40   RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "46   RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "47   RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "67   RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "\n",
       "     total_reads  preprocessed_reads  SARS2_aligned_reads  \\\n",
       "0     1079014964           851849944                    0   \n",
       "102    116117847           116115128                    0   \n",
       "103    148630458           148628067                    0   \n",
       "104    114656800            71848972                    0   \n",
       "105    229948912           223587184                    0   \n",
       "..           ...                 ...                  ...   \n",
       "69     401129274           399741992                  330   \n",
       "40     449907818           447903414                  333   \n",
       "46     213123774           212249334                  405   \n",
       "47     229464470           228684372                 1458   \n",
       "67     260045208           258891974                 7182   \n",
       "\n",
       "     percent_preprocessed_reads_aligning_to_SARS2  \n",
       "0                                        0.000000  \n",
       "102                                      0.000000  \n",
       "103                                      0.000000  \n",
       "104                                      0.000000  \n",
       "105                                      0.000000  \n",
       "..                                            ...  \n",
       "69                                       0.000083  \n",
       "40                                       0.000074  \n",
       "46                                       0.000191  \n",
       "47                                       0.000638  \n",
       "67                                       0.002774  \n",
       "\n",
       "[169 rows x 9 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# columns that we sum across runs for each sample\n",
    "sum_cols = [\n",
    "    \"total_reads\",\n",
    "    \"preprocessed_reads\",\n",
    "    \"SARS2_aligned_reads\",\n",
    "]\n",
    "\n",
    "sars2_aligned_by_metagenomic_sample = (\n",
    "    sars2_aligned_by_run\n",
    "    .query(\"description in @metagenomic_descriptions\")\n",
    "    .groupby(\n",
    "        [\n",
    "            c for c in sars2_aligned_by_run\n",
    "            if c not in sum_cols\n",
    "            and not c.startswith(\"percent_\")\n",
    "            and c not in [\"Run accession\", \"SARS2_covered_bases\"]\n",
    "        ],\n",
    "        as_index=False,\n",
    "        dropna=False,\n",
    "    )\n",
    "    .aggregate({c: \"sum\" for c in sum_cols})\n",
    "    .assign(\n",
    "        percent_preprocessed_reads_aligning_to_SARS2=(\n",
    "            lambda x: x[\"SARS2_aligned_reads\"] / x[\"preprocessed_reads\"] * 100\n",
    "        ),\n",
    "    )\n",
    ")\n",
    "\n",
    "dup_samples = (\n",
    "    sars2_aligned_by_metagenomic_sample\n",
    "    .groupby(\"sample\")\n",
    "    .aggregate(\n",
    "        n=pd.NamedAgg(\"total_reads\", \"count\"),\n",
    "        sample_names=pd.NamedAgg(\"Sample name\", \"unique\"),\n",
    "    )\n",
    "    .query(\"n > 1\")\n",
    ")\n",
    "\n",
    "assert not len(dup_samples), f\"Some duplicated sample names\\n{dup_samples}\"\n",
    "\n",
    "sars2_aligned_by_metagenomic_sample.to_csv(\n",
    "    snakemake.output[\"sars2_aligned_by_metagenomic_sample\"], index=False, float_format=\"%.5g\",\n",
    ")\n",
    "\n",
    "sars2_aligned_by_metagenomic_sample.sort_values(\"SARS2_aligned_reads\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "024f4dc9-93f2-4f56-9380-716b59ca2dd7",
   "metadata": {},
   "source": [
    "## Composition of mitochondrial reads\n",
    "\n",
    "Do some filtering on the runs:\n",
    " - Get just the mitochondrial counts\n",
    " - Only keep runs with the metagenomic description\n",
    " - Exclude runs with insufficient alignment counts to mitochondria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "b258ec79-2e5d-4f6e-9c31-ac4fa339f509",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T19:13:35.037234Z",
     "iopub.status.busy": "2023-04-08T19:13:35.036605Z",
     "iopub.status.idle": "2023-04-08T19:13:36.431919Z",
     "shell.execute_reply": "2023-04-08T19:13:36.431296Z",
     "shell.execute_reply.started": "2023-04-08T19:13:35.037192Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding the following metagenomic runs with insufficient aligned mitochondrial reads:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run accession</th>\n",
       "      <th>total_aligned_reads</th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>description</th>\n",
       "      <th>Isolation source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRR710915</td>\n",
       "      <td>0</td>\n",
       "      <td>1-29-4</td>\n",
       "      <td>Env_0682</td>\n",
       "      <td>2020-01-29</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>East Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRR710914</td>\n",
       "      <td>0</td>\n",
       "      <td>1-27-37</td>\n",
       "      <td>Env_0664</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRR711110</td>\n",
       "      <td>13</td>\n",
       "      <td>C9</td>\n",
       "      <td>Env_0839</td>\n",
       "      <td>2020-02-20</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRR710920</td>\n",
       "      <td>15</td>\n",
       "      <td>8-25-D_Env_0717</td>\n",
       "      <td>Env_0717</td>\n",
       "      <td>2020-02-03</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRR710919</td>\n",
       "      <td>16</td>\n",
       "      <td>8-25-D_Env_0717</td>\n",
       "      <td>Env_0717</td>\n",
       "      <td>2020-02-03</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>CRR711046</td>\n",
       "      <td>940</td>\n",
       "      <td>1-29-23</td>\n",
       "      <td>Env_0701</td>\n",
       "      <td>2020-01-29</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>East Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>CRR710950</td>\n",
       "      <td>959</td>\n",
       "      <td>HJ200015-20200112-1</td>\n",
       "      <td>Env_0530</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>CRR710925</td>\n",
       "      <td>968</td>\n",
       "      <td>HXJC4-1-3</td>\n",
       "      <td>Env_0802</td>\n",
       "      <td>2020-02-09</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>Sewerage well in the surrounding areas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>CRR710939</td>\n",
       "      <td>978</td>\n",
       "      <td>HJ200004-20200112-1</td>\n",
       "      <td>Env_0519</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>CRR711086</td>\n",
       "      <td>983</td>\n",
       "      <td>8-25-D1</td>\n",
       "      <td>Env_0753</td>\n",
       "      <td>2020-02-09</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>211 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Run accession  total_aligned_reads               sample Sample name  \\\n",
       "0       CRR710915                    0               1-29-4    Env_0682   \n",
       "1       CRR710914                    0              1-27-37    Env_0664   \n",
       "2       CRR711110                   13                   C9    Env_0839   \n",
       "3       CRR710920                   15      8-25-D_Env_0717    Env_0717   \n",
       "4       CRR710919                   16      8-25-D_Env_0717    Env_0717   \n",
       "..            ...                  ...                  ...         ...   \n",
       "206     CRR711046                  940              1-29-23    Env_0701   \n",
       "207     CRR710950                  959  HJ200015-20200112-1    Env_0530   \n",
       "208     CRR710925                  968            HXJC4-1-3    Env_0802   \n",
       "209     CRR710939                  978  HJ200004-20200112-1    Env_0519   \n",
       "210     CRR711086                  983              8-25-D1    Env_0753   \n",
       "\n",
       "    Collection date                                        description  \\\n",
       "0        2020-01-29  RNA sequencing of total nucleic acids from env...   \n",
       "1        2020-01-27  RNA sequencing of total nucleic acids from env...   \n",
       "2        2020-02-20  RNA sequencing of total nucleic acids from env...   \n",
       "3        2020-02-03  RNA sequencing of total nucleic acids from env...   \n",
       "4        2020-02-03  RNA sequencing of total nucleic acids from env...   \n",
       "..              ...                                                ...   \n",
       "206      2020-01-29  RNA sequencing of total nucleic acids from env...   \n",
       "207      2020-01-12  RNA sequencing of total nucleic acids from env...   \n",
       "208      2020-02-09  RNA sequencing of total nucleic acids from env...   \n",
       "209      2020-01-12  RNA sequencing of total nucleic acids from env...   \n",
       "210      2020-02-09  RNA sequencing of total nucleic acids from env...   \n",
       "\n",
       "                                Isolation source  \n",
       "0    East Wine of HSM (Sewers or sewerage wells)  \n",
       "1    West Wine of HSM (Sewers or sewerage wells)  \n",
       "2                               West Wine of HSM  \n",
       "3                               West Wine of HSM  \n",
       "4                               West Wine of HSM  \n",
       "..                                           ...  \n",
       "206                             East Wine of HSM  \n",
       "207                             West Wine of HSM  \n",
       "208       Sewerage well in the surrounding areas  \n",
       "209                             West Wine of HSM  \n",
       "210                             West Wine of HSM  \n",
       "\n",
       "[211 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mito_counts = (\n",
    "    counts_and_coverage\n",
    "    .merge(metadata, validate=\"many_to_one\")\n",
    "    .query(\"alignment_reference != @sars2_ref_id\")\n",
    "    .query(\"description in @metagenomic_descriptions\")\n",
    ")\n",
    "\n",
    "insufficient_mito_reads = (\n",
    "    mito_counts\n",
    "    .groupby(\"Run accession\", as_index=False)\n",
    "    .aggregate(total_aligned_reads=pd.NamedAgg(\"aligned_reads\", \"sum\"))\n",
    "    .sort_values(\"total_aligned_reads\")\n",
    "    .query(\"total_aligned_reads < @mito_composition_filters['min_alignments_run_filter']\")\n",
    "    .reset_index(drop=True)\n",
    "    .merge(metadata, how=\"left\")\n",
    ")\n",
    "\n",
    "print(\"Excluding the following metagenomic runs with insufficient aligned mitochondrial reads:\")\n",
    "display(insufficient_mito_reads)\n",
    "\n",
    "mito_counts = mito_counts[\n",
    "    ~mito_counts[\"Run accession\"].isin(insufficient_mito_reads[\"Run accession\"])\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16e75ff2-dc43-4557-a64e-959ee29c7e41",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-04T13:43:25.177040Z",
     "iopub.status.busy": "2023-04-04T13:43:25.176357Z",
     "iopub.status.idle": "2023-04-04T13:43:25.184777Z",
     "shell.execute_reply": "2023-04-04T13:43:25.183760Z",
     "shell.execute_reply.started": "2023-04-04T13:43:25.176998Z"
    },
    "tags": []
   },
   "source": [
    "Now exclude mitochondrial reference genomes that don't have a high enough percentage of read alignments or are not delineated in the list to keep:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "89a56047-8ab0-43ad-bac8-79a6176b6848",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T19:15:55.021462Z",
     "iopub.status.busy": "2023-04-08T19:15:55.020938Z",
     "iopub.status.idle": "2023-04-08T19:15:55.685797Z",
     "shell.execute_reply": "2023-04-08T19:15:55.684834Z",
     "shell.execute_reply.started": "2023-04-08T19:15:55.021419Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping the following mitochondrial genomes:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alignment_reference</th>\n",
       "      <th>max_percent</th>\n",
       "      <th>avg_percent</th>\n",
       "      <th>max_coverage</th>\n",
       "      <th>species</th>\n",
       "      <th>common_name</th>\n",
       "      <th>reason_kept</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_012920.1</td>\n",
       "      <td>99.01</td>\n",
       "      <td>14.72</td>\n",
       "      <td>14000</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>human</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_009728.1</td>\n",
       "      <td>98.63</td>\n",
       "      <td>14.40</td>\n",
       "      <td>73</td>\n",
       "      <td>Osteolaemus tetraspis</td>\n",
       "      <td>dwarf crocodile</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_040902.1</td>\n",
       "      <td>97.71</td>\n",
       "      <td>11.26</td>\n",
       "      <td>16784</td>\n",
       "      <td>Gallus gallus spadiceus</td>\n",
       "      <td>chicken</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NC_022418.1</td>\n",
       "      <td>63.96</td>\n",
       "      <td>6.36</td>\n",
       "      <td>16567</td>\n",
       "      <td>Anas poecilorhyncha</td>\n",
       "      <td>spot-billed duck</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NC_030041.1</td>\n",
       "      <td>87.17</td>\n",
       "      <td>3.91</td>\n",
       "      <td>13056</td>\n",
       "      <td>Ptyas mucosa</td>\n",
       "      <td>Oriental rat snake</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NC_020011.1</td>\n",
       "      <td>77.69</td>\n",
       "      <td>3.58</td>\n",
       "      <td>15491</td>\n",
       "      <td>Channa maculata</td>\n",
       "      <td>snakehead fish</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NC_010288.1</td>\n",
       "      <td>68.76</td>\n",
       "      <td>3.35</td>\n",
       "      <td>16548</td>\n",
       "      <td>Ctenopharyngodon idella</td>\n",
       "      <td>carp</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NC_013700.1</td>\n",
       "      <td>75.49</td>\n",
       "      <td>2.82</td>\n",
       "      <td>15490</td>\n",
       "      <td>Nyctereutes procyonoides</td>\n",
       "      <td>raccoon dog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_006853.1</td>\n",
       "      <td>94.65</td>\n",
       "      <td>2.78</td>\n",
       "      <td>16338</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>cow</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NC_002008.4</td>\n",
       "      <td>80.41</td>\n",
       "      <td>2.52</td>\n",
       "      <td>14587</td>\n",
       "      <td>Canis lupus familiaris</td>\n",
       "      <td>dog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NC_026063.1</td>\n",
       "      <td>40.15</td>\n",
       "      <td>2.00</td>\n",
       "      <td>16597</td>\n",
       "      <td>Ovis orientalis</td>\n",
       "      <td>sheep</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NC_003489.1</td>\n",
       "      <td>70.86</td>\n",
       "      <td>1.80</td>\n",
       "      <td>16497</td>\n",
       "      <td>Ictalurus punctatus</td>\n",
       "      <td>catfish</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NC_026459.1</td>\n",
       "      <td>30.64</td>\n",
       "      <td>1.68</td>\n",
       "      <td>11097</td>\n",
       "      <td>Streptopelia chinensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 25% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NC_001913.1</td>\n",
       "      <td>42.83</td>\n",
       "      <td>1.54</td>\n",
       "      <td>15709</td>\n",
       "      <td>Oryctolagus cuniculus</td>\n",
       "      <td>rabbit</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NC_021478.1</td>\n",
       "      <td>45.60</td>\n",
       "      <td>1.41</td>\n",
       "      <td>13962</td>\n",
       "      <td>Rhizomys pruinosus</td>\n",
       "      <td>hoary bamboo rat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NC_012095.1</td>\n",
       "      <td>22.54</td>\n",
       "      <td>1.24</td>\n",
       "      <td>14559</td>\n",
       "      <td>Sus scrofa domesticus</td>\n",
       "      <td>pig</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NC_011710.1</td>\n",
       "      <td>42.19</td>\n",
       "      <td>1.11</td>\n",
       "      <td>11178</td>\n",
       "      <td>Larimichthys crocea</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 25% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>KX964606.1</td>\n",
       "      <td>56.26</td>\n",
       "      <td>1.01</td>\n",
       "      <td>8737</td>\n",
       "      <td>Erinaceus amurensis</td>\n",
       "      <td>Amur hedgehog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NC_003192.1</td>\n",
       "      <td>39.05</td>\n",
       "      <td>0.97</td>\n",
       "      <td>16527</td>\n",
       "      <td>Monopterus albus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 25% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NC_003408.1</td>\n",
       "      <td>47.74</td>\n",
       "      <td>0.79</td>\n",
       "      <td>14396</td>\n",
       "      <td>Coturnix japonica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 25% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_004926.1</td>\n",
       "      <td>98.83</td>\n",
       "      <td>0.68</td>\n",
       "      <td>12250</td>\n",
       "      <td>Andrias davidianus</td>\n",
       "      <td>Chinese salamander</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NC_023982.1</td>\n",
       "      <td>61.84</td>\n",
       "      <td>0.50</td>\n",
       "      <td>7361</td>\n",
       "      <td>Amaurornis akool</td>\n",
       "      <td>brown bush hen</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NC_050263.1</td>\n",
       "      <td>77.50</td>\n",
       "      <td>0.49</td>\n",
       "      <td>10483</td>\n",
       "      <td>Hystrix brachyura</td>\n",
       "      <td>Malayan porcupine</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NC_037513.1</td>\n",
       "      <td>8.71</td>\n",
       "      <td>0.35</td>\n",
       "      <td>1426</td>\n",
       "      <td>Streptopelia decaocto</td>\n",
       "      <td>collared dove</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NC_014576.1</td>\n",
       "      <td>9.69</td>\n",
       "      <td>0.31</td>\n",
       "      <td>1518</td>\n",
       "      <td>Chrysolophus pictus</td>\n",
       "      <td>golden pheasant</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NC_001700.1</td>\n",
       "      <td>6.71</td>\n",
       "      <td>0.29</td>\n",
       "      <td>4781</td>\n",
       "      <td>Felis catus</td>\n",
       "      <td>cat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NC_028161.1</td>\n",
       "      <td>3.82</td>\n",
       "      <td>0.24</td>\n",
       "      <td>7910</td>\n",
       "      <td>Capra aegagrus</td>\n",
       "      <td>goat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>17.73</td>\n",
       "      <td>0.22</td>\n",
       "      <td>9581</td>\n",
       "      <td>Rattus norvegicus strain Wistar</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NC_018367.1</td>\n",
       "      <td>28.02</td>\n",
       "      <td>0.22</td>\n",
       "      <td>14752</td>\n",
       "      <td>Marmota himalayana</td>\n",
       "      <td>Himalayan marmot</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NC_020645.1</td>\n",
       "      <td>6.21</td>\n",
       "      <td>0.13</td>\n",
       "      <td>6786</td>\n",
       "      <td>Arctonyx collaris</td>\n",
       "      <td>hog badger</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NC_002080.2</td>\n",
       "      <td>5.16</td>\n",
       "      <td>0.10</td>\n",
       "      <td>1162</td>\n",
       "      <td>Erinaceus europaeus</td>\n",
       "      <td>European hedgehog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NC_029888.1</td>\n",
       "      <td>4.39</td>\n",
       "      <td>0.09</td>\n",
       "      <td>1907</td>\n",
       "      <td>Rattus tiomanicus</td>\n",
       "      <td>Mayalan field rat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NC_042233.1</td>\n",
       "      <td>1.98</td>\n",
       "      <td>0.08</td>\n",
       "      <td>105</td>\n",
       "      <td>Numenius tenuirostris</td>\n",
       "      <td>curlew (bird)</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NC_028308.1</td>\n",
       "      <td>2.02</td>\n",
       "      <td>0.07</td>\n",
       "      <td>1394</td>\n",
       "      <td>Felis margarita</td>\n",
       "      <td>sand cat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NC_035814.1</td>\n",
       "      <td>1.26</td>\n",
       "      <td>0.05</td>\n",
       "      <td>97</td>\n",
       "      <td>Aonyx cinerea isolate 16</td>\n",
       "      <td>otter</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NC_023459.1</td>\n",
       "      <td>2.42</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1165</td>\n",
       "      <td>Vulpes zerda</td>\n",
       "      <td>fennec fox</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NC_020637.1</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.04</td>\n",
       "      <td>1484</td>\n",
       "      <td>Mustela sibirica</td>\n",
       "      <td>Siberian weasel</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NC_029403.1</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.03</td>\n",
       "      <td>5896</td>\n",
       "      <td>Paguma larvata</td>\n",
       "      <td>masked palm civet</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NC_036430.1</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.02</td>\n",
       "      <td>640</td>\n",
       "      <td>Muntiacus putaoensis</td>\n",
       "      <td>leaf deer</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>NC_005033.1</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.02</td>\n",
       "      <td>228</td>\n",
       "      <td>Hemiechinus auritus</td>\n",
       "      <td>long-eared hedgehog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NC_026723.1</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.01</td>\n",
       "      <td>179</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NC_041068.1</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.01</td>\n",
       "      <td>174</td>\n",
       "      <td>Elaphe dione</td>\n",
       "      <td>rat snake</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NC_069631.1</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>67</td>\n",
       "      <td>Terrapene carolina</td>\n",
       "      <td>box turtle</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NC_023528.1</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.00</td>\n",
       "      <td>191</td>\n",
       "      <td>Rana dybowski</td>\n",
       "      <td>Dybowski frog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NC_036369.1</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>133</td>\n",
       "      <td>Otocyon megalotis</td>\n",
       "      <td>bat-eared fox</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>NC_020001.1</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>400</td>\n",
       "      <td>Sooglossus thomasseti</td>\n",
       "      <td>frog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NC_008795.1</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "      <td>44</td>\n",
       "      <td>Crocodylus siamensis</td>\n",
       "      <td>Siamese crocodile</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NC_027671.1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>Platymantis vitianus</td>\n",
       "      <td>ground frog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   alignment_reference  max_percent  avg_percent  max_coverage  \\\n",
       "0          NC_012920.1        99.01        14.72         14000   \n",
       "2          NC_009728.1        98.63        14.40            73   \n",
       "3          NC_040902.1        97.71        11.26         16784   \n",
       "12         NC_022418.1        63.96         6.36         16567   \n",
       "5          NC_030041.1        87.17         3.91         13056   \n",
       "7          NC_020011.1        77.69         3.58         15491   \n",
       "11         NC_010288.1        68.76         3.35         16548   \n",
       "9          NC_013700.1        75.49         2.82         15490   \n",
       "4          NC_006853.1        94.65         2.78         16338   \n",
       "6          NC_002008.4        80.41         2.52         14587   \n",
       "19         NC_026063.1        40.15         2.00         16597   \n",
       "10         NC_003489.1        70.86         1.80         16497   \n",
       "21         NC_026459.1        30.64         1.68         11097   \n",
       "17         NC_001913.1        42.83         1.54         15709   \n",
       "16         NC_021478.1        45.60         1.41         13962   \n",
       "23         NC_012095.1        22.54         1.24         14559   \n",
       "18         NC_011710.1        42.19         1.11         11178   \n",
       "14          KX964606.1        56.26         1.01          8737   \n",
       "20         NC_003192.1        39.05         0.97         16527   \n",
       "15         NC_003408.1        47.74         0.79         14396   \n",
       "1          NC_004926.1        98.83         0.68         12250   \n",
       "13         NC_023982.1        61.84         0.50          7361   \n",
       "8          NC_050263.1        77.50         0.49         10483   \n",
       "26         NC_037513.1         8.71         0.35          1426   \n",
       "25         NC_014576.1         9.69         0.31          1518   \n",
       "27         NC_001700.1         6.71         0.29          4781   \n",
       "31         NC_028161.1         3.82         0.24          7910   \n",
       "24         AC_000022.2        17.73         0.22          9581   \n",
       "22         NC_018367.1        28.02         0.22         14752   \n",
       "28         NC_020645.1         6.21         0.13          6786   \n",
       "29         NC_002080.2         5.16         0.10          1162   \n",
       "30         NC_029888.1         4.39         0.09          1907   \n",
       "34         NC_042233.1         1.98         0.08           105   \n",
       "33         NC_028308.1         2.02         0.07          1394   \n",
       "35         NC_035814.1         1.26         0.05            97   \n",
       "32         NC_023459.1         2.42         0.04          1165   \n",
       "38         NC_020637.1         0.94         0.04          1484   \n",
       "37         NC_029403.1         0.96         0.03          5896   \n",
       "39         NC_036430.1         0.82         0.02           640   \n",
       "36         NC_005033.1         0.98         0.02           228   \n",
       "41         NC_026723.1         0.29         0.01           179   \n",
       "40         NC_041068.1         0.33         0.01           174   \n",
       "44         NC_069631.1         0.12         0.00            67   \n",
       "43         NC_023528.1         0.17         0.00           191   \n",
       "42         NC_036369.1         0.21         0.00           133   \n",
       "46         NC_020001.1         0.08         0.00           400   \n",
       "45         NC_008795.1         0.09         0.00            44   \n",
       "47         NC_027671.1         0.00         0.00             0   \n",
       "\n",
       "                            species          common_name  \\\n",
       "0                      Homo sapiens                human   \n",
       "2             Osteolaemus tetraspis      dwarf crocodile   \n",
       "3           Gallus gallus spadiceus              chicken   \n",
       "12              Anas poecilorhyncha     spot-billed duck   \n",
       "5                      Ptyas mucosa   Oriental rat snake   \n",
       "7                   Channa maculata       snakehead fish   \n",
       "11          Ctenopharyngodon idella                 carp   \n",
       "9          Nyctereutes procyonoides          raccoon dog   \n",
       "4                        Bos taurus                  cow   \n",
       "6            Canis lupus familiaris                  dog   \n",
       "19                  Ovis orientalis                sheep   \n",
       "10              Ictalurus punctatus              catfish   \n",
       "21           Streptopelia chinensis                  NaN   \n",
       "17            Oryctolagus cuniculus               rabbit   \n",
       "16               Rhizomys pruinosus     hoary bamboo rat   \n",
       "23            Sus scrofa domesticus                  pig   \n",
       "18              Larimichthys crocea                  NaN   \n",
       "14              Erinaceus amurensis        Amur hedgehog   \n",
       "20                 Monopterus albus                  NaN   \n",
       "15                Coturnix japonica                  NaN   \n",
       "1                Andrias davidianus   Chinese salamander   \n",
       "13                 Amaurornis akool       brown bush hen   \n",
       "8                 Hystrix brachyura    Malayan porcupine   \n",
       "26            Streptopelia decaocto        collared dove   \n",
       "25              Chrysolophus pictus      golden pheasant   \n",
       "27                      Felis catus                  cat   \n",
       "31                   Capra aegagrus                 goat   \n",
       "24  Rattus norvegicus strain Wistar            brown rat   \n",
       "22               Marmota himalayana     Himalayan marmot   \n",
       "28                Arctonyx collaris           hog badger   \n",
       "29              Erinaceus europaeus    European hedgehog   \n",
       "30                Rattus tiomanicus    Mayalan field rat   \n",
       "34            Numenius tenuirostris        curlew (bird)   \n",
       "33                  Felis margarita             sand cat   \n",
       "35         Aonyx cinerea isolate 16                otter   \n",
       "32                     Vulpes zerda           fennec fox   \n",
       "38                 Mustela sibirica      Siberian weasel   \n",
       "37                   Paguma larvata    masked palm civet   \n",
       "39             Muntiacus putaoensis            leaf deer   \n",
       "36              Hemiechinus auritus  long-eared hedgehog   \n",
       "41         Urocyon cinereoargenteus             gray fox   \n",
       "40                     Elaphe dione            rat snake   \n",
       "44               Terrapene carolina           box turtle   \n",
       "43                    Rana dybowski        Dybowski frog   \n",
       "42                Otocyon megalotis        bat-eared fox   \n",
       "46            Sooglossus thomasseti                 frog   \n",
       "45             Crocodylus siamensis    Siamese crocodile   \n",
       "47             Platymantis vitianus          ground frog   \n",
       "\n",
       "                    reason_kept  \n",
       "0   specified to keep in config  \n",
       "2   specified to keep in config  \n",
       "3   specified to keep in config  \n",
       "12  specified to keep in config  \n",
       "5   specified to keep in config  \n",
       "7   specified to keep in config  \n",
       "11  specified to keep in config  \n",
       "9   specified to keep in config  \n",
       "4   specified to keep in config  \n",
       "6   specified to keep in config  \n",
       "19  specified to keep in config  \n",
       "10  specified to keep in config  \n",
       "21      at least 25% in one run  \n",
       "17  specified to keep in config  \n",
       "16  specified to keep in config  \n",
       "23  specified to keep in config  \n",
       "18      at least 25% in one run  \n",
       "14  specified to keep in config  \n",
       "20      at least 25% in one run  \n",
       "15      at least 25% in one run  \n",
       "1   specified to keep in config  \n",
       "13  specified to keep in config  \n",
       "8   specified to keep in config  \n",
       "26  specified to keep in config  \n",
       "25  specified to keep in config  \n",
       "27  specified to keep in config  \n",
       "31  specified to keep in config  \n",
       "24  specified to keep in config  \n",
       "22  specified to keep in config  \n",
       "28  specified to keep in config  \n",
       "29  specified to keep in config  \n",
       "30  specified to keep in config  \n",
       "34  specified to keep in config  \n",
       "33  specified to keep in config  \n",
       "35  specified to keep in config  \n",
       "32  specified to keep in config  \n",
       "38  specified to keep in config  \n",
       "37  specified to keep in config  \n",
       "39  specified to keep in config  \n",
       "36  specified to keep in config  \n",
       "41  specified to keep in config  \n",
       "40  specified to keep in config  \n",
       "44  specified to keep in config  \n",
       "43  specified to keep in config  \n",
       "42  specified to keep in config  \n",
       "46  specified to keep in config  \n",
       "45  specified to keep in config  \n",
       "47  specified to keep in config  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mito_max_primary_percent = ( \n",
    "    mito_counts\n",
    "    .assign(\n",
    "        total=lambda x: x.groupby(\"Run accession\")[\"aligned_reads\"].transform(\"sum\"),\n",
    "        percent=lambda x: 100 * x[\"aligned_reads\"] / x[\"total\"],\n",
    "    )\n",
    "    .groupby(\"alignment_reference\", as_index=False)\n",
    "    .aggregate(\n",
    "        max_percent=pd.NamedAgg(\"percent\", \"max\"),\n",
    "        avg_percent=pd.NamedAgg(\"percent\", \"mean\"),\n",
    "        max_coverage=pd.NamedAgg(\"covered_bases\", \"max\"),\n",
    "    )\n",
    "    .sort_values(\"max_percent\")\n",
    "    .merge(mito_ref_info.rename(columns={\"id\": \"alignment_reference\"}))\n",
    ")\n",
    "\n",
    "min_percent = mito_composition_filters[\"min_percent_aligned_genome_filter\"]\n",
    "mito_to_keep = (\n",
    "    mito_max_primary_percent\n",
    "    .assign(\n",
    "        reason_kept=lambda x: x.apply(\n",
    "            lambda row: (\n",
    "                \"specified to keep in config\"\n",
    "                if any(row[\"species\"].startswith(s) for s in mito_genomes_to_keep.values())\n",
    "                else (\n",
    "                    f\"at least {min_percent}% in one run\"\n",
    "                    if row[\"max_percent\"] >= min_percent\n",
    "                    else\n",
    "                    \"not retained\"\n",
    "                )\n",
    "            ),\n",
    "            axis=1,\n",
    "        ),\n",
    "    )\n",
    "    .query(\"reason_kept != 'not retained'\")\n",
    "    .sort_values(\"max_percent\", ascending=False)\n",
    "    .reset_index(drop=True)\n",
    ")\n",
    "\n",
    "assert all(\n",
    "    any(s_kept.startswith(s_to_keep) for s_kept in mito_to_keep[\"species\"])\n",
    "    for s_to_keep in mito_genomes_to_keep.values()\n",
    ")\n",
    "\n",
    "print(\"Keeping the following mitochondrial genomes:\")\n",
    "display(mito_to_keep.sort_values(\"avg_percent\", ascending=False).round(2))\n",
    "\n",
    "mito_ids_to_keep = mito_to_keep[\"alignment_reference\"].tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a92f6fe-1de8-4418-a325-3588589e31c3",
   "metadata": {},
   "source": [
    "Now get just the counts for those mitochondrial genomes to keep, aggregating all other counts to \"other\" and pivoting / melting to add zero counts:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "5efa6705-0c17-4a85-a8bc-f13c34cbb9c6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T19:25:20.663245Z",
     "iopub.status.busy": "2023-04-08T19:25:20.662362Z",
     "iopub.status.idle": "2023-04-08T19:25:21.408610Z",
     "shell.execute_reply": "2023-04-08T19:25:21.407646Z",
     "shell.execute_reply.started": "2023-04-08T19:25:20.663200Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run accession</th>\n",
       "      <th>reference_id</th>\n",
       "      <th>aligned_reads</th>\n",
       "      <th>species</th>\n",
       "      <th>common_name</th>\n",
       "      <th>SARS2_aligned_reads</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>preprocessed_reads</th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>description</th>\n",
       "      <th>Isolation source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>1</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>3</td>\n",
       "      <td>106545428</td>\n",
       "      <td>90854684</td>\n",
       "      <td>Q37</td>\n",
       "      <td>Env_0552</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>KX964606.1</td>\n",
       "      <td>73</td>\n",
       "      <td>Erinaceus amurensis</td>\n",
       "      <td>Amur hedgehog</td>\n",
       "      <td>3</td>\n",
       "      <td>106545428</td>\n",
       "      <td>90854684</td>\n",
       "      <td>Q37</td>\n",
       "      <td>Env_0552</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>NC_001700.1</td>\n",
       "      <td>0</td>\n",
       "      <td>Felis catus</td>\n",
       "      <td>cat</td>\n",
       "      <td>3</td>\n",
       "      <td>106545428</td>\n",
       "      <td>90854684</td>\n",
       "      <td>Q37</td>\n",
       "      <td>Env_0552</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>NC_001913.1</td>\n",
       "      <td>0</td>\n",
       "      <td>Oryctolagus cuniculus</td>\n",
       "      <td>rabbit</td>\n",
       "      <td>3</td>\n",
       "      <td>106545428</td>\n",
       "      <td>90854684</td>\n",
       "      <td>Q37</td>\n",
       "      <td>Env_0552</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>NC_002008.4</td>\n",
       "      <td>1</td>\n",
       "      <td>Canis lupus</td>\n",
       "      <td>dog</td>\n",
       "      <td>3</td>\n",
       "      <td>106545428</td>\n",
       "      <td>90854684</td>\n",
       "      <td>Q37</td>\n",
       "      <td>Env_0552</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19301</th>\n",
       "      <td>CRR712210</td>\n",
       "      <td>NC_041068.1</td>\n",
       "      <td>0</td>\n",
       "      <td>Elaphe dione</td>\n",
       "      <td>rat snake</td>\n",
       "      <td>0</td>\n",
       "      <td>151171387</td>\n",
       "      <td>25617933</td>\n",
       "      <td>1-27-11</td>\n",
       "      <td>Env_0642</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19302</th>\n",
       "      <td>CRR712210</td>\n",
       "      <td>NC_042233.1</td>\n",
       "      <td>2</td>\n",
       "      <td>Numenius tenuirostris</td>\n",
       "      <td>curlew (bird)</td>\n",
       "      <td>0</td>\n",
       "      <td>151171387</td>\n",
       "      <td>25617933</td>\n",
       "      <td>1-27-11</td>\n",
       "      <td>Env_0642</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19303</th>\n",
       "      <td>CRR712210</td>\n",
       "      <td>NC_050263.1</td>\n",
       "      <td>0</td>\n",
       "      <td>Hystrix brachyura</td>\n",
       "      <td>Malayan porcupine</td>\n",
       "      <td>0</td>\n",
       "      <td>151171387</td>\n",
       "      <td>25617933</td>\n",
       "      <td>1-27-11</td>\n",
       "      <td>Env_0642</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19304</th>\n",
       "      <td>CRR712210</td>\n",
       "      <td>NC_069631.1</td>\n",
       "      <td>0</td>\n",
       "      <td>Terrapene carolina</td>\n",
       "      <td>box turtle</td>\n",
       "      <td>0</td>\n",
       "      <td>151171387</td>\n",
       "      <td>25617933</td>\n",
       "      <td>1-27-11</td>\n",
       "      <td>Env_0642</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19305</th>\n",
       "      <td>CRR712210</td>\n",
       "      <td>other</td>\n",
       "      <td>18</td>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>0</td>\n",
       "      <td>151171387</td>\n",
       "      <td>25617933</td>\n",
       "      <td>1-27-11</td>\n",
       "      <td>Env_0642</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19306 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Run accession reference_id  aligned_reads                species  \\\n",
       "0         CRR710761  AC_000022.2              1      Rattus norvegicus   \n",
       "1         CRR710761   KX964606.1             73    Erinaceus amurensis   \n",
       "2         CRR710761  NC_001700.1              0            Felis catus   \n",
       "3         CRR710761  NC_001913.1              0  Oryctolagus cuniculus   \n",
       "4         CRR710761  NC_002008.4              1            Canis lupus   \n",
       "...             ...          ...            ...                    ...   \n",
       "19301     CRR712210  NC_041068.1              0           Elaphe dione   \n",
       "19302     CRR712210  NC_042233.1              2  Numenius tenuirostris   \n",
       "19303     CRR712210  NC_050263.1              0      Hystrix brachyura   \n",
       "19304     CRR712210  NC_069631.1              0     Terrapene carolina   \n",
       "19305     CRR712210        other             18                  other   \n",
       "\n",
       "             common_name  SARS2_aligned_reads  total_reads  \\\n",
       "0              brown rat                    3    106545428   \n",
       "1          Amur hedgehog                    3    106545428   \n",
       "2                    cat                    3    106545428   \n",
       "3                 rabbit                    3    106545428   \n",
       "4                    dog                    3    106545428   \n",
       "...                  ...                  ...          ...   \n",
       "19301          rat snake                    0    151171387   \n",
       "19302      curlew (bird)                    0    151171387   \n",
       "19303  Malayan porcupine                    0    151171387   \n",
       "19304         box turtle                    0    151171387   \n",
       "19305              other                    0    151171387   \n",
       "\n",
       "       preprocessed_reads   sample Sample name Collection date  \\\n",
       "0                90854684      Q37    Env_0552      2020-01-12   \n",
       "1                90854684      Q37    Env_0552      2020-01-12   \n",
       "2                90854684      Q37    Env_0552      2020-01-12   \n",
       "3                90854684      Q37    Env_0552      2020-01-12   \n",
       "4                90854684      Q37    Env_0552      2020-01-12   \n",
       "...                   ...      ...         ...             ...   \n",
       "19301            25617933  1-27-11    Env_0642      2020-01-27   \n",
       "19302            25617933  1-27-11    Env_0642      2020-01-27   \n",
       "19303            25617933  1-27-11    Env_0642      2020-01-27   \n",
       "19304            25617933  1-27-11    Env_0642      2020-01-27   \n",
       "19305            25617933  1-27-11    Env_0642      2020-01-27   \n",
       "\n",
       "                                             description  \\\n",
       "0      RNA sequencing of total nucleic acids from env...   \n",
       "1      RNA sequencing of total nucleic acids from env...   \n",
       "2      RNA sequencing of total nucleic acids from env...   \n",
       "3      RNA sequencing of total nucleic acids from env...   \n",
       "4      RNA sequencing of total nucleic acids from env...   \n",
       "...                                                  ...   \n",
       "19301  RNA sequencing of total nucleic acids from env...   \n",
       "19302  RNA sequencing of total nucleic acids from env...   \n",
       "19303  RNA sequencing of total nucleic acids from env...   \n",
       "19304  RNA sequencing of total nucleic acids from env...   \n",
       "19305  RNA sequencing of total nucleic acids from env...   \n",
       "\n",
       "                                  Isolation source  \n",
       "0                                 West Wine of HSM  \n",
       "1                                 West Wine of HSM  \n",
       "2                                 West Wine of HSM  \n",
       "3                                 West Wine of HSM  \n",
       "4                                 West Wine of HSM  \n",
       "...                                            ...  \n",
       "19301  West Wine of HSM (Sewers or sewerage wells)  \n",
       "19302  West Wine of HSM (Sewers or sewerage wells)  \n",
       "19303  West Wine of HSM (Sewers or sewerage wells)  \n",
       "19304  West Wine of HSM (Sewers or sewerage wells)  \n",
       "19305  West Wine of HSM (Sewers or sewerage wells)  \n",
       "\n",
       "[19306 rows x 13 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mito_counts_by_metagenomic_run = (\n",
    "    counts_and_coverage\n",
    "    [[\"Run accession\", \"alignment_reference\", \"aligned_reads\"]]\n",
    "    # aggregate all other counts as \"other\"\n",
    "    .assign(\n",
    "        alignment_reference=lambda x: x[\"alignment_reference\"].where(\n",
    "            x[\"alignment_reference\"].isin(mito_ids_to_keep), \"other\"\n",
    "        ),\n",
    "    )\n",
    "    .groupby([\"Run accession\", \"alignment_reference\"], as_index=False)\n",
    "    .aggregate(aligned_reads=pd.NamedAgg(\"aligned_reads\", \"sum\"))\n",
    "    .rename(columns={\"alignment_reference\": \"reference_id\"})\n",
    "    # add more information on references, and shorten species name to first two words\n",
    "    .merge(mito_ref_info.rename(columns={\"id\": \"reference_id\"}), how=\"left\")\n",
    "    .assign(\n",
    "        species=lambda x: numpy.where(\n",
    "            x[\"reference_id\"] == \"other\",\n",
    "            \"other\",\n",
    "            x[\"species\"].map(lambda s: \" \".join(s.split()[: 2]) if not pd.isnull(s) else s),\n",
    "        ),\n",
    "        common_name=lambda x: x[\"common_name\"].where(\n",
    "            x[\"reference_id\"] != \"other\", \"other\",\n",
    "        ),\n",
    "    )\n",
    "    # add other information\n",
    "    .merge(\n",
    "        sars2_aligned_by_run\n",
    "        .drop(columns=[\"percent_preprocessed_reads_aligning_to_SARS2\", \"SARS2_covered_bases\"]),\n",
    "        how=\"left\",\n",
    "    )\n",
    ")\n",
    "\n",
    "mito_counts_by_metagenomic_run.to_csv(\n",
    "    snakemake.output[\"mito_composition_by_metagenomic_run\"], index=False, float_format=\"%.5f\",\n",
    ")\n",
    "\n",
    "# debugging\n",
    "assert mito_counts_by_metagenomic_run.drop(columns=\"common_name\").notnull().all().all(), mito_counts_by_metagenomic_run.notnull().all()\n",
    "\n",
    "mito_counts_by_metagenomic_run"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "917eb8e5-1903-4916-b4cb-bc32cc87906c",
   "metadata": {
    "execution": {
     "iopub.status.busy": "2023-04-05T20:15:01.703158Z",
     "iopub.status.idle": "2023-04-05T20:15:01.704024Z",
     "shell.execute_reply": "2023-04-05T20:15:01.703776Z",
     "shell.execute_reply.started": "2023-04-05T20:15:01.703749Z"
    },
    "tags": []
   },
   "source": [
    "Now we get the mitochondrial counts by **sample** (aggregating counts across run for the same sample:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "8c83b527-11f8-4e80-9376-80addc2ceb38",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T19:26:15.898670Z",
     "iopub.status.busy": "2023-04-08T19:26:15.897987Z",
     "iopub.status.idle": "2023-04-08T19:26:16.445276Z",
     "shell.execute_reply": "2023-04-08T19:26:16.444108Z",
     "shell.execute_reply.started": "2023-04-08T19:26:15.898626Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reference_id</th>\n",
       "      <th>species</th>\n",
       "      <th>common_name</th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>description</th>\n",
       "      <th>Isolation source</th>\n",
       "      <th>aligned_reads</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>preprocessed_reads</th>\n",
       "      <th>SARS2_aligned_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>06-29-abv1</td>\n",
       "      <td>Env_0620</td>\n",
       "      <td>2020-01-23</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>3</td>\n",
       "      <td>1079014964</td>\n",
       "      <td>851849944</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>1-27-11</td>\n",
       "      <td>Env_0642</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "      <td>8</td>\n",
       "      <td>821800418</td>\n",
       "      <td>571936627</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>1-27-12</td>\n",
       "      <td>Env_0643</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "      <td>3</td>\n",
       "      <td>2038484394</td>\n",
       "      <td>2029662756</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>1-27-28</td>\n",
       "      <td>Env_0657</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "      <td>0</td>\n",
       "      <td>290477942</td>\n",
       "      <td>289351106</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>1-27-33</td>\n",
       "      <td>Env_0660</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "      <td>0</td>\n",
       "      <td>92724742</td>\n",
       "      <td>92467336</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8619</th>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>YCHC2-1-2</td>\n",
       "      <td>Env_0798</td>\n",
       "      <td>2020-02-09</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>Sewerage well in the surrounding areas</td>\n",
       "      <td>36</td>\n",
       "      <td>868053694</td>\n",
       "      <td>865532062</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8620</th>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>YCHC2-1-3</td>\n",
       "      <td>Env_0799</td>\n",
       "      <td>2020-02-09</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>Sewerage well in the surrounding areas</td>\n",
       "      <td>11</td>\n",
       "      <td>662632390</td>\n",
       "      <td>660721662</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8621</th>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>w-6-29-33</td>\n",
       "      <td>Env_0828</td>\n",
       "      <td>2020-02-15</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "      <td>77</td>\n",
       "      <td>367737968</td>\n",
       "      <td>366257533</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8622</th>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>w-zong-1</td>\n",
       "      <td>Env_0829</td>\n",
       "      <td>2020-02-15</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "      <td>204</td>\n",
       "      <td>433163214</td>\n",
       "      <td>431193910</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8623</th>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>other</td>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>Env_0830</td>\n",
       "      <td>2020-02-15</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "      <td>306</td>\n",
       "      <td>508197563</td>\n",
       "      <td>506059696</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8624 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     reference_id            species common_name      sample Sample name  \\\n",
       "0     AC_000022.2  Rattus norvegicus   brown rat  06-29-abv1    Env_0620   \n",
       "1     AC_000022.2  Rattus norvegicus   brown rat     1-27-11    Env_0642   \n",
       "2     AC_000022.2  Rattus norvegicus   brown rat     1-27-12    Env_0643   \n",
       "3     AC_000022.2  Rattus norvegicus   brown rat     1-27-28    Env_0657   \n",
       "4     AC_000022.2  Rattus norvegicus   brown rat     1-27-33    Env_0660   \n",
       "...           ...                ...         ...         ...         ...   \n",
       "8619        other              other       other   YCHC2-1-2    Env_0798   \n",
       "8620        other              other       other   YCHC2-1-3    Env_0799   \n",
       "8621        other              other       other   w-6-29-33    Env_0828   \n",
       "8622        other              other       other    w-zong-1    Env_0829   \n",
       "8623        other              other       other    wws-1#-0    Env_0830   \n",
       "\n",
       "     Collection date                                        description  \\\n",
       "0         2020-01-23  RNA sequencing of total nucleic acids from env...   \n",
       "1         2020-01-27  RNA sequencing of total nucleic acids from env...   \n",
       "2         2020-01-27  RNA sequencing of total nucleic acids from env...   \n",
       "3         2020-01-27  RNA sequencing of total nucleic acids from env...   \n",
       "4         2020-01-27  RNA sequencing of total nucleic acids from env...   \n",
       "...              ...                                                ...   \n",
       "8619      2020-02-09  RNA sequencing of total nucleic acids from env...   \n",
       "8620      2020-02-09  RNA sequencing of total nucleic acids from env...   \n",
       "8621      2020-02-15  RNA sequencing of total nucleic acids from env...   \n",
       "8622      2020-02-15  RNA sequencing of total nucleic acids from env...   \n",
       "8623      2020-02-15  RNA sequencing of total nucleic acids from env...   \n",
       "\n",
       "                                 Isolation source  aligned_reads  total_reads  \\\n",
       "0                                West Wine of HSM              3   1079014964   \n",
       "1     West Wine of HSM (Sewers or sewerage wells)              8    821800418   \n",
       "2     West Wine of HSM (Sewers or sewerage wells)              3   2038484394   \n",
       "3     West Wine of HSM (Sewers or sewerage wells)              0    290477942   \n",
       "4     West Wine of HSM (Sewers or sewerage wells)              0     92724742   \n",
       "...                                           ...            ...          ...   \n",
       "8619       Sewerage well in the surrounding areas             36    868053694   \n",
       "8620       Sewerage well in the surrounding areas             11    662632390   \n",
       "8621  West Wine of HSM (Sewers or sewerage wells)             77    367737968   \n",
       "8622  West Wine of HSM (Sewers or sewerage wells)            204    433163214   \n",
       "8623  West Wine of HSM (Sewers or sewerage wells)            306    508197563   \n",
       "\n",
       "      preprocessed_reads  SARS2_aligned_reads  \n",
       "0              851849944                    0  \n",
       "1              571936627                    0  \n",
       "2             2029662756                    0  \n",
       "3              289351106                    0  \n",
       "4               92467336                    0  \n",
       "...                  ...                  ...  \n",
       "8619           865532062                    0  \n",
       "8620           660721662                    0  \n",
       "8621           366257533                    0  \n",
       "8622           431193910                    0  \n",
       "8623           506059696                    0  \n",
       "\n",
       "[8624 rows x 12 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# columns that we sum across runs for each sample\n",
    "sum_cols = [\n",
    "    \"aligned_reads\",\n",
    "    \"total_reads\",\n",
    "    \"preprocessed_reads\",\n",
    "    \"SARS2_aligned_reads\",\n",
    "]\n",
    "\n",
    "mito_counts_by_metagenomic_sample = (\n",
    "    mito_counts_by_metagenomic_run\n",
    "    .groupby(\n",
    "        [\n",
    "            c for c in mito_counts_by_metagenomic_run\n",
    "            if c not in sum_cols\n",
    "            and c not in {\"Run accession\"}\n",
    "        ],\n",
    "        as_index=False,\n",
    "        dropna=False,\n",
    "    )\n",
    "    .aggregate({c: \"sum\" for c in sum_cols})\n",
    ")\n",
    "\n",
    "dup_samples = (\n",
    "    mito_counts_by_metagenomic_sample\n",
    "    .groupby([\"reference_id\", \"sample\"])\n",
    "    .aggregate(\n",
    "        n=pd.NamedAgg(\"aligned_reads\", \"count\"),\n",
    "        sample_names=pd.NamedAgg(\"Sample name\", \"unique\"),\n",
    "    )\n",
    "    .query(\"n > 1\")\n",
    ")\n",
    "\n",
    "assert not len(dup_samples), f\"Some duplicated sample names\\n{dup_samples}\"\n",
    "\n",
    "mito_counts_by_metagenomic_sample.to_csv(\n",
    "    snakemake.output[\"mito_composition_by_metagenomic_sample\"], index=False, float_format=\"%.5g\",\n",
    ")\n",
    "\n",
    "mito_counts_by_metagenomic_sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cae22a2c-4c9f-4af9-b9f1-661e6998bc72",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
