{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aa01eedb-fcbd-45b5-a642-c308b717a25c",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Aggregate all the various counts and metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "320f9987-4dea-4504-b5a2-cfb6f4aa363e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:20.650206Z",
     "iopub.status.busy": "2023-04-07T04:03:20.649583Z",
     "iopub.status.idle": "2023-04-07T04:03:20.674380Z",
     "shell.execute_reply": "2023-04-07T04:03:20.673262Z",
     "shell.execute_reply.started": "2023-04-07T04:03:20.650157Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "######## snakemake preamble start (automatically inserted, do not edit) ########\n",
    "import sys; sys.path.extend(['/fh/fast/bloom_j/software/miniconda3/envs/Huanan_market_samples/lib/python3.10/site-packages', '/home/jbloom/.cache/snakemake/snakemake/source-cache/runtime-cache/tmpq32agzxm/file/fh/fast/bloom_j/computational_notebooks/jbloom/2023/Huanan_market_samples/notebooks', '/fh/fast/bloom_j/computational_notebooks/jbloom/2023/Huanan_market_samples/notebooks']); import pickle; snakemake = pickle.loads(b'\\x80\\x04\\x95\\xacb\\x00\\x00\\x00\\x00\\x00\\x00\\x8c\\x10snakemake.script\\x94\\x8c\\tSnakemake\\x94\\x93\\x94)\\x81\\x94}\\x94(\\x8c\\x05input\\x94\\x8c\\x0csnakemake.io\\x94\\x8c\\nInputFiles\\x94\\x93\\x94)\\x81\\x94(\\x8c.results/tallied_alignment_counts/CRR710761.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710762.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710763.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710764.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710765.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710766.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710767.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710768.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710769.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710770.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710771.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710772.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710773.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710774.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710775.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710776.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710777.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710778.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710779.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710780.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710781.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710782.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710783.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710784.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710785.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710786.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710787.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710788.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710789.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711156.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710790.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710791.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710792.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710793.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711157.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710794.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710795.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710796.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710797.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711158.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710798.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710799.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710800.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710801.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711159.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710802.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710803.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710804.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710805.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710806.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710807.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711160.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710808.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710809.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710810.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710811.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711161.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710812.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710813.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710814.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710815.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710816.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710817.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710818.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710819.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710820.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710821.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710822.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710823.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710824.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710825.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710826.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710827.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710828.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710829.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710830.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710831.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710832.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710833.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710834.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710835.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711143.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710836.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710837.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710838.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710839.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710840.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710841.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710842.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710843.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710844.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710845.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710846.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710847.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710848.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710849.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710850.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710851.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710852.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710853.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710854.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710855.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710856.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710857.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710858.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710859.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710860.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710861.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710862.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710863.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710864.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710865.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710866.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710867.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710868.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710869.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710870.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710871.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710872.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710873.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710874.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710875.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710876.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710877.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710878.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710879.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710880.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710881.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710882.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710883.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710884.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710885.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710886.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710887.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710888.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710889.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710890.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710891.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710892.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710893.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710894.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710895.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710896.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710897.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710898.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710899.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710900.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710901.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710902.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710903.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710904.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710905.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710906.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710907.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710908.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710909.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710910.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710911.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710912.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710913.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710914.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710915.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710916.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710917.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710918.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710919.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710920.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710921.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710922.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710923.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710924.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710925.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710926.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710927.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710928.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710929.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710930.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710931.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710932.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710933.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710934.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710935.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710936.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710937.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710938.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710939.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710940.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710941.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710942.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710943.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710944.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710945.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710946.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710947.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710948.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710949.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710950.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710951.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710952.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710953.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710954.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710955.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710956.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710957.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710958.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710959.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710960.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710961.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710962.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710963.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710964.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710965.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710966.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710967.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710968.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710969.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710970.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710971.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710972.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710973.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710974.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710975.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710976.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710977.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710978.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710979.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710980.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710981.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710982.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710983.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710984.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710985.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710986.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710987.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710988.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710989.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710990.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710991.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710992.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710993.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710994.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710995.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710996.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710997.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710998.csv\\x94\\x8c.results/tallied_alignment_counts/CRR710999.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711000.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711001.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711002.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711003.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711004.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711005.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711006.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711007.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711008.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711009.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711010.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711011.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711012.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711013.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711014.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711015.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711016.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711017.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711018.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711019.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711020.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711021.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711022.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711023.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711024.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711025.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711026.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711027.csv\\x94\\x8c.results/tallied_alignment_counts/CRR712209.csv\\x94\\x8c.results/tallied_alignment_counts/CRR712210.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711029.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711030.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711031.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711032.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711033.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711034.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711035.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711036.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711037.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711038.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711039.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711040.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711041.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711042.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711043.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711044.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711045.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711046.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711047.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711048.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711049.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711050.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711051.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711052.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711053.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711054.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711055.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711056.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711057.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711058.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711059.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711060.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711061.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711062.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711063.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711064.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711065.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711066.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711067.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711068.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711069.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711070.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711071.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711072.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711073.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711074.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711075.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711076.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711077.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711078.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711079.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711080.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711081.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711082.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711083.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711084.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711085.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711086.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711087.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711088.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711089.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711090.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711091.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711092.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711093.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711094.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711095.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711096.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711097.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711098.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711099.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711100.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711101.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711102.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711103.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711104.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711105.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711106.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711107.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711108.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711109.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711110.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711111.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711112.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711113.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711114.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711115.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711116.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711117.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711118.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711119.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711120.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711121.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711122.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711123.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711124.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711125.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711126.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711127.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711128.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711129.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711130.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711131.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711132.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711133.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711134.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711135.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711136.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711137.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711138.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711139.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711140.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711141.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711142.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711152.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711153.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711154.csv\\x94\\x8c.results/tallied_alignment_counts/CRR711155.csv\\x94\\x8c*results/mitochondrial_genomes/retained.csv\\x94\\x8c$results/metadata/merged_metadata.csv\\x94\\x8c)results/read_counts/aggregated_counts.csv\\x94e}\\x94(\\x8c\\x06_names\\x94}\\x94(\\x8c\\x0etallied_counts\\x94K\\x00M\\x8a\\x01\\x86\\x94\\x8c\\rmito_ref_info\\x94M\\x8a\\x01N\\x86\\x94\\x8c\\x08metadata\\x94M\\x8b\\x01N\\x86\\x94\\x8c\\x0bread_counts\\x94M\\x8c\\x01N\\x86\\x94u\\x8c\\x12_allowed_overrides\\x94]\\x94(\\x8c\\x05index\\x94\\x8c\\x04sort\\x94ej\\xa4\\x01\\x00\\x00\\x8c\\tfunctools\\x94\\x8c\\x07partial\\x94\\x93\\x94h\\x06\\x8c\\x19Namedlist._used_attribute\\x94\\x93\\x94\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94\\x8c\\x05_name\\x94j\\xa4\\x01\\x00\\x00sNt\\x94bj\\xa5\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00sNt\\x94bj\\x9a\\x01\\x00\\x00h\\x06\\x8c\\tNamedlist\\x94\\x93\\x94)\\x81\\x94(h\\nh\\x0bh\\x0ch\\rh\\x0eh\\x0fh\\x10h\\x11h\\x12h\\x13h\\x14h\\x15h\\x16h\\x17h\\x18h\\x19h\\x1ah\\x1bh\\x1ch\\x1dh\\x1eh\\x1fh h!h\"h#h$h%h&h\\'h(h)h*h+h,h-h.h/h0h1h2h3h4h5h6h7h8h9h:h;h<h=h>h?h@hAhBhChDhEhFhGhHhIhJhKhLhMhNhOhPhQhRhShThUhVhWhXhYhZh[h\\\\h]h^h_h`hahbhchdhehfhghhhihjhkhlhmhnhohphqhrhshthuhvhwhxhyhzh{h|h}h~h\\x7fh\\x80h\\x81h\\x82h\\x83h\\x84h\\x85h\\x86h\\x87h\\x88h\\x89h\\x8ah\\x8bh\\x8ch\\x8dh\\x8eh\\x8fh\\x90h\\x91h\\x92h\\x93h\\x94h\\x95h\\x96h\\x97h\\x98h\\x99h\\x9ah\\x9bh\\x9ch\\x9dh\\x9eh\\x9fh\\xa0h\\xa1h\\xa2h\\xa3h\\xa4h\\xa5h\\xa6h\\xa7h\\xa8h\\xa9h\\xaah\\xabh\\xach\\xadh\\xaeh\\xafh\\xb0h\\xb1h\\xb2h\\xb3h\\xb4h\\xb5h\\xb6h\\xb7h\\xb8h\\xb9h\\xbah\\xbbh\\xbch\\xbdh\\xbeh\\xbfh\\xc0h\\xc1h\\xc2h\\xc3h\\xc4h\\xc5h\\xc6h\\xc7h\\xc8h\\xc9h\\xcah\\xcbh\\xcch\\xcdh\\xceh\\xcfh\\xd0h\\xd1h\\xd2h\\xd3h\\xd4h\\xd5h\\xd6h\\xd7h\\xd8h\\xd9h\\xdah\\xdbh\\xdch\\xddh\\xdeh\\xdfh\\xe0h\\xe1h\\xe2h\\xe3h\\xe4h\\xe5h\\xe6h\\xe7h\\xe8h\\xe9h\\xeah\\xebh\\xech\\xedh\\xeeh\\xefh\\xf0h\\xf1h\\xf2h\\xf3h\\xf4h\\xf5h\\xf6h\\xf7h\\xf8h\\xf9h\\xfah\\xfbh\\xfch\\xfdh\\xfeh\\xffj\\x00\\x01\\x00\\x00j\\x01\\x01\\x00\\x00j\\x02\\x01\\x00\\x00j\\x03\\x01\\x00\\x00j\\x04\\x01\\x00\\x00j\\x05\\x01\\x00\\x00j\\x06\\x01\\x00\\x00j\\x07\\x01\\x00\\x00j\\x08\\x01\\x00\\x00j\\t\\x01\\x00\\x00j\\n\\x01\\x00\\x00j\\x0b\\x01\\x00\\x00j\\x0c\\x01\\x00\\x00j\\r\\x01\\x00\\x00j\\x0e\\x01\\x00\\x00j\\x0f\\x01\\x00\\x00j\\x10\\x01\\x00\\x00j\\x11\\x01\\x00\\x00j\\x12\\x01\\x00\\x00j\\x13\\x01\\x00\\x00j\\x14\\x01\\x00\\x00j\\x15\\x01\\x00\\x00j\\x16\\x01\\x00\\x00j\\x17\\x01\\x00\\x00j\\x18\\x01\\x00\\x00j\\x19\\x01\\x00\\x00j\\x1a\\x01\\x00\\x00j\\x1b\\x01\\x00\\x00j\\x1c\\x01\\x00\\x00j\\x1d\\x01\\x00\\x00j\\x1e\\x01\\x00\\x00j\\x1f\\x01\\x00\\x00j \\x01\\x00\\x00j!\\x01\\x00\\x00j\"\\x01\\x00\\x00j#\\x01\\x00\\x00j$\\x01\\x00\\x00j%\\x01\\x00\\x00j&\\x01\\x00\\x00j\\'\\x01\\x00\\x00j(\\x01\\x00\\x00j)\\x01\\x00\\x00j*\\x01\\x00\\x00j+\\x01\\x00\\x00j,\\x01\\x00\\x00j-\\x01\\x00\\x00j.\\x01\\x00\\x00j/\\x01\\x00\\x00j0\\x01\\x00\\x00j1\\x01\\x00\\x00j2\\x01\\x00\\x00j3\\x01\\x00\\x00j4\\x01\\x00\\x00j5\\x01\\x00\\x00j6\\x01\\x00\\x00j7\\x01\\x00\\x00j8\\x01\\x00\\x00j9\\x01\\x00\\x00j:\\x01\\x00\\x00j;\\x01\\x00\\x00j<\\x01\\x00\\x00j=\\x01\\x00\\x00j>\\x01\\x00\\x00j?\\x01\\x00\\x00j@\\x01\\x00\\x00jA\\x01\\x00\\x00jB\\x01\\x00\\x00jC\\x01\\x00\\x00jD\\x01\\x00\\x00jE\\x01\\x00\\x00jF\\x01\\x00\\x00jG\\x01\\x00\\x00jH\\x01\\x00\\x00jI\\x01\\x00\\x00jJ\\x01\\x00\\x00jK\\x01\\x00\\x00jL\\x01\\x00\\x00jM\\x01\\x00\\x00jN\\x01\\x00\\x00jO\\x01\\x00\\x00jP\\x01\\x00\\x00jQ\\x01\\x00\\x00jR\\x01\\x00\\x00jS\\x01\\x00\\x00jT\\x01\\x00\\x00jU\\x01\\x00\\x00jV\\x01\\x00\\x00jW\\x01\\x00\\x00jX\\x01\\x00\\x00jY\\x01\\x00\\x00jZ\\x01\\x00\\x00j[\\x01\\x00\\x00j\\\\\\x01\\x00\\x00j]\\x01\\x00\\x00j^\\x01\\x00\\x00j_\\x01\\x00\\x00j`\\x01\\x00\\x00ja\\x01\\x00\\x00jb\\x01\\x00\\x00jc\\x01\\x00\\x00jd\\x01\\x00\\x00je\\x01\\x00\\x00jf\\x01\\x00\\x00jg\\x01\\x00\\x00jh\\x01\\x00\\x00ji\\x01\\x00\\x00jj\\x01\\x00\\x00jk\\x01\\x00\\x00jl\\x01\\x00\\x00jm\\x01\\x00\\x00jn\\x01\\x00\\x00jo\\x01\\x00\\x00jp\\x01\\x00\\x00jq\\x01\\x00\\x00jr\\x01\\x00\\x00js\\x01\\x00\\x00jt\\x01\\x00\\x00ju\\x01\\x00\\x00jv\\x01\\x00\\x00jw\\x01\\x00\\x00jx\\x01\\x00\\x00jy\\x01\\x00\\x00jz\\x01\\x00\\x00j{\\x01\\x00\\x00j|\\x01\\x00\\x00j}\\x01\\x00\\x00j~\\x01\\x00\\x00j\\x7f\\x01\\x00\\x00j\\x80\\x01\\x00\\x00j\\x81\\x01\\x00\\x00j\\x82\\x01\\x00\\x00j\\x83\\x01\\x00\\x00j\\x84\\x01\\x00\\x00j\\x85\\x01\\x00\\x00j\\x86\\x01\\x00\\x00j\\x87\\x01\\x00\\x00j\\x88\\x01\\x00\\x00j\\x89\\x01\\x00\\x00j\\x8a\\x01\\x00\\x00j\\x8b\\x01\\x00\\x00j\\x8c\\x01\\x00\\x00j\\x8d\\x01\\x00\\x00j\\x8e\\x01\\x00\\x00j\\x8f\\x01\\x00\\x00j\\x90\\x01\\x00\\x00j\\x91\\x01\\x00\\x00j\\x92\\x01\\x00\\x00j\\x93\\x01\\x00\\x00e}\\x94(j\\x98\\x01\\x00\\x00}\\x94j\\xa2\\x01\\x00\\x00]\\x94(j\\xa4\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00ej\\xa4\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa4\\x01\\x00\\x00sNt\\x94bj\\xa5\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00sNt\\x94bubj\\x9c\\x01\\x00\\x00j\\x94\\x01\\x00\\x00j\\x9e\\x01\\x00\\x00j\\x95\\x01\\x00\\x00j\\xa0\\x01\\x00\\x00j\\x96\\x01\\x00\\x00ub\\x8c\\x06output\\x94h\\x06\\x8c\\x0bOutputFiles\\x94\\x93\\x94)\\x81\\x94(\\x8c2results/aggregated_counts/sars2_aligned_by_run.csv\\x94\\x8cAresults/aggregated_counts/sars2_aligned_by_metagenomic_sample.csv\\x94\\x8cAresults/aggregated_counts/mito_composition_by_metagenomic_run.csv\\x94\\x8cDresults/aggregated_counts/mito_composition_by_metagenomic_sample.csv\\x94e}\\x94(j\\x98\\x01\\x00\\x00}\\x94(\\x8c\\x14sars2_aligned_by_run\\x94K\\x00N\\x86\\x94\\x8c#sars2_aligned_by_metagenomic_sample\\x94K\\x01N\\x86\\x94\\x8c#mito_composition_by_metagenomic_run\\x94K\\x02N\\x86\\x94\\x8c&mito_composition_by_metagenomic_sample\\x94K\\x03N\\x86\\x94uj\\xa2\\x01\\x00\\x00]\\x94(j\\xa4\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00ej\\xa4\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa4\\x01\\x00\\x00sNt\\x94bj\\xa5\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00sNt\\x94bj\\xcc\\x01\\x00\\x00j\\xc6\\x01\\x00\\x00j\\xce\\x01\\x00\\x00j\\xc7\\x01\\x00\\x00j\\xd0\\x01\\x00\\x00j\\xc8\\x01\\x00\\x00j\\xd2\\x01\\x00\\x00j\\xc9\\x01\\x00\\x00ub\\x8c\\x06params\\x94h\\x06\\x8c\\x06Params\\x94\\x93\\x94)\\x81\\x94(\\x8c\\x0bNC_045512v2\\x94}\\x94(\\x8c\\rAmur hedgehog\\x94\\x8c\\x13Erinaceus amurensis\\x94\\x8c\\x03pig\\x94\\x8c\\x15Sus scrofa domesticus\\x94\\x8c\\tbrown rat\\x94\\x8c\\x1fRattus norvegicus strain Wistar\\x94\\x8c\\x06rabbit\\x94\\x8c\\x15Oryctolagus cuniculus\\x94\\x8c\\x03dog\\x94\\x8c\\x16Canis lupus familiaris\\x94\\x8c\\x11European hedgehog\\x94\\x8c\\x13Erinaceus europaeus\\x94\\x8c\\x13long-eared hedgehog\\x94\\x8c\\x13Hemiechinus auritus\\x94\\x8c\\x03cow\\x94\\x8c\\nBos taurus\\x94\\x8c\\x05human\\x94\\x8c\\x0cHomo sapiens\\x94\\x8c\\x0braccoon dog\\x94\\x8c\\x18Nyctereutes procyonoides\\x94\\x8c\\x10Himalayan marmot\\x94\\x8c\\x12Marmota himalayana\\x94\\x8c\\x0fSiberian weasel\\x94\\x8c\\x10Mustela sibirica\\x94\\x8c\\nhog badger\\x94\\x8c\\x11Arctonyx collaris\\x94\\x8c\\x10hoary bamboo rat\\x94\\x8c\\x12Rhizomys pruinosus\\x94\\x8c\\nfennec fox\\x94\\x8c\\x0cVulpes zerda\\x94\\x8c\\x05sheep\\x94\\x8c\\x0fOvis orientalis\\x94\\x8c\\x08gray fox\\x94\\x8c\\x18Urocyon cinereoargenteus\\x94\\x8c\\x04goat\\x94\\x8c\\x0eCapra aegagrus\\x94\\x8c\\x08sand cat\\x94\\x8c\\x0fFelis margarita\\x94\\x8c\\x11masked palm civet\\x94\\x8c\\x0ePaguma larvata\\x94\\x8c\\x11Mayalan field rat\\x94\\x8c\\x11Rattus tiomanicus\\x94\\x8c\\rbat-eared fox\\x94\\x8c\\x11Otocyon megalotis\\x94\\x8c\\tleaf deer\\x94\\x8c\\x14Muntiacus putaoensis\\x94\\x8c\\x11Malayan porcupine\\x94\\x8c\\x11Hystrix brachyura\\x94\\x8c\\x03cat\\x94\\x8c\\x0bFelis catus\\x94\\x8c\\x07chicken\\x94\\x8c\\x17Gallus gallus spadiceus\\x94\\x8c\\x04duck\\x94\\x8c\\x13Anas poecilorhyncha\\x94\\x8c\\tcrocodile\\x94\\x8c\\x15Osteolaemus tetraspis\\x94\\x8c\\x13Japanese salamander\\x94\\x8c\\x11Andrias japonicus\\x94\\x8c\\trat snake\\x94\\x8c\\x0cElaphe dione\\x94\\x8c\\x0esnakehead fish\\x94\\x8c\\x0fChanna maculata\\x94\\x8c\\x0ebrown bush hen\\x94\\x8c\\x10Amaurornis akool\\x94\\x8c\\rcollared dove\\x94\\x8c\\x15Streptopelia decaocto\\x94\\x8c\\x0fgolden pheasant\\x94\\x8c\\x13Chrysolophus pictus\\x94u}\\x94(\\x8c\\x19min_alignments_run_filter\\x94M\\xe8\\x03\\x8c!min_percent_aligned_genome_filter\\x94K\\nu]\\x94\\x8cORNA sequencing of total nucleic acids from environmental swabs for metagenomics\\x94ae}\\x94(j\\x98\\x01\\x00\\x00}\\x94(\\x8c\\x0csars2_ref_id\\x94K\\x00N\\x86\\x94\\x8c\\x14mito_genomes_to_keep\\x94K\\x01N\\x86\\x94\\x8c\\x18mito_composition_filters\\x94K\\x02N\\x86\\x94\\x8c\\x18metagenomic_descriptions\\x94K\\x03N\\x86\\x94uj\\xa2\\x01\\x00\\x00]\\x94(j\\xa4\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00ej\\xa4\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa4\\x01\\x00\\x00sNt\\x94bj\\xa5\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00sNt\\x94bj.\\x02\\x00\\x00j\\xe1\\x01\\x00\\x00j0\\x02\\x00\\x00j\\xe2\\x01\\x00\\x00j2\\x02\\x00\\x00j\\'\\x02\\x00\\x00j4\\x02\\x00\\x00j*\\x02\\x00\\x00ub\\x8c\\twildcards\\x94h\\x06\\x8c\\tWildcards\\x94\\x93\\x94)\\x81\\x94}\\x94(j\\x98\\x01\\x00\\x00}\\x94j\\xa2\\x01\\x00\\x00]\\x94(j\\xa4\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00ej\\xa4\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa4\\x01\\x00\\x00sNt\\x94bj\\xa5\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00sNt\\x94bub\\x8c\\x07threads\\x94K\\x01\\x8c\\tresources\\x94h\\x06\\x8c\\tResources\\x94\\x93\\x94)\\x81\\x94(K\\x01K\\x01\\x8c\\x04/tmp\\x94e}\\x94(j\\x98\\x01\\x00\\x00}\\x94(\\x8c\\x06_cores\\x94K\\x00N\\x86\\x94\\x8c\\x06_nodes\\x94K\\x01N\\x86\\x94\\x8c\\x06tmpdir\\x94K\\x02N\\x86\\x94uj\\xa2\\x01\\x00\\x00]\\x94(j\\xa4\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00ej\\xa4\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa4\\x01\\x00\\x00sNt\\x94bj\\xa5\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00sNt\\x94bjV\\x02\\x00\\x00K\\x01jX\\x02\\x00\\x00K\\x01jZ\\x02\\x00\\x00jS\\x02\\x00\\x00ub\\x8c\\x03log\\x94h\\x06\\x8c\\x03Log\\x94\\x93\\x94)\\x81\\x94\\x8c4results/aggregated_counts/aggregate_all_counts.ipynb\\x94a}\\x94(j\\x98\\x01\\x00\\x00}\\x94\\x8c\\x08notebook\\x94K\\x00N\\x86\\x94sj\\xa2\\x01\\x00\\x00]\\x94(j\\xa4\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00ej\\xa4\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa4\\x01\\x00\\x00sNt\\x94bj\\xa5\\x01\\x00\\x00j\\xa8\\x01\\x00\\x00j\\xaa\\x01\\x00\\x00\\x85\\x94R\\x94(j\\xaa\\x01\\x00\\x00)}\\x94j\\xae\\x01\\x00\\x00j\\xa5\\x01\\x00\\x00sNt\\x94bjl\\x02\\x00\\x00ji\\x02\\x00\\x00ub\\x8c\\x06config\\x94}\\x94(\\x8c\\x14crits_christoph_data\\x94}\\x94(\\x8c\\tfile_info\\x94\\x8c\\x95https://raw.githubusercontent.com/sars-cov-2-origins/huanan-environmental/bd2afdb19392a0efb872c09eb94045de9d265a6e/tables/TableS1_FileInformation.csv\\x94\\x8c\\x0csample_names\\x94\\x8c\\x91https://raw.githubusercontent.com/sars-cov-2-origins/huanan-environmental/bd2afdb19392a0efb872c09eb94045de9d265a6e/tables/TableS2_SampleNames.csv\\x94\\x8c\\x0bread_counts\\x94\\x8c\\x96https://raw.githubusercontent.com/sars-cov-2-origins/huanan-environmental/bd2afdb19392a0efb872c09eb94045de9d265a6e/tables/TableS3_MappedReadCounts.csv\\x94u\\x8c\\x15mitochondrial_genomes\\x94\\x8cYhttps://ftp.ncbi.nlm.nih.gov/refseq/release/mitochondrion/mitochondrion.1.genomic.gbff.gz\\x94\\x8c\\x1bextra_mitochondrial_genomes\\x94]\\x94\\x8c\\nKX964606.1\\x94a\\x8c\\x1amitochondrial_taxa_to_keep\\x94}\\x94(\\x8c\\x06phylum\\x94]\\x94\\x8c\\x08Chordata\\x94a\\x8c\\tsubphylum\\x94]\\x94u\\x8c\\x1dmitochondrial_genomes_to_keep\\x94j\\xe2\\x01\\x00\\x00\\x8c\"mitochondrial_genome_min_mash_dist\\x94G?\\xb1\\xeb\\x85\\x1e\\xb8Q\\xec\\x8c\\tsars2_ref\\x94\\x8cGhttp://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/bigZips/wuhCor1.fa.gz\\x94\\x8c\\x0csars2_ref_id\\x94j\\xe1\\x01\\x00\\x00\\x8c\\x16sars2_ref_trim_3_polyA\\x94\\x88\\x8c\\x18metagenomic_descriptions\\x94]\\x94j+\\x02\\x00\\x00a\\x8c\\x18mito_composition_filters\\x94j\\'\\x02\\x00\\x00u\\x8c\\x04rule\\x94\\x8c\\x14aggregate_all_counts\\x94\\x8c\\x0fbench_iteration\\x94N\\x8c\\tscriptdir\\x94\\x8cT/fh/fast/bloom_j/computational_notebooks/jbloom/2023/Huanan_market_samples/notebooks\\x94ub.'); from snakemake.logging import logger; logger.printshellcmds = False; import os; os.chdir(r'/fh/fast/bloom_j/computational_notebooks/jbloom/2023/Huanan_market_samples');\n",
    "######## snakemake preamble end #########\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "080d9351-837f-4e60-b199-de54a8e67b66",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:20.676179Z",
     "iopub.status.busy": "2023-04-07T04:03:20.675834Z",
     "iopub.status.idle": "2023-04-07T04:03:20.679915Z",
     "shell.execute_reply": "2023-04-07T04:03:20.679154Z",
     "shell.execute_reply.started": "2023-04-07T04:03:20.676151Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "442943df-4fd0-455e-ab33-1c0c16a61496",
   "metadata": {},
   "source": [
    "Get variables from `snakemake`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "301ac539-6b0b-4d37-b08b-43a059f521e8",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:20.680921Z",
     "iopub.status.busy": "2023-04-07T04:03:20.680715Z",
     "iopub.status.idle": "2023-04-07T04:03:20.685132Z",
     "shell.execute_reply": "2023-04-07T04:03:20.684394Z",
     "shell.execute_reply.started": "2023-04-07T04:03:20.680902Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "tallied_count_files = snakemake.input.tallied_counts\n",
    "mito_ref_info_csv = snakemake.input.mito_ref_info\n",
    "metadata_csv = snakemake.input.metadata\n",
    "read_counts_csv = snakemake.input.read_counts\n",
    "sars2_ref_id = snakemake.params.sars2_ref_id\n",
    "mito_genomes_to_keep = snakemake.params.mito_genomes_to_keep\n",
    "mito_composition_filters = snakemake.params.mito_composition_filters\n",
    "metagenomic_descriptions = snakemake.params.metagenomic_descriptions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bfc352d8-105f-4f1e-9412-b376809d0b25",
   "metadata": {},
   "source": [
    "Read information on mitochondrial reference genomes, and get species name as just first two words:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "9936a779-9b99-4b35-b2b1-cf41bdd23edc",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:04:24.034538Z",
     "iopub.status.busy": "2023-04-07T04:04:24.033811Z",
     "iopub.status.idle": "2023-04-07T04:04:24.055120Z",
     "shell.execute_reply": "2023-04-07T04:04:24.054374Z",
     "shell.execute_reply.started": "2023-04-07T04:04:24.034491Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>species</th>\n",
       "      <th>common_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KX964606.1</td>\n",
       "      <td>Erinaceus amurensis</td>\n",
       "      <td>Amur hedgehog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_012095.1</td>\n",
       "      <td>Sus scrofa domesticus</td>\n",
       "      <td>pig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>Rattus norvegicus strain Wistar</td>\n",
       "      <td>brown rat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_001913.1</td>\n",
       "      <td>Oryctolagus cuniculus</td>\n",
       "      <td>rabbit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_002008.4</td>\n",
       "      <td>Canis lupus familiaris</td>\n",
       "      <td>dog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4166</th>\n",
       "      <td>NC_029381.1</td>\n",
       "      <td>Tlacuatzin canescens</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4167</th>\n",
       "      <td>NC_026049.1</td>\n",
       "      <td>Pelusios castaneus voucher AHNU26080283</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4168</th>\n",
       "      <td>NC_044441.1</td>\n",
       "      <td>Aphyosemion cognatum voucher AMNH-I-251785</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4169</th>\n",
       "      <td>NC_051037.1</td>\n",
       "      <td>Cinclus mexicanus</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4170</th>\n",
       "      <td>NC_060309.1</td>\n",
       "      <td>Nyctalus aviator</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4171 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               id                                     species    common_name\n",
       "0      KX964606.1                         Erinaceus amurensis  Amur hedgehog\n",
       "1     NC_012095.1                       Sus scrofa domesticus            pig\n",
       "2     AC_000022.2             Rattus norvegicus strain Wistar      brown rat\n",
       "3     NC_001913.1                       Oryctolagus cuniculus         rabbit\n",
       "4     NC_002008.4                      Canis lupus familiaris            dog\n",
       "...           ...                                         ...            ...\n",
       "4166  NC_029381.1                        Tlacuatzin canescens            NaN\n",
       "4167  NC_026049.1     Pelusios castaneus voucher AHNU26080283            NaN\n",
       "4168  NC_044441.1  Aphyosemion cognatum voucher AMNH-I-251785            NaN\n",
       "4169  NC_051037.1                           Cinclus mexicanus            NaN\n",
       "4170  NC_060309.1                            Nyctalus aviator            NaN\n",
       "\n",
       "[4171 rows x 3 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mito_ref_info = pd.read_csv(mito_ref_info_csv)\n",
    "\n",
    "mito_ref_info"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28989fa2-3ee3-4f93-9c2d-1cfdd8b50765",
   "metadata": {},
   "source": [
    "Read metadata on samples.\n",
    "For sample titles that have multiple sample names, append the name to the title:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "66dd0295-14a9-4004-a670-a7c609ba2459",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:20.710119Z",
     "iopub.status.busy": "2023-04-07T04:03:20.709738Z",
     "iopub.status.idle": "2023-04-07T04:03:20.726322Z",
     "shell.execute_reply": "2023-04-07T04:03:20.725418Z",
     "shell.execute_reply.started": "2023-04-07T04:03:20.710096Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "metadata_all = pd.read_csv(metadata_csv)\n",
    "\n",
    "# downsize to most relevant metadata to keep\n",
    "metadata = (\n",
    "    metadata_all\n",
    "    # make sample titles unique if they correspond to multiple names\n",
    "    .assign(\n",
    "        n_sample_names=lambda x: (\n",
    "            x.groupby(\"Sample title\")[\"Sample name\"].transform(\"nunique\")\n",
    "        ),\n",
    "        sample=lambda x: x[\"Sample title\"].where(\n",
    "            x[\"n_sample_names\"] == 1,\n",
    "            x[\"Sample title\"] + \"_\" + x[\"Sample name\"],\n",
    "        )\n",
    "    )\n",
    "    # rename and organize\n",
    "    .rename(columns={\"Public description\": \"description\"})\n",
    "    [\n",
    "        [\n",
    "            \"Run accession\",\n",
    "            \"sample\",\n",
    "            \"Sample name\",\n",
    "            \"Collection date\",\n",
    "            \"description\",\n",
    "            \"Isolation source\",\n",
    "        ]\n",
    "    ]\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bf5e0344-11fa-406b-99b1-5d9d10e238f1",
   "metadata": {},
   "source": [
    "Tally all the alignment counts:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "be252664-0415-4f43-b4f1-25d6c642bf5f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:20.727576Z",
     "iopub.status.busy": "2023-04-07T04:03:20.727187Z",
     "iopub.status.idle": "2023-04-07T04:03:24.351823Z",
     "shell.execute_reply": "2023-04-07T04:03:24.350927Z",
     "shell.execute_reply.started": "2023-04-07T04:03:20.727556Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "tallied_counts = pd.concat(\n",
    "    [pd.read_csv(f) for f in tallied_count_files],\n",
    "    ignore_index=True,\n",
    ")\n",
    "\n",
    "# make sure all alignment references are SARS2 or a known mitochondrial genome\n",
    "assert set(tallied_counts[\"alignment_reference\"]).issubset(\n",
    "    set(mito_ref_info[\"id\"]).union([sars2_ref_id])\n",
    ")\n",
    "\n",
    "assert (\n",
    "    len(tallied_counts)\n",
    "    == len(tallied_counts.drop_duplicates())\n",
    "    == len(tallied_counts.groupby([\"alignment_reference\", \"Run accession\"]))\n",
    ")\n",
    "\n",
    "# only keep these types of counts\n",
    "count_cols = [\"n_primary\", \"n_primary_and_unique_best\"]\n",
    "\n",
    "tallied_counts_tidy = tallied_counts.melt(\n",
    "    id_vars=[\"Run accession\", \"alignment_reference\"],\n",
    "    value_vars=count_cols,\n",
    "    var_name=\"alignment_count_type\",\n",
    "    value_name=\"alignment_count\",\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6955a97c-9227-4993-a247-1ccf487fd55b",
   "metadata": {},
   "source": [
    "Get the read and alignment counts:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "b1e5d096-cfe9-4bb1-b255-c178daa5a605",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:24.353092Z",
     "iopub.status.busy": "2023-04-07T04:03:24.352770Z",
     "iopub.status.idle": "2023-04-07T04:03:24.359079Z",
     "shell.execute_reply": "2023-04-07T04:03:24.358204Z",
     "shell.execute_reply.started": "2023-04-07T04:03:24.353072Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "read_counts = pd.read_csv(read_counts_csv)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6748fae6-e8a7-4b99-85c2-980abbe1ac04",
   "metadata": {},
   "source": [
    "## SARS-CoV-2 read counts\n",
    "Now get the SARS-CoV-2 read counts as a percentage of all preprocessed reads.\n",
    "\n",
    "We first do this by run:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a8807c49-2033-43ec-a14a-d68ce74ab27a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:24.360386Z",
     "iopub.status.busy": "2023-04-07T04:03:24.360055Z",
     "iopub.status.idle": "2023-04-07T04:03:24.432682Z",
     "shell.execute_reply": "2023-04-07T04:03:24.431898Z",
     "shell.execute_reply.started": "2023-04-07T04:03:24.360365Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run accession</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>preprocessed_reads</th>\n",
       "      <th>alignment_count_type</th>\n",
       "      <th>SARS2_aligned_reads</th>\n",
       "      <th>percent_preprocessed_reads_aligning_to_SARS2</th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>description</th>\n",
       "      <th>Isolation source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>106545428</td>\n",
       "      <td>90854684</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>5</td>\n",
       "      <td>5.503294e-06</td>\n",
       "      <td>Q37</td>\n",
       "      <td>Env_0552</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRR710761</td>\n",
       "      <td>106545428</td>\n",
       "      <td>90854684</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>5</td>\n",
       "      <td>5.503294e-06</td>\n",
       "      <td>Q37</td>\n",
       "      <td>Env_0552</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRR710762</td>\n",
       "      <td>213214887</td>\n",
       "      <td>206952411</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>1</td>\n",
       "      <td>4.832029e-07</td>\n",
       "      <td>Q61</td>\n",
       "      <td>Env_0576</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRR710762</td>\n",
       "      <td>213214887</td>\n",
       "      <td>206952411</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>1</td>\n",
       "      <td>4.832029e-07</td>\n",
       "      <td>Q61</td>\n",
       "      <td>Env_0576</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRR710763</td>\n",
       "      <td>120407578</td>\n",
       "      <td>120404558</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>5</td>\n",
       "      <td>4.152667e-06</td>\n",
       "      <td>Q64</td>\n",
       "      <td>Env_0579</td>\n",
       "      <td>2020-01-12</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>783</th>\n",
       "      <td>CRR711140</td>\n",
       "      <td>60481208</td>\n",
       "      <td>60460916</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>CCDC-758</td>\n",
       "      <td>Env_0907</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>784</th>\n",
       "      <td>CRR711141</td>\n",
       "      <td>124565478</td>\n",
       "      <td>124415388</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>CCDC-760</td>\n",
       "      <td>Env_0909</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>785</th>\n",
       "      <td>CRR711141</td>\n",
       "      <td>124565478</td>\n",
       "      <td>124415388</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>CCDC-760</td>\n",
       "      <td>Env_0909</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>786</th>\n",
       "      <td>CRR711142</td>\n",
       "      <td>123643938</td>\n",
       "      <td>123541212</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>CCDC-760</td>\n",
       "      <td>Env_0909</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>787</th>\n",
       "      <td>CRR711142</td>\n",
       "      <td>123643938</td>\n",
       "      <td>123541212</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>CCDC-760</td>\n",
       "      <td>Env_0909</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>788 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Run accession  total_reads  preprocessed_reads       alignment_count_type  \\\n",
       "0       CRR710761    106545428            90854684                  n_primary   \n",
       "1       CRR710761    106545428            90854684  n_primary_and_unique_best   \n",
       "2       CRR710762    213214887           206952411                  n_primary   \n",
       "3       CRR710762    213214887           206952411  n_primary_and_unique_best   \n",
       "4       CRR710763    120407578           120404558                  n_primary   \n",
       "..            ...          ...                 ...                        ...   \n",
       "783     CRR711140     60481208            60460916  n_primary_and_unique_best   \n",
       "784     CRR711141    124565478           124415388                  n_primary   \n",
       "785     CRR711141    124565478           124415388  n_primary_and_unique_best   \n",
       "786     CRR711142    123643938           123541212                  n_primary   \n",
       "787     CRR711142    123643938           123541212  n_primary_and_unique_best   \n",
       "\n",
       "     SARS2_aligned_reads  percent_preprocessed_reads_aligning_to_SARS2  \\\n",
       "0                      5                                  5.503294e-06   \n",
       "1                      5                                  5.503294e-06   \n",
       "2                      1                                  4.832029e-07   \n",
       "3                      1                                  4.832029e-07   \n",
       "4                      5                                  4.152667e-06   \n",
       "..                   ...                                           ...   \n",
       "783                    0                                  0.000000e+00   \n",
       "784                    0                                  0.000000e+00   \n",
       "785                    0                                  0.000000e+00   \n",
       "786                    0                                  0.000000e+00   \n",
       "787                    0                                  0.000000e+00   \n",
       "\n",
       "       sample Sample name Collection date  \\\n",
       "0         Q37    Env_0552      2020-01-12   \n",
       "1         Q37    Env_0552      2020-01-12   \n",
       "2         Q61    Env_0576      2020-01-12   \n",
       "3         Q61    Env_0576      2020-01-12   \n",
       "4         Q64    Env_0579      2020-01-12   \n",
       "..        ...         ...             ...   \n",
       "783  CCDC-758    Env_0907      2020-03-02   \n",
       "784  CCDC-760    Env_0909      2020-03-02   \n",
       "785  CCDC-760    Env_0909      2020-03-02   \n",
       "786  CCDC-760    Env_0909      2020-03-02   \n",
       "787  CCDC-760    Env_0909      2020-03-02   \n",
       "\n",
       "                                           description  Isolation source  \n",
       "0    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "1    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "2    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "3    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "4    RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "..                                                 ...               ...  \n",
       "783  RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "784  RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "785  RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "786  RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "787  RNA sequencing of total nucleic acids from env...  West Wine of HSM  \n",
       "\n",
       "[788 rows x 11 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sars2_aligned_by_run = (\n",
    "    tallied_counts_tidy\n",
    "    .query(\"alignment_reference == @sars2_ref_id\")\n",
    "    .drop(columns=\"alignment_reference\")\n",
    "    .groupby(\n",
    "        [\"Run accession\", \"alignment_count_type\"],\n",
    "        as_index=False,\n",
    "    )\n",
    "    .aggregate(SARS2_aligned_reads=pd.NamedAgg(\"alignment_count\", \"sum\"))\n",
    "    # pivot and melt to fill zero values\n",
    "    .pivot_table(\n",
    "        index=\"Run accession\",\n",
    "        columns=\"alignment_count_type\",\n",
    "        values=\"SARS2_aligned_reads\",\n",
    "        fill_value=0,\n",
    "    )\n",
    "    .reset_index()\n",
    "    .merge(read_counts, validate=\"one_to_one\", how=\"outer\")\n",
    "    .assign(**{c: lambda x: x[c].fillna(0) for c in count_cols})\n",
    "    .melt(\n",
    "        id_vars=[\"Run accession\", \"total_reads\", \"preprocessed_reads\"],\n",
    "        value_vars=count_cols,\n",
    "        var_name=\"alignment_count_type\",\n",
    "        value_name=\"SARS2_aligned_reads\",\n",
    "    )\n",
    "    .assign(\n",
    "        percent_preprocessed_reads_aligning_to_SARS2=(\n",
    "            lambda x: x[\"SARS2_aligned_reads\"] / x[\"preprocessed_reads\"] * 100\n",
    "        ),\n",
    "    )\n",
    "    .merge(\n",
    "        metadata,\n",
    "        on=\"Run accession\",\n",
    "        validate=\"many_to_one\",\n",
    "        how=\"outer\",\n",
    "    )\n",
    "    .assign(SARS2_aligned_reads=lambda x: x[\"SARS2_aligned_reads\"].astype(\"Int64\"))\n",
    ")\n",
    "\n",
    "sars2_aligned_by_run.to_csv(\n",
    "    snakemake.output[\"sars2_aligned_by_run\"], index=False, float_format=\"%.5g\",\n",
    ")\n",
    "\n",
    "sars2_aligned_by_run"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d6385ce-0467-40cc-a449-aad91e9aa762",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-04T12:31:56.085557Z",
     "iopub.status.busy": "2023-04-04T12:31:56.085007Z",
     "iopub.status.idle": "2023-04-04T12:31:56.093519Z",
     "shell.execute_reply": "2023-04-04T12:31:56.092581Z",
     "shell.execute_reply.started": "2023-04-04T12:31:56.085514Z"
    },
    "tags": []
   },
   "source": [
    "Now we do this by sample (aggregating runs for each sample) for **just the metagenomic samples**."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "f9d7b1f6-50a1-4151-a843-c89e86cc2a84",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:24.434068Z",
     "iopub.status.busy": "2023-04-07T04:03:24.433687Z",
     "iopub.status.idle": "2023-04-07T04:03:24.492597Z",
     "shell.execute_reply": "2023-04-07T04:03:24.491714Z",
     "shell.execute_reply.started": "2023-04-07T04:03:24.434045Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alignment_count_type</th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>description</th>\n",
       "      <th>Isolation source</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>preprocessed_reads</th>\n",
       "      <th>SARS2_aligned_reads</th>\n",
       "      <th>percent_preprocessed_reads_aligning_to_SARS2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>n_primary</td>\n",
       "      <td>06-29-abv1</td>\n",
       "      <td>Env_0620</td>\n",
       "      <td>2020-01-23</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>1079014964</td>\n",
       "      <td>851849944</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>CCDC-758</td>\n",
       "      <td>Env_0907</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>118189266</td>\n",
       "      <td>118129164</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>CCDC-746</td>\n",
       "      <td>Env_0895</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>189764532</td>\n",
       "      <td>189578868</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>CCDC-744</td>\n",
       "      <td>Env_0893</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>224276396</td>\n",
       "      <td>224133266</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>CCDC-454</td>\n",
       "      <td>Env_0873</td>\n",
       "      <td>2020-02-25</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>240593678</td>\n",
       "      <td>240395146</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>D32</td>\n",
       "      <td>Env_0213</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>197394836</td>\n",
       "      <td>192577410</td>\n",
       "      <td>284</td>\n",
       "      <td>0.000147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>B5</td>\n",
       "      <td>Env_0126</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>229464470</td>\n",
       "      <td>228684372</td>\n",
       "      <td>433</td>\n",
       "      <td>0.000189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>n_primary</td>\n",
       "      <td>B5</td>\n",
       "      <td>Env_0126</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>229464470</td>\n",
       "      <td>228684372</td>\n",
       "      <td>433</td>\n",
       "      <td>0.000189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>n_primary</td>\n",
       "      <td>F100</td>\n",
       "      <td>Env_0400</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>260045208</td>\n",
       "      <td>258891974</td>\n",
       "      <td>5246</td>\n",
       "      <td>0.002026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>F100</td>\n",
       "      <td>Env_0400</td>\n",
       "      <td>2020-01-01</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "      <td>260045208</td>\n",
       "      <td>258891974</td>\n",
       "      <td>5246</td>\n",
       "      <td>0.002026</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>338 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          alignment_count_type      sample Sample name Collection date  \\\n",
       "0                    n_primary  06-29-abv1    Env_0620      2020-01-23   \n",
       "223  n_primary_and_unique_best    CCDC-758    Env_0907      2020-03-02   \n",
       "222  n_primary_and_unique_best    CCDC-746    Env_0895      2020-03-02   \n",
       "221  n_primary_and_unique_best    CCDC-744    Env_0893      2020-03-02   \n",
       "220  n_primary_and_unique_best    CCDC-454    Env_0873      2020-02-25   \n",
       "..                         ...         ...         ...             ...   \n",
       "228  n_primary_and_unique_best         D32    Env_0213      2020-01-01   \n",
       "216  n_primary_and_unique_best          B5    Env_0126      2020-01-01   \n",
       "47                   n_primary          B5    Env_0126      2020-01-01   \n",
       "67                   n_primary        F100    Env_0400      2020-01-01   \n",
       "236  n_primary_and_unique_best        F100    Env_0400      2020-01-01   \n",
       "\n",
       "                                           description  Isolation source  \\\n",
       "0    RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "223  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "222  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "221  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "220  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "..                                                 ...               ...   \n",
       "228  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "216  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "47   RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "67   RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "236  RNA sequencing of total nucleic acids from env...  West Wine of HSM   \n",
       "\n",
       "     total_reads  preprocessed_reads  SARS2_aligned_reads  \\\n",
       "0     1079014964           851849944                    0   \n",
       "223    118189266           118129164                    0   \n",
       "222    189764532           189578868                    0   \n",
       "221    224276396           224133266                    0   \n",
       "220    240593678           240395146                    0   \n",
       "..           ...                 ...                  ...   \n",
       "228    197394836           192577410                  284   \n",
       "216    229464470           228684372                  433   \n",
       "47     229464470           228684372                  433   \n",
       "67     260045208           258891974                 5246   \n",
       "236    260045208           258891974                 5246   \n",
       "\n",
       "     percent_preprocessed_reads_aligning_to_SARS2  \n",
       "0                                             0.0  \n",
       "223                                           0.0  \n",
       "222                                           0.0  \n",
       "221                                           0.0  \n",
       "220                                           0.0  \n",
       "..                                            ...  \n",
       "228                                      0.000147  \n",
       "216                                      0.000189  \n",
       "47                                       0.000189  \n",
       "67                                       0.002026  \n",
       "236                                      0.002026  \n",
       "\n",
       "[338 rows x 10 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# columns that we sum across runs for each sample\n",
    "sum_cols = [\n",
    "    \"total_reads\",\n",
    "    \"preprocessed_reads\",\n",
    "    \"SARS2_aligned_reads\",\n",
    "]\n",
    "\n",
    "sars2_aligned_by_metagenomic_sample = (\n",
    "    sars2_aligned_by_run\n",
    "    .query(\"description in @metagenomic_descriptions\")\n",
    "    .groupby(\n",
    "        [\n",
    "            c for c in sars2_aligned_by_run\n",
    "            if c not in sum_cols\n",
    "            and not c.startswith(\"percent_\")\n",
    "            and c != \"Run accession\"\n",
    "        ],\n",
    "        as_index=False,\n",
    "        dropna=False,\n",
    "    )\n",
    "    .aggregate({c: \"sum\" for c in sum_cols})\n",
    "    .assign(\n",
    "        percent_preprocessed_reads_aligning_to_SARS2=(\n",
    "            lambda x: x[\"SARS2_aligned_reads\"] / x[\"preprocessed_reads\"] * 100\n",
    "        ),\n",
    "    )\n",
    ")\n",
    "\n",
    "dup_samples = (\n",
    "    sars2_aligned_by_metagenomic_sample\n",
    "    .groupby([\"alignment_count_type\", \"sample\"])\n",
    "    .aggregate(\n",
    "        n=pd.NamedAgg(\"total_reads\", \"count\"),\n",
    "        sample_names=pd.NamedAgg(\"Sample name\", \"unique\"),\n",
    "    )\n",
    "    .query(\"n > 1\")\n",
    ")\n",
    "\n",
    "assert not len(dup_samples), f\"Some duplicated sample names\\n{dup_samples}\"\n",
    "\n",
    "sars2_aligned_by_metagenomic_sample.to_csv(\n",
    "    snakemake.output[\"sars2_aligned_by_metagenomic_sample\"], index=False, float_format=\"%.5g\",\n",
    ")\n",
    "\n",
    "sars2_aligned_by_metagenomic_sample.sort_values(\"SARS2_aligned_reads\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "024f4dc9-93f2-4f56-9380-716b59ca2dd7",
   "metadata": {},
   "source": [
    "## Composition of mitochondrial reads\n",
    "\n",
    "Do some filtering on the runs:\n",
    " - Get just the mitochondrial counts\n",
    " - Only keep runs with the metagenomic description\n",
    " - Exclude runs with insufficient primary read alignment counts to mitochondria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "b258ec79-2e5d-4f6e-9c31-ac4fa339f509",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:03:24.493681Z",
     "iopub.status.busy": "2023-04-07T04:03:24.493454Z",
     "iopub.status.idle": "2023-04-07T04:03:24.976786Z",
     "shell.execute_reply": "2023-04-07T04:03:24.975710Z",
     "shell.execute_reply.started": "2023-04-07T04:03:24.493662Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding the following metagenomic runs with insufficient aligned mitochondrial reads:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run accession</th>\n",
       "      <th>total_aligned_reads</th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample name</th>\n",
       "      <th>Collection date</th>\n",
       "      <th>description</th>\n",
       "      <th>Isolation source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRR711139</td>\n",
       "      <td>155</td>\n",
       "      <td>CCDC-758</td>\n",
       "      <td>Env_0907</td>\n",
       "      <td>2020-03-02</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRR710932</td>\n",
       "      <td>174</td>\n",
       "      <td>SJ-L3</td>\n",
       "      <td>Env_0865</td>\n",
       "      <td>2020-02-22</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>Warehouses related to west wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRR710919</td>\n",
       "      <td>195</td>\n",
       "      <td>8-25-D_Env_0717</td>\n",
       "      <td>Env_0717</td>\n",
       "      <td>2020-02-03</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRR711090</td>\n",
       "      <td>223</td>\n",
       "      <td>8-25-Long</td>\n",
       "      <td>Env_0756</td>\n",
       "      <td>2020-02-09</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRR711068</td>\n",
       "      <td>229</td>\n",
       "      <td>8-25-BX</td>\n",
       "      <td>Env_0714</td>\n",
       "      <td>2020-02-03</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>CRR711073</td>\n",
       "      <td>947</td>\n",
       "      <td>8-25-CK</td>\n",
       "      <td>Env_0716</td>\n",
       "      <td>2020-02-03</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>CRR710801</td>\n",
       "      <td>951</td>\n",
       "      <td>WWS-1#-2</td>\n",
       "      <td>Env_0817</td>\n",
       "      <td>2020-02-15</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>CRR711045</td>\n",
       "      <td>956</td>\n",
       "      <td>1-29-23</td>\n",
       "      <td>Env_0701</td>\n",
       "      <td>2020-01-29</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>East Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>CRR711075</td>\n",
       "      <td>962</td>\n",
       "      <td>8-25-CK</td>\n",
       "      <td>Env_0716</td>\n",
       "      <td>2020-02-03</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>CRR710913</td>\n",
       "      <td>996</td>\n",
       "      <td>1-27-33</td>\n",
       "      <td>Env_0660</td>\n",
       "      <td>2020-01-27</td>\n",
       "      <td>RNA sequencing of total nucleic acids from env...</td>\n",
       "      <td>West Wine of HSM (Sewers or sewerage wells)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Run accession  total_aligned_reads           sample Sample name  \\\n",
       "0      CRR711139                  155         CCDC-758    Env_0907   \n",
       "1      CRR710932                  174            SJ-L3    Env_0865   \n",
       "2      CRR710919                  195  8-25-D_Env_0717    Env_0717   \n",
       "3      CRR711090                  223        8-25-Long    Env_0756   \n",
       "4      CRR711068                  229          8-25-BX    Env_0714   \n",
       "..           ...                  ...              ...         ...   \n",
       "60     CRR711073                  947          8-25-CK    Env_0716   \n",
       "61     CRR710801                  951         WWS-1#-2    Env_0817   \n",
       "62     CRR711045                  956          1-29-23    Env_0701   \n",
       "63     CRR711075                  962          8-25-CK    Env_0716   \n",
       "64     CRR710913                  996          1-27-33    Env_0660   \n",
       "\n",
       "   Collection date                                        description  \\\n",
       "0       2020-03-02  RNA sequencing of total nucleic acids from env...   \n",
       "1       2020-02-22  RNA sequencing of total nucleic acids from env...   \n",
       "2       2020-02-03  RNA sequencing of total nucleic acids from env...   \n",
       "3       2020-02-09  RNA sequencing of total nucleic acids from env...   \n",
       "4       2020-02-03  RNA sequencing of total nucleic acids from env...   \n",
       "..             ...                                                ...   \n",
       "60      2020-02-03  RNA sequencing of total nucleic acids from env...   \n",
       "61      2020-02-15  RNA sequencing of total nucleic acids from env...   \n",
       "62      2020-01-29  RNA sequencing of total nucleic acids from env...   \n",
       "63      2020-02-03  RNA sequencing of total nucleic acids from env...   \n",
       "64      2020-01-27  RNA sequencing of total nucleic acids from env...   \n",
       "\n",
       "                               Isolation source  \n",
       "0                              West Wine of HSM  \n",
       "1        Warehouses related to west wine of HSM  \n",
       "2                              West Wine of HSM  \n",
       "3                              West Wine of HSM  \n",
       "4                              West Wine of HSM  \n",
       "..                                          ...  \n",
       "60                             West Wine of HSM  \n",
       "61  West Wine of HSM (Sewers or sewerage wells)  \n",
       "62                             East Wine of HSM  \n",
       "63                             West Wine of HSM  \n",
       "64  West Wine of HSM (Sewers or sewerage wells)  \n",
       "\n",
       "[65 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mito_counts = (\n",
    "    tallied_counts_tidy\n",
    "    .merge(metadata, validate=\"many_to_one\")\n",
    "    .query(\"alignment_reference != @sars2_ref_id\")\n",
    "    .query(\"description in @metagenomic_descriptions\")\n",
    "    .assign(alignment_count=lambda x: x[\"alignment_count\"].astype(int))\n",
    ")\n",
    "\n",
    "insufficient_mito_reads = (\n",
    "    mito_counts\n",
    "    .query(\"alignment_count_type == 'n_primary'\")\n",
    "    .groupby(\"Run accession\", as_index=False)\n",
    "    .aggregate(total_aligned_reads=pd.NamedAgg(\"alignment_count\", \"sum\"))\n",
    "    .sort_values(\"total_aligned_reads\")\n",
    "    .query(\"total_aligned_reads < @mito_composition_filters['min_alignments_run_filter']\")\n",
    "    .reset_index(drop=True)\n",
    "    .merge(metadata, how=\"left\")\n",
    ")\n",
    "\n",
    "print(\"Excluding the following metagenomic runs with insufficient aligned mitochondrial reads:\")\n",
    "display(insufficient_mito_reads)\n",
    "\n",
    "mito_counts = mito_counts[\n",
    "    ~mito_counts[\"Run accession\"].isin(insufficient_mito_reads[\"Run accession\"])\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16e75ff2-dc43-4557-a64e-959ee29c7e41",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-04T13:43:25.177040Z",
     "iopub.status.busy": "2023-04-04T13:43:25.176357Z",
     "iopub.status.idle": "2023-04-04T13:43:25.184777Z",
     "shell.execute_reply": "2023-04-04T13:43:25.183760Z",
     "shell.execute_reply.started": "2023-04-04T13:43:25.176998Z"
    },
    "tags": []
   },
   "source": [
    "Now exclude mitochondrial reference genomes that don't have a high enough percentage of primary read alignments or are not delineated in the list to keep:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "89a56047-8ab0-43ad-bac8-79a6176b6848",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-07T04:42:09.278986Z",
     "iopub.status.busy": "2023-04-07T04:42:09.278454Z",
     "iopub.status.idle": "2023-04-07T04:42:09.782614Z",
     "shell.execute_reply": "2023-04-07T04:42:09.781567Z",
     "shell.execute_reply.started": "2023-04-07T04:42:09.278947Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Keeping the following mitochondrial genomes:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alignment_reference</th>\n",
       "      <th>max_percent</th>\n",
       "      <th>avg_percent</th>\n",
       "      <th>species</th>\n",
       "      <th>common_name</th>\n",
       "      <th>reason_kept</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_009728.1</td>\n",
       "      <td>89.93</td>\n",
       "      <td>18.13</td>\n",
       "      <td>Osteolaemus tetraspis</td>\n",
       "      <td>crocodile</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_012920.1</td>\n",
       "      <td>89.10</td>\n",
       "      <td>6.21</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>human</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NC_008143.1</td>\n",
       "      <td>48.41</td>\n",
       "      <td>5.17</td>\n",
       "      <td>Crocodylus porosus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NC_035814.1</td>\n",
       "      <td>30.46</td>\n",
       "      <td>4.93</td>\n",
       "      <td>Aonyx cinerea isolate 16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NC_050988.1</td>\n",
       "      <td>20.26</td>\n",
       "      <td>4.89</td>\n",
       "      <td>Antechinus arktos voucher AA100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_040902.1</td>\n",
       "      <td>76.62</td>\n",
       "      <td>4.43</td>\n",
       "      <td>Gallus gallus spadiceus</td>\n",
       "      <td>chicken</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NC_042233.1</td>\n",
       "      <td>28.10</td>\n",
       "      <td>4.09</td>\n",
       "      <td>Numenius tenuirostris isolate lib4-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NC_020011.1</td>\n",
       "      <td>42.29</td>\n",
       "      <td>2.62</td>\n",
       "      <td>Channa maculata</td>\n",
       "      <td>snakehead fish</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NC_022418.1</td>\n",
       "      <td>36.10</td>\n",
       "      <td>2.58</td>\n",
       "      <td>Anas poecilorhyncha</td>\n",
       "      <td>duck</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_007446.1</td>\n",
       "      <td>65.53</td>\n",
       "      <td>2.17</td>\n",
       "      <td>Andrias japonicus</td>\n",
       "      <td>Japanese salamander</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_030041.1</td>\n",
       "      <td>66.64</td>\n",
       "      <td>2.15</td>\n",
       "      <td>Ptyas mucosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NC_010288.1</td>\n",
       "      <td>54.22</td>\n",
       "      <td>2.12</td>\n",
       "      <td>Ctenopharyngodon idella</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NC_037513.1</td>\n",
       "      <td>42.87</td>\n",
       "      <td>2.06</td>\n",
       "      <td>Streptopelia decaocto</td>\n",
       "      <td>collared dove</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NC_057520.1</td>\n",
       "      <td>18.68</td>\n",
       "      <td>1.80</td>\n",
       "      <td>Lutreolina crassicaudata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NC_057519.1</td>\n",
       "      <td>19.85</td>\n",
       "      <td>1.77</td>\n",
       "      <td>Didelphis pernigra</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NC_013700.1</td>\n",
       "      <td>21.21</td>\n",
       "      <td>1.51</td>\n",
       "      <td>Nyctereutes procyonoides</td>\n",
       "      <td>raccoon dog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NC_026794.1</td>\n",
       "      <td>10.04</td>\n",
       "      <td>1.48</td>\n",
       "      <td>Colisa fasciata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NC_006853.1</td>\n",
       "      <td>50.82</td>\n",
       "      <td>1.35</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>cow</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NC_008680.1</td>\n",
       "      <td>11.52</td>\n",
       "      <td>1.27</td>\n",
       "      <td>Vaillantella maassi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NC_027671.1</td>\n",
       "      <td>20.46</td>\n",
       "      <td>1.21</td>\n",
       "      <td>Platymantis vitianus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NC_029159.1</td>\n",
       "      <td>21.20</td>\n",
       "      <td>1.12</td>\n",
       "      <td>Ictalurus cf. pricei USON-01115-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NC_021478.1</td>\n",
       "      <td>17.39</td>\n",
       "      <td>1.09</td>\n",
       "      <td>Rhizomys pruinosus</td>\n",
       "      <td>hoary bamboo rat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NC_031868.1</td>\n",
       "      <td>20.15</td>\n",
       "      <td>1.09</td>\n",
       "      <td>Columba jouyi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NC_023528.1</td>\n",
       "      <td>58.30</td>\n",
       "      <td>1.08</td>\n",
       "      <td>Rana dybowskii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NC_026307.1</td>\n",
       "      <td>10.59</td>\n",
       "      <td>0.89</td>\n",
       "      <td>Anoxypristis cuspidata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NC_020001.1</td>\n",
       "      <td>20.50</td>\n",
       "      <td>0.86</td>\n",
       "      <td>Sooglossus thomasseti</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NC_026063.1</td>\n",
       "      <td>12.81</td>\n",
       "      <td>0.84</td>\n",
       "      <td>Ovis orientalis</td>\n",
       "      <td>sheep</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NC_066473.1</td>\n",
       "      <td>10.36</td>\n",
       "      <td>0.82</td>\n",
       "      <td>Tropidophorus hainanus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NC_002008.4</td>\n",
       "      <td>19.82</td>\n",
       "      <td>0.78</td>\n",
       "      <td>Canis lupus familiaris</td>\n",
       "      <td>dog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NC_003192.1</td>\n",
       "      <td>14.05</td>\n",
       "      <td>0.72</td>\n",
       "      <td>Monopterus albus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NC_026723.1</td>\n",
       "      <td>4.73</td>\n",
       "      <td>0.67</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NC_024645.1</td>\n",
       "      <td>10.96</td>\n",
       "      <td>0.63</td>\n",
       "      <td>Culter alburnus x Ancherythroculter nigrocauda</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NC_001913.1</td>\n",
       "      <td>10.99</td>\n",
       "      <td>0.63</td>\n",
       "      <td>Oryctolagus cuniculus</td>\n",
       "      <td>rabbit</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NC_023982.1</td>\n",
       "      <td>26.13</td>\n",
       "      <td>0.61</td>\n",
       "      <td>Amaurornis akool</td>\n",
       "      <td>brown bush hen</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NC_012095.1</td>\n",
       "      <td>14.61</td>\n",
       "      <td>0.61</td>\n",
       "      <td>Sus scrofa domesticus</td>\n",
       "      <td>pig</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NC_008776.1</td>\n",
       "      <td>13.85</td>\n",
       "      <td>0.53</td>\n",
       "      <td>Heloderma suspectum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NC_014576.1</td>\n",
       "      <td>16.64</td>\n",
       "      <td>0.50</td>\n",
       "      <td>Chrysolophus pictus</td>\n",
       "      <td>golden pheasant</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NC_010781.1</td>\n",
       "      <td>12.22</td>\n",
       "      <td>0.44</td>\n",
       "      <td>Lophura ignita</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>KX964606.1</td>\n",
       "      <td>16.44</td>\n",
       "      <td>0.43</td>\n",
       "      <td>Erinaceus amurensis</td>\n",
       "      <td>Amur hedgehog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NC_018367.1</td>\n",
       "      <td>15.83</td>\n",
       "      <td>0.34</td>\n",
       "      <td>Marmota himalayana</td>\n",
       "      <td>Himalayan marmot</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NC_003408.1</td>\n",
       "      <td>16.91</td>\n",
       "      <td>0.33</td>\n",
       "      <td>Coturnix japonica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NC_069631.1</td>\n",
       "      <td>20.67</td>\n",
       "      <td>0.32</td>\n",
       "      <td>Terrapene carolina</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NC_029382.1</td>\n",
       "      <td>12.73</td>\n",
       "      <td>0.30</td>\n",
       "      <td>Zebrias crossolepis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NC_050263.1</td>\n",
       "      <td>12.04</td>\n",
       "      <td>0.26</td>\n",
       "      <td>Hystrix brachyura</td>\n",
       "      <td>Malayan porcupine</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>NC_001700.1</td>\n",
       "      <td>4.24</td>\n",
       "      <td>0.25</td>\n",
       "      <td>Felis catus</td>\n",
       "      <td>cat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NC_065160.1</td>\n",
       "      <td>14.93</td>\n",
       "      <td>0.23</td>\n",
       "      <td>Tupaia nicobarica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>AC_000022.2</td>\n",
       "      <td>4.23</td>\n",
       "      <td>0.21</td>\n",
       "      <td>Rattus norvegicus strain Wistar</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>NC_029888.1</td>\n",
       "      <td>3.87</td>\n",
       "      <td>0.18</td>\n",
       "      <td>Rattus tiomanicus</td>\n",
       "      <td>Mayalan field rat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>NC_007692.1</td>\n",
       "      <td>12.43</td>\n",
       "      <td>0.14</td>\n",
       "      <td>Testudo graeca</td>\n",
       "      <td>NaN</td>\n",
       "      <td>at least 10% in one run</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>NC_028308.1</td>\n",
       "      <td>2.43</td>\n",
       "      <td>0.12</td>\n",
       "      <td>Felis margarita</td>\n",
       "      <td>sand cat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>NC_020645.1</td>\n",
       "      <td>2.20</td>\n",
       "      <td>0.11</td>\n",
       "      <td>Arctonyx collaris</td>\n",
       "      <td>hog badger</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>NC_029403.1</td>\n",
       "      <td>1.09</td>\n",
       "      <td>0.10</td>\n",
       "      <td>Paguma larvata</td>\n",
       "      <td>masked palm civet</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>NC_002080.2</td>\n",
       "      <td>3.28</td>\n",
       "      <td>0.10</td>\n",
       "      <td>Erinaceus europaeus</td>\n",
       "      <td>European hedgehog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>NC_028161.1</td>\n",
       "      <td>1.52</td>\n",
       "      <td>0.09</td>\n",
       "      <td>Capra aegagrus</td>\n",
       "      <td>goat</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>NC_041068.1</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.08</td>\n",
       "      <td>Elaphe dione</td>\n",
       "      <td>rat snake</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>NC_023459.1</td>\n",
       "      <td>1.02</td>\n",
       "      <td>0.07</td>\n",
       "      <td>Vulpes zerda</td>\n",
       "      <td>fennec fox</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>NC_005033.1</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Hemiechinus auritus</td>\n",
       "      <td>long-eared hedgehog</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>NC_036430.1</td>\n",
       "      <td>1.77</td>\n",
       "      <td>0.04</td>\n",
       "      <td>Muntiacus putaoensis</td>\n",
       "      <td>leaf deer</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>NC_036369.1</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.04</td>\n",
       "      <td>Otocyon megalotis</td>\n",
       "      <td>bat-eared fox</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>NC_020637.1</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.03</td>\n",
       "      <td>Mustela sibirica</td>\n",
       "      <td>Siberian weasel</td>\n",
       "      <td>specified to keep in config</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   alignment_reference  max_percent  avg_percent  \\\n",
       "0          NC_009728.1        89.93        18.13   \n",
       "1          NC_012920.1        89.10         6.21   \n",
       "8          NC_008143.1        48.41         5.17   \n",
       "12         NC_035814.1        30.46         4.93   \n",
       "20         NC_050988.1        20.26         4.89   \n",
       "2          NC_040902.1        76.62         4.43   \n",
       "13         NC_042233.1        28.10         4.09   \n",
       "10         NC_020011.1        42.29         2.62   \n",
       "11         NC_022418.1        36.10         2.58   \n",
       "4          NC_007446.1        65.53         2.17   \n",
       "3          NC_030041.1        66.64         2.15   \n",
       "6          NC_010288.1        54.22         2.12   \n",
       "9          NC_037513.1        42.87         2.06   \n",
       "24         NC_057520.1        18.68         1.80   \n",
       "22         NC_057519.1        19.85         1.77   \n",
       "15         NC_013700.1        21.21         1.51   \n",
       "44         NC_026794.1        10.04         1.48   \n",
       "7          NC_006853.1        50.82         1.35   \n",
       "39         NC_008680.1        11.52         1.27   \n",
       "19         NC_027671.1        20.46         1.21   \n",
       "16         NC_029159.1        21.20         1.12   \n",
       "25         NC_021478.1        17.39         1.09   \n",
       "21         NC_031868.1        20.15         1.09   \n",
       "5          NC_023528.1        58.30         1.08   \n",
       "42         NC_026307.1        10.59         0.89   \n",
       "18         NC_020001.1        20.50         0.86   \n",
       "34         NC_026063.1        12.81         0.84   \n",
       "43         NC_066473.1        10.36         0.82   \n",
       "23         NC_002008.4        19.82         0.78   \n",
       "32         NC_003192.1        14.05         0.72   \n",
       "45         NC_026723.1         4.73         0.67   \n",
       "41         NC_024645.1        10.96         0.63   \n",
       "40         NC_001913.1        10.99         0.63   \n",
       "14         NC_023982.1        26.13         0.61   \n",
       "31         NC_012095.1        14.61         0.61   \n",
       "33         NC_008776.1        13.85         0.53   \n",
       "27         NC_014576.1        16.64         0.50   \n",
       "37         NC_010781.1        12.22         0.44   \n",
       "28          KX964606.1        16.44         0.43   \n",
       "29         NC_018367.1        15.83         0.34   \n",
       "26         NC_003408.1        16.91         0.33   \n",
       "17         NC_069631.1        20.67         0.32   \n",
       "35         NC_029382.1        12.73         0.30   \n",
       "38         NC_050263.1        12.04         0.26   \n",
       "46         NC_001700.1         4.24         0.25   \n",
       "30         NC_065160.1        14.93         0.23   \n",
       "47         AC_000022.2         4.23         0.21   \n",
       "48         NC_029888.1         3.87         0.18   \n",
       "36         NC_007692.1        12.43         0.14   \n",
       "50         NC_028308.1         2.43         0.12   \n",
       "51         NC_020645.1         2.20         0.11   \n",
       "54         NC_029403.1         1.09         0.10   \n",
       "49         NC_002080.2         3.28         0.10   \n",
       "53         NC_028161.1         1.52         0.09   \n",
       "56         NC_041068.1         0.73         0.08   \n",
       "55         NC_023459.1         1.02         0.07   \n",
       "57         NC_005033.1         0.66         0.05   \n",
       "52         NC_036430.1         1.77         0.04   \n",
       "58         NC_036369.1         0.36         0.04   \n",
       "59         NC_020637.1         0.28         0.03   \n",
       "\n",
       "                                           species          common_name  \\\n",
       "0                            Osteolaemus tetraspis            crocodile   \n",
       "1                                     Homo sapiens                human   \n",
       "8                               Crocodylus porosus                  NaN   \n",
       "12                        Aonyx cinerea isolate 16                  NaN   \n",
       "20                 Antechinus arktos voucher AA100                  NaN   \n",
       "2                          Gallus gallus spadiceus              chicken   \n",
       "13            Numenius tenuirostris isolate lib4-2                  NaN   \n",
       "10                                 Channa maculata       snakehead fish   \n",
       "11                             Anas poecilorhyncha                 duck   \n",
       "4                                Andrias japonicus  Japanese salamander   \n",
       "3                                     Ptyas mucosa                  NaN   \n",
       "6                          Ctenopharyngodon idella                  NaN   \n",
       "9                            Streptopelia decaocto        collared dove   \n",
       "24                        Lutreolina crassicaudata                  NaN   \n",
       "22                              Didelphis pernigra                  NaN   \n",
       "15                        Nyctereutes procyonoides          raccoon dog   \n",
       "44                                 Colisa fasciata                  NaN   \n",
       "7                                       Bos taurus                  cow   \n",
       "39                             Vaillantella maassi                  NaN   \n",
       "19                            Platymantis vitianus                  NaN   \n",
       "16               Ictalurus cf. pricei USON-01115-1                  NaN   \n",
       "25                              Rhizomys pruinosus     hoary bamboo rat   \n",
       "21                                   Columba jouyi                  NaN   \n",
       "5                                   Rana dybowskii                  NaN   \n",
       "42                          Anoxypristis cuspidata                  NaN   \n",
       "18                           Sooglossus thomasseti                  NaN   \n",
       "34                                 Ovis orientalis                sheep   \n",
       "43                          Tropidophorus hainanus                  NaN   \n",
       "23                          Canis lupus familiaris                  dog   \n",
       "32                                Monopterus albus                  NaN   \n",
       "45                        Urocyon cinereoargenteus             gray fox   \n",
       "41  Culter alburnus x Ancherythroculter nigrocauda                  NaN   \n",
       "40                           Oryctolagus cuniculus               rabbit   \n",
       "14                                Amaurornis akool       brown bush hen   \n",
       "31                           Sus scrofa domesticus                  pig   \n",
       "33                             Heloderma suspectum                  NaN   \n",
       "27                             Chrysolophus pictus      golden pheasant   \n",
       "37                                  Lophura ignita                  NaN   \n",
       "28                             Erinaceus amurensis        Amur hedgehog   \n",
       "29                              Marmota himalayana     Himalayan marmot   \n",
       "26                               Coturnix japonica                  NaN   \n",
       "17                              Terrapene carolina                  NaN   \n",
       "35                             Zebrias crossolepis                  NaN   \n",
       "38                               Hystrix brachyura    Malayan porcupine   \n",
       "46                                     Felis catus                  cat   \n",
       "30                               Tupaia nicobarica                  NaN   \n",
       "47                 Rattus norvegicus strain Wistar            brown rat   \n",
       "48                               Rattus tiomanicus    Mayalan field rat   \n",
       "36                                  Testudo graeca                  NaN   \n",
       "50                                 Felis margarita             sand cat   \n",
       "51                               Arctonyx collaris           hog badger   \n",
       "54                                  Paguma larvata    masked palm civet   \n",
       "49                             Erinaceus europaeus    European hedgehog   \n",
       "53                                  Capra aegagrus                 goat   \n",
       "56                                    Elaphe dione            rat snake   \n",
       "55                                    Vulpes zerda           fennec fox   \n",
       "57                             Hemiechinus auritus  long-eared hedgehog   \n",
       "52                            Muntiacus putaoensis            leaf deer   \n",
       "58                               Otocyon megalotis        bat-eared fox   \n",
       "59                                Mustela sibirica      Siberian weasel   \n",
       "\n",
       "                    reason_kept  \n",
       "0   specified to keep in config  \n",
       "1   specified to keep in config  \n",
       "8       at least 10% in one run  \n",
       "12      at least 10% in one run  \n",
       "20      at least 10% in one run  \n",
       "2   specified to keep in config  \n",
       "13      at least 10% in one run  \n",
       "10  specified to keep in config  \n",
       "11  specified to keep in config  \n",
       "4   specified to keep in config  \n",
       "3       at least 10% in one run  \n",
       "6       at least 10% in one run  \n",
       "9   specified to keep in config  \n",
       "24      at least 10% in one run  \n",
       "22      at least 10% in one run  \n",
       "15  specified to keep in config  \n",
       "44      at least 10% in one run  \n",
       "7   specified to keep in config  \n",
       "39      at least 10% in one run  \n",
       "19      at least 10% in one run  \n",
       "16      at least 10% in one run  \n",
       "25  specified to keep in config  \n",
       "21      at least 10% in one run  \n",
       "5       at least 10% in one run  \n",
       "42      at least 10% in one run  \n",
       "18      at least 10% in one run  \n",
       "34  specified to keep in config  \n",
       "43      at least 10% in one run  \n",
       "23  specified to keep in config  \n",
       "32      at least 10% in one run  \n",
       "45  specified to keep in config  \n",
       "41      at least 10% in one run  \n",
       "40  specified to keep in config  \n",
       "14  specified to keep in config  \n",
       "31  specified to keep in config  \n",
       "33      at least 10% in one run  \n",
       "27  specified to keep in config  \n",
       "37      at least 10% in one run  \n",
       "28  specified to keep in config  \n",
       "29  specified to keep in config  \n",
       "26      at least 10% in one run  \n",
       "17      at least 10% in one run  \n",
       "35      at least 10% in one run  \n",
       "38  specified to keep in config  \n",
       "46  specified to keep in config  \n",
       "30      at least 10% in one run  \n",
       "47  specified to keep in config  \n",
       "48  specified to keep in config  \n",
       "36      at least 10% in one run  \n",
       "50  specified to keep in config  \n",
       "51  specified to keep in config  \n",
       "54  specified to keep in config  \n",
       "49  specified to keep in config  \n",
       "53  specified to keep in config  \n",
       "56  specified to keep in config  \n",
       "55  specified to keep in config  \n",
       "57  specified to keep in config  \n",
       "52  specified to keep in config  \n",
       "58  specified to keep in config  \n",
       "59  specified to keep in config  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mito_max_primary_percent = ( \n",
    "    mito_counts\n",
    "    .query(\"alignment_count_type == 'n_primary'\")\n",
    "    .assign(\n",
    "        total=lambda x: x.groupby(\"Run accession\")[\"alignment_count\"].transform(\"sum\"),\n",
    "        percent=lambda x: 100 * x[\"alignment_count\"] / x[\"total\"],\n",
    "    )\n",
    "    .groupby(\"alignment_reference\", as_index=False)\n",
    "    .aggregate(\n",
    "        max_percent=pd.NamedAgg(\"percent\", \"max\"),\n",
    "        avg_percent=pd.NamedAgg(\"percent\", \"mean\"),\n",
    "    )\n",
    "    .sort_values(\"max_percent\")\n",
    "    .merge(mito_ref_info.rename(columns={\"id\": \"alignment_reference\"}))\n",
    ")\n",
    "\n",
    "min_percent = mito_composition_filters[\"min_percent_aligned_genome_filter\"]\n",
    "mito_to_keep = (\n",
    "    mito_max_primary_percent\n",
    "    .assign(\n",
    "        reason_kept=lambda x: x.apply(\n",
    "            lambda row: (\n",
    "                \"specified to keep in config\"\n",
    "                if any(row[\"species\"].startswith(s) for s in mito_genomes_to_keep.values())\n",
    "                else (\n",
    "                    f\"at least {min_percent}% in one run\"\n",
    "                    if row[\"max_percent\"] >= min_percent\n",
    "                    else\n",
    "                    \"not retained\"\n",
    "                )\n",
    "            ),\n",
    "            axis=1,\n",
    "        ),\n",
    "    )\n",
    "    .query(\"reason_kept != 'not retained'\")\n",
    "    .sort_values(\"max_percent\", ascending=False)\n",
    "    .reset_index(drop=True)\n",
    ")\n",
    "\n",
    "assert all(\n",
    "    any(s_kept.startswith(s_to_keep) for s_kept in mito_to_keep[\"species\"])\n",
    "    for s_to_keep in mito_genomes_to_keep.values()\n",
    ")\n",
    "\n",
    "print(\"Keeping the following mitochondrial genomes:\")\n",
    "display(mito_to_keep.sort_values(\"avg_percent\", ascending=False).round(2))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a92f6fe-1de8-4418-a325-3588589e31c3",
   "metadata": {},
   "source": [
    "Now get just the counts for those mitochondrial genomes to keep:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "382ab5d1-59a0-4969-ae08-54fdb5a924ae",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mito_counts_by_run = (\n",
    "    mito_counts\n",
    "    .merge(\n",
    "        mito_to_keep[[\"alignment_reference\", \"species\", \"common_name\"]],\n",
    "        validate=\"many_to_one\",\n",
    "        on=\"alignment_reference\",\n",
    "        how=\"inner\",\n",
    "    )\n",
    ")\n",
    "\n",
    "mito_counts_by_run.to_csv(\n",
    "    snakemake.output[\"mito_composition_by_metagenomic_run\"], index=False, float_format=\"%.5f\",\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "917eb8e5-1903-4916-b4cb-bc32cc87906c",
   "metadata": {
    "execution": {
     "iopub.status.busy": "2023-04-05T20:15:01.703158Z",
     "iopub.status.idle": "2023-04-05T20:15:01.704024Z",
     "shell.execute_reply": "2023-04-05T20:15:01.703776Z",
     "shell.execute_reply.started": "2023-04-05T20:15:01.703749Z"
    },
    "tags": []
   },
   "source": [
    "Now we get the mitochondrial counts by **sample** (aggregating counts across run for the same sample:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c83b527-11f8-4e80-9376-80addc2ceb38",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# columns that we sum across runs for each sample\n",
    "sum_cols = [\n",
    "    \"alignment_count\",\n",
    "]\n",
    "\n",
    "mito_counts_by_sample = (\n",
    "    mito_counts_by_run\n",
    "    .groupby(\n",
    "        [\n",
    "            c for c in mito_counts_by_run\n",
    "            if c not in sum_cols and c != \"Run accession\"\n",
    "        ],\n",
    "        as_index=False,\n",
    "        dropna=False,\n",
    "    )\n",
    "    .aggregate({c: \"sum\" for c in sum_cols})\n",
    ")\n",
    "\n",
    "dup_samples = (\n",
    "    mito_counts_by_sample\n",
    "    .groupby([\"alignment_reference\", \"alignment_count_type\", \"sample\"])\n",
    "    .aggregate(\n",
    "        n=pd.NamedAgg(\"alignment_count\", \"count\"),\n",
    "        sample_names=pd.NamedAgg(\"Sample name\", \"unique\"),\n",
    "    )\n",
    "    .query(\"n > 1\")\n",
    ")\n",
    "\n",
    "assert not len(dup_samples), f\"Some duplicated sample names\\n{dup_samples}\"\n",
    "\n",
    "mito_counts_by_sample.to_csv(\n",
    "    snakemake.output[\"mito_composition_by_metagenomic_sample\"], index=False, float_format=\"%.5g\",\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cae22a2c-4c9f-4af9-b9f1-661e6998bc72",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
