{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bdf42685-7bac-45b8-b061-0851c35f3a59",
   "metadata": {},
   "source": [
    "# Make plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ba902a21-e1c6-44f4-a59b-69da2d001cc1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:20.209733Z",
     "iopub.status.busy": "2023-04-08T22:51:20.209198Z",
     "iopub.status.idle": "2023-04-08T22:51:20.407894Z",
     "shell.execute_reply": "2023-04-08T22:51:20.406525Z",
     "shell.execute_reply.started": "2023-04-08T22:51:20.209692Z"
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "######## snakemake preamble start (automatically inserted, do not edit) ########\n",
    "import sys; sys.path.extend(['/fh/fast/bloom_j/software/miniconda3/envs/Huanan_market_samples/lib/python3.10/site-packages', '/home/jbloom/.cache/snakemake/snakemake/source-cache/runtime-cache/tmppc1j3g1v/file/fh/fast/bloom_j/computational_notebooks/jbloom/2023/Huanan_market_samples/notebooks', '/fh/fast/bloom_j/computational_notebooks/jbloom/2023/Huanan_market_samples/notebooks']); import pickle; snakemake = pickle.loads(b'\\x80\\x04\\x95\\'\\x12\\x00\\x00\\x00\\x00\\x00\\x00\\x8c\\x10snakemake.script\\x94\\x8c\\tSnakemake\\x94\\x93\\x94)\\x81\\x94}\\x94(\\x8c\\x05input\\x94\\x8c\\x0csnakemake.io\\x94\\x8c\\nInputFiles\\x94\\x93\\x94)\\x81\\x94(\\x8c2results/aggregated_counts/sars2_aligned_by_run.csv\\x94\\x8cAresults/aggregated_counts/sars2_aligned_by_metagenomic_sample.csv\\x94\\x8cAresults/aggregated_counts/mito_composition_by_metagenomic_run.csv\\x94\\x8cDresults/aggregated_counts/mito_composition_by_metagenomic_sample.csv\\x94\\x8c,results/crits_christoph_data/read_counts.csv\\x94\\x8c@results/crits_christoph_data/check_sha512_vs_crits_christoph.csv\\x94e}\\x94(\\x8c\\x06_names\\x94}\\x94(\\x8c\\x14sars2_aligned_by_run\\x94K\\x00N\\x86\\x94\\x8c#sars2_aligned_by_metagenomic_sample\\x94K\\x01N\\x86\\x94\\x8c#mito_composition_by_metagenomic_run\\x94K\\x02N\\x86\\x94\\x8c&mito_composition_by_metagenomic_sample\\x94K\\x03N\\x86\\x94\\x8c\\x1bcrits_christoph_read_counts\\x94K\\x04N\\x86\\x94\\x8c\\x17ngdc_to_crits_christoph\\x94K\\x05N\\x86\\x94u\\x8c\\x12_allowed_overrides\\x94]\\x94(\\x8c\\x05index\\x94\\x8c\\x04sort\\x94eh!\\x8c\\tfunctools\\x94\\x8c\\x07partial\\x94\\x93\\x94h\\x06\\x8c\\x19Namedlist._used_attribute\\x94\\x93\\x94\\x85\\x94R\\x94(h\\')}\\x94\\x8c\\x05_name\\x94h!sNt\\x94bh\"h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h\"sNt\\x94b\\x8c\\x14sars2_aligned_by_run\\x94h\\n\\x8c#sars2_aligned_by_metagenomic_sample\\x94h\\x0b\\x8c#mito_composition_by_metagenomic_run\\x94h\\x0c\\x8c&mito_composition_by_metagenomic_sample\\x94h\\rh\\x1bh\\x0eh\\x1dh\\x0fub\\x8c\\x06output\\x94h\\x06\\x8c\\x0bOutputFiles\\x94\\x93\\x94)\\x81\\x94\\x8c\\rresults/plots\\x94a}\\x94(h\\x11}\\x94\\x8c\\x08plotsdir\\x94K\\x00N\\x86\\x94sh\\x1f]\\x94(h!h\"eh!h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h!sNt\\x94bh\"h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h\"sNt\\x94bh<h9ub\\x8c\\x06params\\x94h\\x06\\x8c\\x06Params\\x94\\x93\\x94)\\x81\\x94}\\x94(h\\x11}\\x94h\\x1f]\\x94(h!h\"eh!h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h!sNt\\x94bh\"h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h\"sNt\\x94bub\\x8c\\twildcards\\x94h\\x06\\x8c\\tWildcards\\x94\\x93\\x94)\\x81\\x94}\\x94(h\\x11}\\x94h\\x1f]\\x94(h!h\"eh!h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h!sNt\\x94bh\"h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h\"sNt\\x94bub\\x8c\\x07threads\\x94K\\x01\\x8c\\tresources\\x94h\\x06\\x8c\\tResources\\x94\\x93\\x94)\\x81\\x94(K\\x01K\\x01M\\xe8\\x03M\\xba\\x03M\\xe8\\x03M\\xba\\x03\\x8c\\x15/loc/scratch/17960424\\x94e}\\x94(h\\x11}\\x94(\\x8c\\x06_cores\\x94K\\x00N\\x86\\x94\\x8c\\x06_nodes\\x94K\\x01N\\x86\\x94\\x8c\\x06mem_mb\\x94K\\x02N\\x86\\x94\\x8c\\x07mem_mib\\x94K\\x03N\\x86\\x94\\x8c\\x07disk_mb\\x94K\\x04N\\x86\\x94\\x8c\\x08disk_mib\\x94K\\x05N\\x86\\x94\\x8c\\x06tmpdir\\x94K\\x06N\\x86\\x94uh\\x1f]\\x94(h!h\"eh!h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h!sNt\\x94bh\"h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h\"sNt\\x94bhmK\\x01hoK\\x01hqM\\xe8\\x03hsM\\xba\\x03huM\\xe8\\x03hwM\\xba\\x03hyhjub\\x8c\\x03log\\x94h\\x06\\x8c\\x03Log\\x94\\x93\\x94)\\x81\\x94\\x8c\\x1eresults/plots/make_plots.ipynb\\x94a}\\x94(h\\x11}\\x94\\x8c\\x08notebook\\x94K\\x00N\\x86\\x94sh\\x1f]\\x94(h!h\"eh!h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h!sNt\\x94bh\"h%h\\'\\x85\\x94R\\x94(h\\')}\\x94h+h\"sNt\\x94bh\\x8bh\\x88ub\\x8c\\x06config\\x94}\\x94(\\x8c\\x14crits_christoph_data\\x94}\\x94(\\x8c\\tfile_info\\x94\\x8c\\x95https://raw.githubusercontent.com/sars-cov-2-origins/huanan-environmental/bd2afdb19392a0efb872c09eb94045de9d265a6e/tables/TableS1_FileInformation.csv\\x94\\x8c\\x0csample_names\\x94\\x8c\\x91https://raw.githubusercontent.com/sars-cov-2-origins/huanan-environmental/bd2afdb19392a0efb872c09eb94045de9d265a6e/tables/TableS2_SampleNames.csv\\x94\\x8c\\x0bread_counts\\x94\\x8c\\x96https://raw.githubusercontent.com/sars-cov-2-origins/huanan-environmental/bd2afdb19392a0efb872c09eb94045de9d265a6e/tables/TableS3_MappedReadCounts.csv\\x94u\\x8c\\x15mitochondrial_genomes\\x94\\x8cYhttps://ftp.ncbi.nlm.nih.gov/refseq/release/mitochondrion/mitochondrion.1.genomic.gbff.gz\\x94\\x8c\\x1bextra_mitochondrial_genomes\\x94]\\x94\\x8c\\nKX964606.1\\x94a\\x8c\\x1amitochondrial_taxa_to_keep\\x94}\\x94(\\x8c\\x06phylum\\x94]\\x94\\x8c\\x08Chordata\\x94a\\x8c\\tsubphylum\\x94]\\x94u\\x8c\\x1dmitochondrial_genomes_to_keep\\x94}\\x94(\\x8c\\rAmur hedgehog\\x94\\x8c\\x13Erinaceus amurensis\\x94\\x8c\\x03pig\\x94\\x8c\\x15Sus scrofa domesticus\\x94\\x8c\\tbrown rat\\x94\\x8c\\x1fRattus norvegicus strain Wistar\\x94\\x8c\\x06rabbit\\x94\\x8c\\x15Oryctolagus cuniculus\\x94\\x8c\\x03dog\\x94\\x8c\\x16Canis lupus familiaris\\x94\\x8c\\x11European hedgehog\\x94\\x8c\\x13Erinaceus europaeus\\x94\\x8c\\x13long-eared hedgehog\\x94\\x8c\\x13Hemiechinus auritus\\x94\\x8c\\x03cow\\x94\\x8c\\nBos taurus\\x94\\x8c\\x05human\\x94\\x8c\\x0cHomo sapiens\\x94\\x8c\\x0braccoon dog\\x94\\x8c\\x18Nyctereutes procyonoides\\x94\\x8c\\x10Himalayan marmot\\x94\\x8c\\x12Marmota himalayana\\x94\\x8c\\x0fSiberian weasel\\x94\\x8c\\x10Mustela sibirica\\x94\\x8c\\nhog badger\\x94\\x8c\\x11Arctonyx collaris\\x94\\x8c\\x10hoary bamboo rat\\x94\\x8c\\x12Rhizomys pruinosus\\x94\\x8c\\nfennec fox\\x94\\x8c\\x0cVulpes zerda\\x94\\x8c\\x05sheep\\x94\\x8c\\x0fOvis orientalis\\x94\\x8c\\x08gray fox\\x94\\x8c\\x18Urocyon cinereoargenteus\\x94\\x8c\\x04goat\\x94\\x8c\\x0eCapra aegagrus\\x94\\x8c\\x08sand cat\\x94\\x8c\\x0fFelis margarita\\x94\\x8c\\x11masked palm civet\\x94\\x8c\\x0ePaguma larvata\\x94\\x8c\\x11Mayalan field rat\\x94\\x8c\\x11Rattus tiomanicus\\x94\\x8c\\rbat-eared fox\\x94\\x8c\\x11Otocyon megalotis\\x94\\x8c\\tleaf deer\\x94\\x8c\\x14Muntiacus putaoensis\\x94\\x8c\\x11Malayan porcupine\\x94\\x8c\\x11Hystrix brachyura\\x94\\x8c\\x05quail\\x94\\x8c\\x11Coturnix japonica\\x94\\x8c\\x15yellow croaker (fish)\\x94\\x8c\\x13Larimichthys crocea\\x94\\x8c\\x0cspotted dove\\x94\\x8c\\x16Streptopelia chinensis\\x94\\x8c\\tswamp eel\\x94\\x8c\\x10Monopterus albus\\x94\\x8c\\x07chicken\\x94\\x8c\\x17Gallus gallus spadiceus\\x94\\x8c\\x12Oriental rat snake\\x94\\x8c\\x0cPtyas mucosa\\x94\\x8c\\x04carp\\x94\\x8c\\x17Ctenopharyngodon idella\\x94\\x8c\\x0esnakehead fish\\x94\\x8c\\x0fChanna maculata\\x94\\x8c\\x10spot-billed duck\\x94\\x8c\\x13Anas poecilorhyncha\\x94\\x8c\\x07catfish\\x94\\x8c\\x13Ictalurus punctatus\\x94\\x8c\\x12Chinese salamander\\x94\\x8c\\x12Andrias davidianus\\x94\\x8c\\x0ebrown bush hen\\x94\\x8c\\x10Amaurornis akool\\x94\\x8c\\rChinese cobra\\x94\\x8c\\tNaja atra\\x94\\x8c\\x0fLargemouth bass\\x94\\x8c\\x1fMicropterus salmoides salmoides\\x94\\x8c\\x06pigeon\\x94\\x8c\\x11Columba rupestris\\x94u\\x8c\"mitochondrial_genome_min_mash_dist\\x94G?\\xb1\\xeb\\x85\\x1e\\xb8Q\\xec\\x8c\\tsars2_ref\\x94\\x8cGhttp://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/bigZips/wuhCor1.fa.gz\\x94\\x8c\\x0csars2_ref_id\\x94\\x8c\\x0bNC_045512v2\\x94\\x8c\\x16sars2_ref_trim_3_polyA\\x94\\x88\\x8c\\x08min_mapq\\x94K\\x04\\x8c\\x0ccoverm_flags\\x94}\\x94(\\x8c\\x17min_read_aligned_length\\x94K(\\x8c\\x14contig_end_exclusion\\x94Kd\\x8c\\x19min_read_percent_identity\\x94G?\\xeeffffffu\\x8c\\x18metagenomic_descriptions\\x94]\\x94\\x8cORNA sequencing of total nucleic acids from environmental swabs for metagenomics\\x94a\\x8c\\x18mito_composition_filters\\x94}\\x94(\\x8c\\x19min_alignments_run_filter\\x94M\\xf4\\x01\\x8c!min_percent_aligned_genome_filter\\x94K\\x14\\x8c\"min_coverage_aligned_genome_filter\\x94M\\xa0\\x0fuu\\x8c\\x04rule\\x94\\x8c\\nmake_plots\\x94\\x8c\\x0fbench_iteration\\x94N\\x8c\\tscriptdir\\x94\\x8cT/fh/fast/bloom_j/computational_notebooks/jbloom/2023/Huanan_market_samples/notebooks\\x94ub.'); from snakemake.logging import logger; logger.printshellcmds = False; import os; os.chdir(r'/fh/fast/bloom_j/computational_notebooks/jbloom/2023/Huanan_market_samples');\n",
    "######## snakemake preamble end #########\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30fe09f9-1157-4594-a37d-c60db2190d6c",
   "metadata": {},
   "source": [
    "Import Python modules:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "814bf235-7c93-47f7-adbc-ba242b931670",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:20.413241Z",
     "iopub.status.busy": "2023-04-08T22:51:20.412882Z",
     "iopub.status.idle": "2023-04-08T22:51:21.593793Z",
     "shell.execute_reply": "2023-04-08T22:51:21.592523Z",
     "shell.execute_reply.started": "2023-04-08T22:51:20.413205Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "import itertools\n",
    "import os\n",
    "\n",
    "import altair as alt\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "_ = alt.data_transformers.disable_max_rows()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "897e563c-8a14-48e7-958c-4c69e57bfd69",
   "metadata": {},
   "source": [
    "Get data and variable from `snakemake`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1e4be094-39bd-4972-b49d-76e3b400d3ab",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:21.599071Z",
     "iopub.status.busy": "2023-04-08T22:51:21.598652Z",
     "iopub.status.idle": "2023-04-08T22:51:21.685805Z",
     "shell.execute_reply": "2023-04-08T22:51:21.684809Z",
     "shell.execute_reply.started": "2023-04-08T22:51:21.599043Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "sars2_aligned_by_run = pd.read_csv(snakemake.input.sars2_aligned_by_run)\n",
    "\n",
    "sars2_aligned_by_metagenomic_sample = pd.read_csv(snakemake.input.sars2_aligned_by_metagenomic_sample)\n",
    "\n",
    "mito_composition_by_metagenomic_run = pd.read_csv(snakemake.input.mito_composition_by_metagenomic_run)\n",
    "\n",
    "mito_composition_by_metagenomic_sample = pd.read_csv(snakemake.input.mito_composition_by_metagenomic_sample)\n",
    "\n",
    "crits_christoph_read_counts = pd.read_csv(snakemake.input.crits_christoph_read_counts)\n",
    "\n",
    "ngdc_to_crits_christoph = pd.read_csv(snakemake.input.ngdc_to_crits_christoph)\n",
    "\n",
    "plotsdir = snakemake.output.plotsdir"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f887eb07-373c-42b4-9717-5b7702b84f0c",
   "metadata": {},
   "source": [
    "## Compare mitochondrial DNA composition to Crits-Christoph et al\n",
    "First get Crits-Christoph read counts in tidy format, assigning NGDC run accessions and summing counts for runs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b2a9fb08-f0fd-4539-9e7a-59f1ae2a9c44",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:21.691636Z",
     "iopub.status.busy": "2023-04-08T22:51:21.691358Z",
     "iopub.status.idle": "2023-04-08T22:51:21.728340Z",
     "shell.execute_reply": "2023-04-08T22:51:21.727356Z",
     "shell.execute_reply.started": "2023-04-08T22:51:21.691614Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Run accession</th>\n",
       "      <th>species</th>\n",
       "      <th>aligned reads Crits-Christoph</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Arctonyx collaris</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Canis lupus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Capra aegagrus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Erinaceus amurensis</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2923</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Rattus tiomanicus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2924</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Rhizomys pruinosus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2925</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Sus scrofa</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2926</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2927</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Vulpes zerda</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2928 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample Run accession                   species  \\\n",
       "0       1-29-8     CRR710770         Arctonyx collaris   \n",
       "1       1-29-8     CRR710770                Bos taurus   \n",
       "2       1-29-8     CRR710770               Canis lupus   \n",
       "3       1-29-8     CRR710770            Capra aegagrus   \n",
       "4       1-29-8     CRR710770       Erinaceus amurensis   \n",
       "...        ...           ...                       ...   \n",
       "2923  wws-1#-0     CRR711161         Rattus tiomanicus   \n",
       "2924  wws-1#-0     CRR711161        Rhizomys pruinosus   \n",
       "2925  wws-1#-0     CRR711161                Sus scrofa   \n",
       "2926  wws-1#-0     CRR711161  Urocyon cinereoargenteus   \n",
       "2927  wws-1#-0     CRR711161              Vulpes zerda   \n",
       "\n",
       "      aligned reads Crits-Christoph  \n",
       "0                                 0  \n",
       "1                                 0  \n",
       "2                                 0  \n",
       "3                                 0  \n",
       "4                                 0  \n",
       "...                             ...  \n",
       "2923                              0  \n",
       "2924                              0  \n",
       "2925                             32  \n",
       "2926                              0  \n",
       "2927                              0  \n",
       "\n",
       "[2928 rows x 4 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crits_christoph_read_counts_tidy = (\n",
    "    crits_christoph_read_counts\n",
    "    .drop(columns=[\"Location\", \"Sample_name\"])\n",
    "    .merge(\n",
    "        ngdc_to_crits_christoph\n",
    "        .assign(Filename=lambda x: x[\"fastq Crits-Christoph\"].map(lambda s: s.split(\".\")[0]))\n",
    "        .rename(columns={\"Run accession NGDC\": \"Run accession\"})\n",
    "        [[\"Filename\", \"Run accession\"]]\n",
    "        .drop_duplicates(),\n",
    "        validate=\"one_to_one\",\n",
    "    )\n",
    "    .merge(\n",
    "        sars2_aligned_by_run[[\"Run accession\", \"sample\"]].drop_duplicates(),\n",
    "        validate=\"many_to_one\",\n",
    "        how=\"left\",\n",
    "    )\n",
    "    .melt(\n",
    "        id_vars=[\"sample\", \"Run accession\", \"Filename\"],\n",
    "        var_name=\"species\",\n",
    "        value_name=\"aligned reads Crits-Christoph\",\n",
    "    )\n",
    "    .groupby([\"sample\", \"Run accession\", \"species\"], as_index=False)\n",
    "    .aggregate({\"aligned reads Crits-Christoph\": \"sum\"})\n",
    ")\n",
    "\n",
    "crits_christoph_read_counts_tidy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8c67148f-2ca1-4a69-a8ae-3ead3f37c31b",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:21.732396Z",
     "iopub.status.busy": "2023-04-08T22:51:21.731980Z",
     "iopub.status.idle": "2023-04-08T22:51:21.737787Z",
     "shell.execute_reply": "2023-04-08T22:51:21.736970Z",
     "shell.execute_reply.started": "2023-04-08T22:51:21.732369Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Run accession', 'reference_id', 'aligned_reads', 'covered_bases',\n",
       "       'species', 'common_name', 'SARS2_aligned_reads', 'total_reads',\n",
       "       'preprocessed_reads', 'sample', 'Sample name', 'Collection date',\n",
       "       'description', 'Isolation source'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    " mito_composition_by_metagenomic_run.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81141895-958a-4cc7-b1d0-02e33f8c0064",
   "metadata": {},
   "source": [
    "Now merge current read counts with those from Crits-Christoph for all species:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "116340c5-ebe7-4da3-8472-d1acf24694ef",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:21.741547Z",
     "iopub.status.busy": "2023-04-08T22:51:21.741189Z",
     "iopub.status.idle": "2023-04-08T22:51:21.761670Z",
     "shell.execute_reply": "2023-04-08T22:51:21.760534Z",
     "shell.execute_reply.started": "2023-04-08T22:51:21.741521Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_species = crits_christoph_read_counts_tidy[\"species\"].unique().tolist()\n",
    "\n",
    "mito_composition_to_compare = (\n",
    "    mito_composition_by_metagenomic_run\n",
    "    [[\"Run accession\", \"species\", \"common_name\", \"aligned_reads\", \"covered_bases\"]]\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"aligned_reads\": \"aligned reads current study\",\n",
    "            \"covered_bases\": \"covered bases current study\",\n",
    "        }\n",
    "    )\n",
    ")\n",
    "\n",
    "assert set(crits_christoph_species).issubset(mito_composition_to_compare[\"species\"])\n",
    "\n",
    "crits_christoph_vs_current = (\n",
    "    crits_christoph_read_counts_tidy\n",
    "    .merge(\n",
    "        mito_composition_to_compare,\n",
    "        validate=\"one_to_one\",\n",
    "        on=[\"Run accession\", \"species\"],\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f2893ad-9059-4a99-b026-971364fc4177",
   "metadata": {},
   "source": [
    "Now get the correlations by species and plot them:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fad10471-0e9e-47e4-b488-a89c2dc09d0a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:21.765647Z",
     "iopub.status.busy": "2023-04-08T22:51:21.765305Z",
     "iopub.status.idle": "2023-04-08T22:51:21.800037Z",
     "shell.execute_reply": "2023-04-08T22:51:21.799275Z",
     "shell.execute_reply.started": "2023-04-08T22:51:21.765626Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>species</th>\n",
       "      <th>common_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>aligned_reads_Crits_Christoph</th>\n",
       "      <th>aligned_reads_current_study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>0.055346</td>\n",
       "      <td>552</td>\n",
       "      <td>400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>brown rat</td>\n",
       "      <td>0.537612</td>\n",
       "      <td>10141</td>\n",
       "      <td>175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Rattus tiomanicus</td>\n",
       "      <td>Mayalan field rat</td>\n",
       "      <td>0.757205</td>\n",
       "      <td>394</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Mustela sibirica</td>\n",
       "      <td>Siberian weasel</td>\n",
       "      <td>0.929420</td>\n",
       "      <td>216</td>\n",
       "      <td>238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Arctonyx collaris</td>\n",
       "      <td>hog badger</td>\n",
       "      <td>0.972038</td>\n",
       "      <td>188</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Vulpes zerda</td>\n",
       "      <td>fennec fox</td>\n",
       "      <td>0.978385</td>\n",
       "      <td>165</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Capra aegagrus</td>\n",
       "      <td>goat</td>\n",
       "      <td>0.985783</td>\n",
       "      <td>2710</td>\n",
       "      <td>2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Otocyon megalotis</td>\n",
       "      <td>bat-eared fox</td>\n",
       "      <td>0.985863</td>\n",
       "      <td>255</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Muntiacus putaoensis</td>\n",
       "      <td>leaf deer</td>\n",
       "      <td>0.987589</td>\n",
       "      <td>89</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Oryctolagus cuniculus</td>\n",
       "      <td>rabbit</td>\n",
       "      <td>0.989770</td>\n",
       "      <td>1018</td>\n",
       "      <td>846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Felis margarita</td>\n",
       "      <td>sand cat</td>\n",
       "      <td>0.992632</td>\n",
       "      <td>914</td>\n",
       "      <td>966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Paguma larvata</td>\n",
       "      <td>masked palm civet</td>\n",
       "      <td>0.993973</td>\n",
       "      <td>536</td>\n",
       "      <td>409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Rhizomys pruinosus</td>\n",
       "      <td>hoary bamboo rat</td>\n",
       "      <td>0.994724</td>\n",
       "      <td>1263</td>\n",
       "      <td>813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Ovis orientalis</td>\n",
       "      <td>sheep</td>\n",
       "      <td>0.995994</td>\n",
       "      <td>28274</td>\n",
       "      <td>28518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Canis lupus</td>\n",
       "      <td>dog</td>\n",
       "      <td>0.997246</td>\n",
       "      <td>7941</td>\n",
       "      <td>5561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Hemiechinus auritus</td>\n",
       "      <td>long-eared hedgehog</td>\n",
       "      <td>0.997643</td>\n",
       "      <td>151</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Erinaceus europaeus</td>\n",
       "      <td>European hedgehog</td>\n",
       "      <td>0.997962</td>\n",
       "      <td>1236</td>\n",
       "      <td>284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>human</td>\n",
       "      <td>0.999483</td>\n",
       "      <td>92592</td>\n",
       "      <td>79912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Sus scrofa</td>\n",
       "      <td>pig</td>\n",
       "      <td>0.999515</td>\n",
       "      <td>9821</td>\n",
       "      <td>10016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Erinaceus amurensis</td>\n",
       "      <td>Amur hedgehog</td>\n",
       "      <td>0.999833</td>\n",
       "      <td>7035</td>\n",
       "      <td>2831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>cow</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>61287</td>\n",
       "      <td>61970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Hystrix brachyura</td>\n",
       "      <td>Malayan porcupine</td>\n",
       "      <td>0.999996</td>\n",
       "      <td>6022</td>\n",
       "      <td>2713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Nyctereutes procyonoides</td>\n",
       "      <td>raccoon dog</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>17086</td>\n",
       "      <td>10223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Marmota himalayana</td>\n",
       "      <td>Himalayan marmot</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>796</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     species          common_name  correlation  \\\n",
       "22  Urocyon cinereoargenteus             gray fox     0.055346   \n",
       "18         Rattus norvegicus            brown rat     0.537612   \n",
       "19         Rattus tiomanicus    Mayalan field rat     0.757205   \n",
       "12          Mustela sibirica      Siberian weasel     0.929420   \n",
       "0          Arctonyx collaris           hog badger     0.972038   \n",
       "23              Vulpes zerda           fennec fox     0.978385   \n",
       "3             Capra aegagrus                 goat     0.985783   \n",
       "15         Otocyon megalotis        bat-eared fox     0.985863   \n",
       "11      Muntiacus putaoensis            leaf deer     0.987589   \n",
       "14     Oryctolagus cuniculus               rabbit     0.989770   \n",
       "6            Felis margarita             sand cat     0.992632   \n",
       "17            Paguma larvata    masked palm civet     0.993973   \n",
       "20        Rhizomys pruinosus     hoary bamboo rat     0.994724   \n",
       "16           Ovis orientalis                sheep     0.995994   \n",
       "2                Canis lupus                  dog     0.997246   \n",
       "7        Hemiechinus auritus  long-eared hedgehog     0.997643   \n",
       "5        Erinaceus europaeus    European hedgehog     0.997962   \n",
       "8               Homo sapiens                human     0.999483   \n",
       "21                Sus scrofa                  pig     0.999515   \n",
       "4        Erinaceus amurensis        Amur hedgehog     0.999833   \n",
       "1                 Bos taurus                  cow     0.999990   \n",
       "9          Hystrix brachyura    Malayan porcupine     0.999996   \n",
       "13  Nyctereutes procyonoides          raccoon dog     0.999999   \n",
       "10        Marmota himalayana     Himalayan marmot     1.000000   \n",
       "\n",
       "    aligned_reads_Crits_Christoph  aligned_reads_current_study  \n",
       "22                            552                          400  \n",
       "18                          10141                          175  \n",
       "19                            394                          229  \n",
       "12                            216                          238  \n",
       "0                             188                          102  \n",
       "23                            165                           17  \n",
       "3                            2710                         2394  \n",
       "15                            255                            9  \n",
       "11                             89                           29  \n",
       "14                           1018                          846  \n",
       "6                             914                          966  \n",
       "17                            536                          409  \n",
       "20                           1263                          813  \n",
       "16                          28274                        28518  \n",
       "2                            7941                         5561  \n",
       "7                             151                           56  \n",
       "5                            1236                          284  \n",
       "8                           92592                        79912  \n",
       "21                           9821                        10016  \n",
       "4                            7035                         2831  \n",
       "1                           61287                        61970  \n",
       "9                            6022                         2713  \n",
       "13                          17086                        10223  \n",
       "10                            796                          476  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crits_christoph_vs_current_species_corr = (\n",
    "    crits_christoph_vs_current\n",
    "    .groupby([\"species\", \"common_name\"])\n",
    "    [[\"aligned reads Crits-Christoph\", \"aligned reads current study\"]]\n",
    "    .corr(method=\"pearson\")\n",
    "    .reset_index()\n",
    "    .query(\"level_2 == 'aligned reads Crits-Christoph'\")\n",
    "    .rename(columns={\"aligned reads current study\": \"correlation\"})\n",
    "    .drop(columns=[\"level_2\", \"aligned reads Crits-Christoph\"])\n",
    "    .query(\"correlation.notnull()\")\n",
    "    .merge(\n",
    "        crits_christoph_vs_current\n",
    "        .groupby(\"species\", as_index=False)\n",
    "        .aggregate(\n",
    "            aligned_reads_Crits_Christoph=pd.NamedAgg(\"aligned reads Crits-Christoph\", \"sum\"),\n",
    "            aligned_reads_current_study=pd.NamedAgg(\"aligned reads current study\", \"sum\"),\n",
    "        )\n",
    "    )\n",
    "    .sort_values(\"correlation\")\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_species_corr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "14913bb6-f68f-4c02-9fe9-fb824f2a28ec",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:21.804100Z",
     "iopub.status.busy": "2023-04-08T22:51:21.803343Z",
     "iopub.status.idle": "2023-04-08T22:51:21.886483Z",
     "shell.execute_reply": "2023-04-08T22:51:21.885367Z",
     "shell.execute_reply.started": "2023-04-08T22:51:21.804074Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<div id=\"altair-viz-7bea589bc21c4108b4d85d066a3a8377\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-7bea589bc21c4108b4d85d066a3a8377\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-7bea589bc21c4108b4d85d066a3a8377\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm/vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm/vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm/vega-lite@5.6.1?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm/vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"5.6.1\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 300, \"continuousHeight\": 300}}, \"data\": {\"name\": \"data-1c60e67cd111be1144c908ffe786b4d7\"}, \"mark\": {\"type\": \"circle\", \"opacity\": 1, \"size\": 50}, \"encoding\": {\"tooltip\": [{\"field\": \"species\", \"type\": \"nominal\"}, {\"field\": \"common_name\", \"type\": \"nominal\"}, {\"field\": \"correlation\", \"format\": \".3g\", \"type\": \"quantitative\"}, {\"field\": \"aligned_reads_Crits_Christoph\", \"type\": \"quantitative\"}, {\"field\": \"aligned_reads_current_study\", \"type\": \"quantitative\"}], \"x\": {\"field\": \"correlation\", \"type\": \"quantitative\"}, \"y\": {\"field\": \"common_name\", \"sort\": {\"field\": \"correlation\", \"order\": \"descending\"}, \"title\": null, \"type\": \"nominal\"}}, \"height\": {\"step\": 13}, \"width\": 185, \"$schema\": \"https://vega.github.io/schema/vega-lite/v5.6.1.json\", \"datasets\": {\"data-1c60e67cd111be1144c908ffe786b4d7\": [{\"species\": \"Urocyon cinereoargenteus\", \"common_name\": \"gray fox\", \"correlation\": 0.05534568459553415, \"aligned_reads_Crits_Christoph\": 552, \"aligned_reads_current_study\": 400}, {\"species\": \"Rattus norvegicus\", \"common_name\": \"brown rat\", \"correlation\": 0.5376121428819014, \"aligned_reads_Crits_Christoph\": 10141, \"aligned_reads_current_study\": 175}, {\"species\": \"Rattus tiomanicus\", \"common_name\": \"Mayalan field rat\", \"correlation\": 0.7572046987939867, \"aligned_reads_Crits_Christoph\": 394, \"aligned_reads_current_study\": 229}, {\"species\": \"Mustela sibirica\", \"common_name\": \"Siberian weasel\", \"correlation\": 0.9294196752574672, \"aligned_reads_Crits_Christoph\": 216, \"aligned_reads_current_study\": 238}, {\"species\": \"Arctonyx collaris\", \"common_name\": \"hog badger\", \"correlation\": 0.9720377466046901, \"aligned_reads_Crits_Christoph\": 188, \"aligned_reads_current_study\": 102}, {\"species\": \"Vulpes zerda\", \"common_name\": \"fennec fox\", \"correlation\": 0.9783851329968032, \"aligned_reads_Crits_Christoph\": 165, \"aligned_reads_current_study\": 17}, {\"species\": \"Capra aegagrus\", \"common_name\": \"goat\", \"correlation\": 0.9857826685244839, \"aligned_reads_Crits_Christoph\": 2710, \"aligned_reads_current_study\": 2394}, {\"species\": \"Otocyon megalotis\", \"common_name\": \"bat-eared fox\", \"correlation\": 0.9858627896553395, \"aligned_reads_Crits_Christoph\": 255, \"aligned_reads_current_study\": 9}, {\"species\": \"Muntiacus putaoensis\", \"common_name\": \"leaf deer\", \"correlation\": 0.9875892569606639, \"aligned_reads_Crits_Christoph\": 89, \"aligned_reads_current_study\": 29}, {\"species\": \"Oryctolagus cuniculus\", \"common_name\": \"rabbit\", \"correlation\": 0.9897696192491712, \"aligned_reads_Crits_Christoph\": 1018, \"aligned_reads_current_study\": 846}, {\"species\": \"Felis margarita\", \"common_name\": \"sand cat\", \"correlation\": 0.9926318834553671, \"aligned_reads_Crits_Christoph\": 914, \"aligned_reads_current_study\": 966}, {\"species\": \"Paguma larvata\", \"common_name\": \"masked palm civet\", \"correlation\": 0.9939726979618859, \"aligned_reads_Crits_Christoph\": 536, \"aligned_reads_current_study\": 409}, {\"species\": \"Rhizomys pruinosus\", \"common_name\": \"hoary bamboo rat\", \"correlation\": 0.9947241311849124, \"aligned_reads_Crits_Christoph\": 1263, \"aligned_reads_current_study\": 813}, {\"species\": \"Ovis orientalis\", \"common_name\": \"sheep\", \"correlation\": 0.9959944840045968, \"aligned_reads_Crits_Christoph\": 28274, \"aligned_reads_current_study\": 28518}, {\"species\": \"Canis lupus\", \"common_name\": \"dog\", \"correlation\": 0.9972461732095451, \"aligned_reads_Crits_Christoph\": 7941, \"aligned_reads_current_study\": 5561}, {\"species\": \"Hemiechinus auritus\", \"common_name\": \"long-eared hedgehog\", \"correlation\": 0.9976429551184095, \"aligned_reads_Crits_Christoph\": 151, \"aligned_reads_current_study\": 56}, {\"species\": \"Erinaceus europaeus\", \"common_name\": \"European hedgehog\", \"correlation\": 0.997961686107016, \"aligned_reads_Crits_Christoph\": 1236, \"aligned_reads_current_study\": 284}, {\"species\": \"Homo sapiens\", \"common_name\": \"human\", \"correlation\": 0.9994825968333004, \"aligned_reads_Crits_Christoph\": 92592, \"aligned_reads_current_study\": 79912}, {\"species\": \"Sus scrofa\", \"common_name\": \"pig\", \"correlation\": 0.9995149182888202, \"aligned_reads_Crits_Christoph\": 9821, \"aligned_reads_current_study\": 10016}, {\"species\": \"Erinaceus amurensis\", \"common_name\": \"Amur hedgehog\", \"correlation\": 0.999832609744006, \"aligned_reads_Crits_Christoph\": 7035, \"aligned_reads_current_study\": 2831}, {\"species\": \"Bos taurus\", \"common_name\": \"cow\", \"correlation\": 0.999989794817, \"aligned_reads_Crits_Christoph\": 61287, \"aligned_reads_current_study\": 61970}, {\"species\": \"Hystrix brachyura\", \"common_name\": \"Malayan porcupine\", \"correlation\": 0.9999955814005329, \"aligned_reads_Crits_Christoph\": 6022, \"aligned_reads_current_study\": 2713}, {\"species\": \"Nyctereutes procyonoides\", \"common_name\": \"raccoon dog\", \"correlation\": 0.9999987522581765, \"aligned_reads_Crits_Christoph\": 17086, \"aligned_reads_current_study\": 10223}, {\"species\": \"Marmota himalayana\", \"common_name\": \"Himalayan marmot\", \"correlation\": 0.9999996487023596, \"aligned_reads_Crits_Christoph\": 796, \"aligned_reads_current_study\": 476}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crits_christoph_vs_current_species_corr_chart = (\n",
    "    alt.Chart(crits_christoph_vs_current_species_corr)\n",
    "    .encode(\n",
    "        x=\"correlation\",\n",
    "        y=alt.Y(\n",
    "            \"common_name\",\n",
    "            sort=alt.SortField(\"correlation\", order=\"descending\"),\n",
    "            title=None,\n",
    "        ),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".3g\")\n",
    "            if crits_christoph_vs_current_species_corr[c].dtype == float\n",
    "            else c\n",
    "            for c in crits_christoph_vs_current_species_corr.columns.tolist()\n",
    "        ],\n",
    "    )\n",
    "    .mark_circle(size=50, opacity=1)\n",
    "    .properties(\n",
    "        height=alt.Step(13),\n",
    "        width=185,\n",
    "    )\n",
    ") \n",
    "    \n",
    "crits_christoph_vs_current_species_corr_chart"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08f3611f-4831-4925-a23b-c6fc817cd43d",
   "metadata": {},
   "source": [
    "Now get the correlations by run:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "197c548e-b73a-46e8-ae3b-c3401ed36d6c",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:51:21.890993Z",
     "iopub.status.busy": "2023-04-08T22:51:21.890635Z",
     "iopub.status.idle": "2023-04-08T22:51:21.938097Z",
     "shell.execute_reply": "2023-04-08T22:51:21.937082Z",
     "shell.execute_reply.started": "2023-04-08T22:51:21.890971Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run accession</th>\n",
       "      <th>sample</th>\n",
       "      <th>correlation</th>\n",
       "      <th>aligned_reads_Crits_Christoph</th>\n",
       "      <th>aligned_reads_current_study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>CRR710905</td>\n",
       "      <td>F54</td>\n",
       "      <td>-0.004219</td>\n",
       "      <td>5</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CRR710804</td>\n",
       "      <td>WWS-2#-3</td>\n",
       "      <td>0.748373</td>\n",
       "      <td>24</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>CRR710861</td>\n",
       "      <td>A90</td>\n",
       "      <td>0.903233</td>\n",
       "      <td>13</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>CRR710839</td>\n",
       "      <td>A33</td>\n",
       "      <td>0.910729</td>\n",
       "      <td>30</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>CRR710860</td>\n",
       "      <td>A90</td>\n",
       "      <td>0.935283</td>\n",
       "      <td>17</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>CRR710882</td>\n",
       "      <td>E48</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>12569</td>\n",
       "      <td>12752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>CRR710880</td>\n",
       "      <td>E48</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>12752</td>\n",
       "      <td>12872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>CRR710883</td>\n",
       "      <td>E48</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>12473</td>\n",
       "      <td>12674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>CRR710881</td>\n",
       "      <td>E48</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>12437</td>\n",
       "      <td>12594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>CRR710859</td>\n",
       "      <td>A88</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>44</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>108 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Run accession    sample  correlation  aligned_reads_Crits_Christoph  \\\n",
       "97     CRR710905       F54    -0.004219                              5   \n",
       "16     CRR710804  WWS-2#-3     0.748373                             24   \n",
       "56     CRR710861       A90     0.903233                             13   \n",
       "36     CRR710839       A33     0.910729                             30   \n",
       "55     CRR710860       A90     0.935283                             17   \n",
       "..           ...       ...          ...                            ...   \n",
       "75     CRR710882       E48     0.999999                          12569   \n",
       "73     CRR710880       E48     1.000000                          12752   \n",
       "76     CRR710883       E48     1.000000                          12473   \n",
       "74     CRR710881       E48     1.000000                          12437   \n",
       "54     CRR710859       A88     1.000000                             44   \n",
       "\n",
       "    aligned_reads_current_study  \n",
       "97                          237  \n",
       "16                           23  \n",
       "56                           20  \n",
       "36                           64  \n",
       "55                           27  \n",
       "..                          ...  \n",
       "75                        12752  \n",
       "73                        12872  \n",
       "76                        12674  \n",
       "74                        12594  \n",
       "54                           45  \n",
       "\n",
       "[108 rows x 5 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crits_christoph_vs_current_run_corr = (\n",
    "    crits_christoph_vs_current\n",
    "    .groupby([\"Run accession\", \"sample\"])\n",
    "    [[\"aligned reads Crits-Christoph\", \"aligned reads current study\"]]\n",
    "    .corr(method=\"pearson\")\n",
    "    .reset_index()\n",
    "    .query(\"level_2 == 'aligned reads Crits-Christoph'\")\n",
    "    .rename(columns={\"aligned reads current study\": \"correlation\"})\n",
    "    .drop(columns=[\"level_2\", \"aligned reads Crits-Christoph\"])\n",
    "    .query(\"correlation.notnull()\")\n",
    "    .merge(\n",
    "        crits_christoph_vs_current\n",
    "        .groupby(\"Run accession\", as_index=False)\n",
    "        .aggregate(\n",
    "            aligned_reads_Crits_Christoph=pd.NamedAgg(\"aligned reads Crits-Christoph\", \"sum\"),\n",
    "            aligned_reads_current_study=pd.NamedAgg(\"aligned reads current study\", \"sum\"),\n",
    "        )\n",
    "    )\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_run_corr.sort_values(\"correlation\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "060ef350-67e0-4690-b2b6-65af8245a757",
   "metadata": {},
   "source": [
    "Plot the correlations by run:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c8aed3b3-6113-40ab-83df-eec514e825b1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T22:53:07.547933Z",
     "iopub.status.busy": "2023-04-08T22:53:07.547261Z",
     "iopub.status.idle": "2023-04-08T22:53:07.652517Z",
     "shell.execute_reply": "2023-04-08T22:53:07.651784Z",
     "shell.execute_reply.started": "2023-04-08T22:53:07.547891Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<div id=\"altair-viz-afce4f26ee24493db3dda9997db9c47b\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-afce4f26ee24493db3dda9997db9c47b\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-afce4f26ee24493db3dda9997db9c47b\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm/vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm/vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm/vega-lite@5.6.1?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm/vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"5.6.1\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 300, \"continuousHeight\": 300}}, \"data\": {\"name\": \"data-71182559c1343452cd1048f0968b1e2c\"}, \"mark\": {\"type\": \"circle\", \"opacity\": 1, \"size\": 50}, \"encoding\": {\"tooltip\": [{\"field\": \"Run accession\", \"type\": \"nominal\"}, {\"field\": \"sample\", \"type\": \"nominal\"}, {\"field\": \"correlation\", \"format\": \".4g\", \"type\": \"quantitative\"}, {\"field\": \"aligned_reads_Crits_Christoph\", \"type\": \"quantitative\"}, {\"field\": \"aligned_reads_current_study\", \"type\": \"quantitative\"}], \"x\": {\"field\": \"Run accession\", \"sort\": {\"field\": \"correlation\", \"order\": \"descending\"}, \"type\": \"nominal\"}, \"y\": {\"field\": \"correlation\", \"title\": \"Pearson correlation\", \"type\": \"quantitative\"}}, \"height\": 150, \"params\": [{\"name\": \"param_5\", \"bind\": {\"input\": \"range\", \"max\": 1000, \"min\": 0, \"name\": \"minimum aligned reads in current study\"}, \"value\": 50}, {\"name\": \"param_6\", \"bind\": {\"input\": \"range\", \"max\": 1000, \"min\": 0, \"name\": \"minimum aligned reads in Crits-Christoph et al\"}, \"value\": 50}], \"title\": \"Correlation in read counts aligned to mitochondrial genomes of different species in Crits-Christoph et al and current study\", \"transform\": [{\"filter\": \"(datum['aligned_reads_current_study'] >= param_5)\"}, {\"filter\": \"(datum['aligned_reads_Crits_Christoph'] >= param_6)\"}], \"width\": {\"step\": 11}, \"$schema\": \"https://vega.github.io/schema/vega-lite/v5.6.1.json\", \"datasets\": {\"data-71182559c1343452cd1048f0968b1e2c\": [{\"Run accession\": \"CRR710762\", \"sample\": \"Q61\", \"correlation\": 0.999424607872248, \"aligned_reads_Crits_Christoph\": 21889, \"aligned_reads_current_study\": 12315}, {\"Run accession\": \"CRR710763\", \"sample\": \"Q64\", \"correlation\": 0.9524131770979063, \"aligned_reads_Crits_Christoph\": 1291, \"aligned_reads_current_study\": 715}, {\"Run accession\": \"CRR710764\", \"sample\": \"Q68\", \"correlation\": 0.9954185265399657, \"aligned_reads_Crits_Christoph\": 1364, \"aligned_reads_current_study\": 782}, {\"Run accession\": \"CRR710765\", \"sample\": \"Q69\", \"correlation\": 0.9999006345626711, \"aligned_reads_Crits_Christoph\": 4627, \"aligned_reads_current_study\": 2814}, {\"Run accession\": \"CRR710766\", \"sample\": \"Q70\", \"correlation\": 0.9999787581297611, \"aligned_reads_Crits_Christoph\": 6153, \"aligned_reads_current_study\": 2818}, {\"Run accession\": \"CRR710770\", \"sample\": \"1-29-8\", \"correlation\": 0.9964868403220724, \"aligned_reads_Crits_Christoph\": 67, \"aligned_reads_current_study\": 69}, {\"Run accession\": \"CRR710772\", \"sample\": \"8-25-M1\", \"correlation\": 0.9835861103045237, \"aligned_reads_Crits_Christoph\": 46, \"aligned_reads_current_study\": 47}, {\"Run accession\": \"CRR710775\", \"sample\": \"WS-1-1\", \"correlation\": 0.9988054306421934, \"aligned_reads_Crits_Christoph\": 313, \"aligned_reads_current_study\": 116}, {\"Run accession\": \"CRR710778\", \"sample\": \"WS-2-1\", \"correlation\": 0.9994711379840925, \"aligned_reads_Crits_Christoph\": 628, \"aligned_reads_current_study\": 254}, {\"Run accession\": \"CRR710781\", \"sample\": \"WS-2-3\", \"correlation\": 0.9999368415728592, \"aligned_reads_Crits_Christoph\": 3101, \"aligned_reads_current_study\": 1110}, {\"Run accession\": \"CRR710782\", \"sample\": \"CSSQ-1-3\", \"correlation\": 0.9876308645138362, \"aligned_reads_Crits_Christoph\": 73164, \"aligned_reads_current_study\": 52921}, {\"Run accession\": \"CRR710786\", \"sample\": \"W-8-25-D1\", \"correlation\": 0.9729399569292977, \"aligned_reads_Crits_Christoph\": 604, \"aligned_reads_current_study\": 278}, {\"Run accession\": \"CRR710790\", \"sample\": \"W-8-25-L\", \"correlation\": 0.9743061550671747, \"aligned_reads_Crits_Christoph\": 96, \"aligned_reads_current_study\": 48}, {\"Run accession\": \"CRR710794\", \"sample\": \"EWS-2#-2\", \"correlation\": 0.9974250393887791, \"aligned_reads_Crits_Christoph\": 129, \"aligned_reads_current_study\": 94}, {\"Run accession\": \"CRR710798\", \"sample\": \"WWS-1#\", \"correlation\": 0.9561910985347094, \"aligned_reads_Crits_Christoph\": 91, \"aligned_reads_current_study\": 63}, {\"Run accession\": \"CRR710803\", \"sample\": \"WWS-1#-3\", \"correlation\": 0.9944187919647264, \"aligned_reads_Crits_Christoph\": 148, \"aligned_reads_current_study\": 118}, {\"Run accession\": \"CRR710804\", \"sample\": \"WWS-2#-3\", \"correlation\": 0.7483734042101511, \"aligned_reads_Crits_Christoph\": 24, \"aligned_reads_current_study\": 23}, {\"Run accession\": \"CRR710812\", \"sample\": \"A101\", \"correlation\": 0.9993970769956425, \"aligned_reads_Crits_Christoph\": 1168, \"aligned_reads_current_study\": 1196}, {\"Run accession\": \"CRR710813\", \"sample\": \"A101\", \"correlation\": 0.999708187609852, \"aligned_reads_Crits_Christoph\": 1187, \"aligned_reads_current_study\": 1207}, {\"Run accession\": \"CRR710814\", \"sample\": \"A101\", \"correlation\": 0.9996194776163037, \"aligned_reads_Crits_Christoph\": 1199, \"aligned_reads_current_study\": 1213}, {\"Run accession\": \"CRR710815\", \"sample\": \"A101\", \"correlation\": 0.999118909220885, \"aligned_reads_Crits_Christoph\": 1149, \"aligned_reads_current_study\": 1201}, {\"Run accession\": \"CRR710820\", \"sample\": \"A15\", \"correlation\": 0.9999993070006165, \"aligned_reads_Crits_Christoph\": 162, \"aligned_reads_current_study\": 159}, {\"Run accession\": \"CRR710821\", \"sample\": \"A15\", \"correlation\": 0.9997296577663751, \"aligned_reads_Crits_Christoph\": 96, \"aligned_reads_current_study\": 88}, {\"Run accession\": \"CRR710822\", \"sample\": \"A15\", \"correlation\": 0.9993430624388526, \"aligned_reads_Crits_Christoph\": 129, \"aligned_reads_current_study\": 120}, {\"Run accession\": \"CRR710823\", \"sample\": \"A15\", \"correlation\": 0.9985146484943374, \"aligned_reads_Crits_Christoph\": 123, \"aligned_reads_current_study\": 124}, {\"Run accession\": \"CRR710824\", \"sample\": \"A18\", \"correlation\": 0.9998168293979716, \"aligned_reads_Crits_Christoph\": 620, \"aligned_reads_current_study\": 639}, {\"Run accession\": \"CRR710825\", \"sample\": \"A18\", \"correlation\": 0.9998262357728124, \"aligned_reads_Crits_Christoph\": 743, \"aligned_reads_current_study\": 777}, {\"Run accession\": \"CRR710826\", \"sample\": \"A18\", \"correlation\": 0.9998318829642231, \"aligned_reads_Crits_Christoph\": 649, \"aligned_reads_current_study\": 673}, {\"Run accession\": \"CRR710827\", \"sample\": \"A18\", \"correlation\": 0.9997836100905426, \"aligned_reads_Crits_Christoph\": 594, \"aligned_reads_current_study\": 633}, {\"Run accession\": \"CRR710828\", \"sample\": \"A2\", \"correlation\": 0.998518459586196, \"aligned_reads_Crits_Christoph\": 288, \"aligned_reads_current_study\": 319}, {\"Run accession\": \"CRR710829\", \"sample\": \"A2\", \"correlation\": 0.9988365236415419, \"aligned_reads_Crits_Christoph\": 321, \"aligned_reads_current_study\": 354}, {\"Run accession\": \"CRR710830\", \"sample\": \"A2\", \"correlation\": 0.9982980066422349, \"aligned_reads_Crits_Christoph\": 265, \"aligned_reads_current_study\": 313}, {\"Run accession\": \"CRR710831\", \"sample\": \"A2\", \"correlation\": 0.9974895234658114, \"aligned_reads_Crits_Christoph\": 247, \"aligned_reads_current_study\": 268}, {\"Run accession\": \"CRR710836\", \"sample\": \"A33\", \"correlation\": 0.9589206041284241, \"aligned_reads_Crits_Christoph\": 29, \"aligned_reads_current_study\": 49}, {\"Run accession\": \"CRR710837\", \"sample\": \"A33\", \"correlation\": 0.9684423617636697, \"aligned_reads_Crits_Christoph\": 53, \"aligned_reads_current_study\": 72}, {\"Run accession\": \"CRR710838\", \"sample\": \"A33\", \"correlation\": 0.9766764085846004, \"aligned_reads_Crits_Christoph\": 34, \"aligned_reads_current_study\": 55}, {\"Run accession\": \"CRR710839\", \"sample\": \"A33\", \"correlation\": 0.9107292651682936, \"aligned_reads_Crits_Christoph\": 30, \"aligned_reads_current_study\": 64}, {\"Run accession\": \"CRR710840\", \"sample\": \"A55\", \"correlation\": 0.9997457911631907, \"aligned_reads_Crits_Christoph\": 2376, \"aligned_reads_current_study\": 2505}, {\"Run accession\": \"CRR710841\", \"sample\": \"A55\", \"correlation\": 0.9992290308231411, \"aligned_reads_Crits_Christoph\": 2389, \"aligned_reads_current_study\": 2495}, {\"Run accession\": \"CRR710844\", \"sample\": \"A61\", \"correlation\": 0.99988248569791, \"aligned_reads_Crits_Christoph\": 3486, \"aligned_reads_current_study\": 3505}, {\"Run accession\": \"CRR710845\", \"sample\": \"A61\", \"correlation\": 0.9999359228583691, \"aligned_reads_Crits_Christoph\": 3589, \"aligned_reads_current_study\": 3594}, {\"Run accession\": \"CRR710846\", \"sample\": \"A61\", \"correlation\": 0.9999376890202138, \"aligned_reads_Crits_Christoph\": 3524, \"aligned_reads_current_study\": 3541}, {\"Run accession\": \"CRR710847\", \"sample\": \"A61\", \"correlation\": 0.9998082335541708, \"aligned_reads_Crits_Christoph\": 3337, \"aligned_reads_current_study\": 3348}, {\"Run accession\": \"CRR710848\", \"sample\": \"A63\", \"correlation\": 0.9999367068157158, \"aligned_reads_Crits_Christoph\": 609, \"aligned_reads_current_study\": 629}, {\"Run accession\": \"CRR710849\", \"sample\": \"A63\", \"correlation\": 0.9994853764690164, \"aligned_reads_Crits_Christoph\": 541, \"aligned_reads_current_study\": 589}, {\"Run accession\": \"CRR710850\", \"sample\": \"A63\", \"correlation\": 0.9995143234415115, \"aligned_reads_Crits_Christoph\": 535, \"aligned_reads_current_study\": 601}, {\"Run accession\": \"CRR710851\", \"sample\": \"A63\", \"correlation\": 0.9997914981750939, \"aligned_reads_Crits_Christoph\": 598, \"aligned_reads_current_study\": 648}, {\"Run accession\": \"CRR710852\", \"sample\": \"A87\", \"correlation\": 0.9983496323728203, \"aligned_reads_Crits_Christoph\": 742, \"aligned_reads_current_study\": 768}, {\"Run accession\": \"CRR710853\", \"sample\": \"A87\", \"correlation\": 0.9980417848752063, \"aligned_reads_Crits_Christoph\": 734, \"aligned_reads_current_study\": 772}, {\"Run accession\": \"CRR710854\", \"sample\": \"A87\", \"correlation\": 0.9983941995847404, \"aligned_reads_Crits_Christoph\": 787, \"aligned_reads_current_study\": 825}, {\"Run accession\": \"CRR710855\", \"sample\": \"A87\", \"correlation\": 0.9996098481354891, \"aligned_reads_Crits_Christoph\": 854, \"aligned_reads_current_study\": 870}, {\"Run accession\": \"CRR710856\", \"sample\": \"A88\", \"correlation\": 0.9987794241309935, \"aligned_reads_Crits_Christoph\": 55, \"aligned_reads_current_study\": 60}, {\"Run accession\": \"CRR710857\", \"sample\": \"A88\", \"correlation\": 0.9994796388656999, \"aligned_reads_Crits_Christoph\": 62, \"aligned_reads_current_study\": 64}, {\"Run accession\": \"CRR710858\", \"sample\": \"A88\", \"correlation\": 0.9760499462162847, \"aligned_reads_Crits_Christoph\": 36, \"aligned_reads_current_study\": 47}, {\"Run accession\": \"CRR710859\", \"sample\": \"A88\", \"correlation\": 0.9999999999999999, \"aligned_reads_Crits_Christoph\": 44, \"aligned_reads_current_study\": 45}, {\"Run accession\": \"CRR710860\", \"sample\": \"A90\", \"correlation\": 0.9352826756918655, \"aligned_reads_Crits_Christoph\": 17, \"aligned_reads_current_study\": 27}, {\"Run accession\": \"CRR710861\", \"sample\": \"A90\", \"correlation\": 0.9032329495627082, \"aligned_reads_Crits_Christoph\": 13, \"aligned_reads_current_study\": 20}, {\"Run accession\": \"CRR710864\", \"sample\": \"A96\", \"correlation\": 0.9857257854390061, \"aligned_reads_Crits_Christoph\": 52, \"aligned_reads_current_study\": 61}, {\"Run accession\": \"CRR710865\", \"sample\": \"A96\", \"correlation\": 0.9924840775704771, \"aligned_reads_Crits_Christoph\": 63, \"aligned_reads_current_study\": 72}, {\"Run accession\": \"CRR710866\", \"sample\": \"A96\", \"correlation\": 0.9882594547790707, \"aligned_reads_Crits_Christoph\": 50, \"aligned_reads_current_study\": 65}, {\"Run accession\": \"CRR710867\", \"sample\": \"A96\", \"correlation\": 0.9800754467602446, \"aligned_reads_Crits_Christoph\": 83, \"aligned_reads_current_study\": 98}, {\"Run accession\": \"CRR710868\", \"sample\": \"B17\", \"correlation\": 0.999167733426343, \"aligned_reads_Crits_Christoph\": 770, \"aligned_reads_current_study\": 793}, {\"Run accession\": \"CRR710869\", \"sample\": \"B17\", \"correlation\": 0.9997698692714668, \"aligned_reads_Crits_Christoph\": 785, \"aligned_reads_current_study\": 819}, {\"Run accession\": \"CRR710870\", \"sample\": \"B17\", \"correlation\": 0.9996666576204437, \"aligned_reads_Crits_Christoph\": 773, \"aligned_reads_current_study\": 809}, {\"Run accession\": \"CRR710871\", \"sample\": \"B17\", \"correlation\": 0.9993707628564273, \"aligned_reads_Crits_Christoph\": 765, \"aligned_reads_current_study\": 805}, {\"Run accession\": \"CRR710872\", \"sample\": \"B5\", \"correlation\": 0.9998888427225808, \"aligned_reads_Crits_Christoph\": 4747, \"aligned_reads_current_study\": 4784}, {\"Run accession\": \"CRR710873\", \"sample\": \"B5\", \"correlation\": 0.9998136514462002, \"aligned_reads_Crits_Christoph\": 4628, \"aligned_reads_current_study\": 4670}, {\"Run accession\": \"CRR710874\", \"sample\": \"B5\", \"correlation\": 0.9997711516012906, \"aligned_reads_Crits_Christoph\": 4932, \"aligned_reads_current_study\": 4971}, {\"Run accession\": \"CRR710875\", \"sample\": \"B5\", \"correlation\": 0.9997975029819488, \"aligned_reads_Crits_Christoph\": 5101, \"aligned_reads_current_study\": 5145}, {\"Run accession\": \"CRR710876\", \"sample\": \"D32\", \"correlation\": 0.9998779054830333, \"aligned_reads_Crits_Christoph\": 2458, \"aligned_reads_current_study\": 2509}, {\"Run accession\": \"CRR710877\", \"sample\": \"D32\", \"correlation\": 0.9999287297906232, \"aligned_reads_Crits_Christoph\": 2367, \"aligned_reads_current_study\": 2416}, {\"Run accession\": \"CRR710878\", \"sample\": \"D32\", \"correlation\": 0.999978581772743, \"aligned_reads_Crits_Christoph\": 2278, \"aligned_reads_current_study\": 2307}, {\"Run accession\": \"CRR710879\", \"sample\": \"D32\", \"correlation\": 0.9999565726591887, \"aligned_reads_Crits_Christoph\": 2216, \"aligned_reads_current_study\": 2244}, {\"Run accession\": \"CRR710880\", \"sample\": \"E48\", \"correlation\": 0.9999995685234461, \"aligned_reads_Crits_Christoph\": 12752, \"aligned_reads_current_study\": 12872}, {\"Run accession\": \"CRR710881\", \"sample\": \"E48\", \"correlation\": 0.9999998772181435, \"aligned_reads_Crits_Christoph\": 12437, \"aligned_reads_current_study\": 12594}, {\"Run accession\": \"CRR710882\", \"sample\": \"E48\", \"correlation\": 0.999999348058754, \"aligned_reads_Crits_Christoph\": 12569, \"aligned_reads_current_study\": 12752}, {\"Run accession\": \"CRR710883\", \"sample\": \"E48\", \"correlation\": 0.9999997075902821, \"aligned_reads_Crits_Christoph\": 12473, \"aligned_reads_current_study\": 12674}, {\"Run accession\": \"CRR710884\", \"sample\": \"E61\", \"correlation\": 0.979147020107963, \"aligned_reads_Crits_Christoph\": 37, \"aligned_reads_current_study\": 47}, {\"Run accession\": \"CRR710885\", \"sample\": \"E61\", \"correlation\": 0.9918809804773326, \"aligned_reads_Crits_Christoph\": 60, \"aligned_reads_current_study\": 71}, {\"Run accession\": \"CRR710886\", \"sample\": \"E61\", \"correlation\": 0.9908749527276367, \"aligned_reads_Crits_Christoph\": 77, \"aligned_reads_current_study\": 84}, {\"Run accession\": \"CRR710887\", \"sample\": \"E61\", \"correlation\": 0.9742919871500083, \"aligned_reads_Crits_Christoph\": 59, \"aligned_reads_current_study\": 67}, {\"Run accession\": \"CRR710888\", \"sample\": \"E7\", \"correlation\": 0.9994242088716081, \"aligned_reads_Crits_Christoph\": 1099, \"aligned_reads_current_study\": 1131}, {\"Run accession\": \"CRR710889\", \"sample\": \"E7\", \"correlation\": 0.9988839608504738, \"aligned_reads_Crits_Christoph\": 1125, \"aligned_reads_current_study\": 1193}, {\"Run accession\": \"CRR710890\", \"sample\": \"E7\", \"correlation\": 0.9997358567884033, \"aligned_reads_Crits_Christoph\": 1068, \"aligned_reads_current_study\": 1101}, {\"Run accession\": \"CRR710891\", \"sample\": \"E7\", \"correlation\": 0.9995141491489651, \"aligned_reads_Crits_Christoph\": 1028, \"aligned_reads_current_study\": 1067}, {\"Run accession\": \"CRR710892\", \"sample\": \"F100\", \"correlation\": 0.9651273801207719, \"aligned_reads_Crits_Christoph\": 38, \"aligned_reads_current_study\": 49}, {\"Run accession\": \"CRR710893\", \"sample\": \"F100\", \"correlation\": 0.9581263150246111, \"aligned_reads_Crits_Christoph\": 31, \"aligned_reads_current_study\": 42}, {\"Run accession\": \"CRR710894\", \"sample\": \"F100\", \"correlation\": 0.9836483522608589, \"aligned_reads_Crits_Christoph\": 30, \"aligned_reads_current_study\": 39}, {\"Run accession\": \"CRR710895\", \"sample\": \"F100\", \"correlation\": 0.9813591194834214, \"aligned_reads_Crits_Christoph\": 31, \"aligned_reads_current_study\": 40}, {\"Run accession\": \"CRR710896\", \"sample\": \"F13\", \"correlation\": 0.9990166955652234, \"aligned_reads_Crits_Christoph\": 103, \"aligned_reads_current_study\": 103}, {\"Run accession\": \"CRR710897\", \"sample\": \"F13\", \"correlation\": 0.9986477831256982, \"aligned_reads_Crits_Christoph\": 67, \"aligned_reads_current_study\": 67}, {\"Run accession\": \"CRR710898\", \"sample\": \"F13\", \"correlation\": 0.9981482362877387, \"aligned_reads_Crits_Christoph\": 89, \"aligned_reads_current_study\": 94}, {\"Run accession\": \"CRR710899\", \"sample\": \"F13\", \"correlation\": 0.9998078574634005, \"aligned_reads_Crits_Christoph\": 105, \"aligned_reads_current_study\": 104}, {\"Run accession\": \"CRR710900\", \"sample\": \"F46\", \"correlation\": 0.9999612515758858, \"aligned_reads_Crits_Christoph\": 1138, \"aligned_reads_current_study\": 1043}, {\"Run accession\": \"CRR710901\", \"sample\": \"F46\", \"correlation\": 0.9999794593570166, \"aligned_reads_Crits_Christoph\": 1130, \"aligned_reads_current_study\": 1039}, {\"Run accession\": \"CRR710902\", \"sample\": \"F46\", \"correlation\": 0.99999020518377, \"aligned_reads_Crits_Christoph\": 1175, \"aligned_reads_current_study\": 1079}, {\"Run accession\": \"CRR710903\", \"sample\": \"F46\", \"correlation\": 0.9999790018108304, \"aligned_reads_Crits_Christoph\": 1172, \"aligned_reads_current_study\": 1076}, {\"Run accession\": \"CRR710905\", \"sample\": \"F54\", \"correlation\": -0.004219010743820937, \"aligned_reads_Crits_Christoph\": 5, \"aligned_reads_current_study\": 237}, {\"Run accession\": \"CRR710907\", \"sample\": \"F54\", \"correlation\": 0.9995618216625154, \"aligned_reads_Crits_Christoph\": 144, \"aligned_reads_current_study\": 211}, {\"Run accession\": \"CRR710908\", \"sample\": \"F98\", \"correlation\": 0.9999483278281182, \"aligned_reads_Crits_Christoph\": 423, \"aligned_reads_current_study\": 439}, {\"Run accession\": \"CRR710909\", \"sample\": \"F98\", \"correlation\": 0.9998487691695912, \"aligned_reads_Crits_Christoph\": 393, \"aligned_reads_current_study\": 412}, {\"Run accession\": \"CRR710910\", \"sample\": \"F98\", \"correlation\": 0.9997949510596071, \"aligned_reads_Crits_Christoph\": 432, \"aligned_reads_current_study\": 442}, {\"Run accession\": \"CRR710911\", \"sample\": \"F98\", \"correlation\": 0.9998575371589218, \"aligned_reads_Crits_Christoph\": 343, \"aligned_reads_current_study\": 358}, {\"Run accession\": \"CRR711156\", \"sample\": \"W-8-25-D2\", \"correlation\": 0.9740265811578742, \"aligned_reads_Crits_Christoph\": 6493, \"aligned_reads_current_study\": 2985}, {\"Run accession\": \"CRR711157\", \"sample\": \"W-8-25-L2\", \"correlation\": 0.9963362261024077, \"aligned_reads_Crits_Christoph\": 2979, \"aligned_reads_current_study\": 1222}, {\"Run accession\": \"CRR711158\", \"sample\": \"EWS-3#-2\", \"correlation\": 0.9924850198997489, \"aligned_reads_Crits_Christoph\": 69, \"aligned_reads_current_study\": 48}, {\"Run accession\": \"CRR711159\", \"sample\": \"WWS-1#-2\", \"correlation\": 0.9615240401533396, \"aligned_reads_Crits_Christoph\": 81, \"aligned_reads_current_study\": 70}, {\"Run accession\": \"CRR711161\", \"sample\": \"wws-1#-0\", \"correlation\": 0.9937976081201559, \"aligned_reads_Crits_Christoph\": 210, \"aligned_reads_current_study\": 164}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aligned_reads_current_slider = alt.param(\n",
    "    bind=alt.binding_range(\n",
    "        name=\"minimum aligned reads in current study\",\n",
    "        min=0,\n",
    "        max=1000,\n",
    "    ),\n",
    "    value=50,\n",
    ")\n",
    "\n",
    "aligned_reads_crits_christoph_slider = alt.param(\n",
    "    bind=alt.binding_range(\n",
    "        name=\"minimum aligned reads in Crits-Christoph et al\",\n",
    "        min=0,\n",
    "        max=1000,\n",
    "    ),\n",
    "    value=50,\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_run_corr_chart = (\n",
    "    alt.Chart(crits_christoph_vs_current_run_corr)\n",
    "    .encode(\n",
    "        x=alt.X(\n",
    "            \"Run accession\",\n",
    "            sort=alt.SortField(\"correlation\", order=\"descending\"),\n",
    "        ),\n",
    "        y=alt.Y(\"correlation\", title=\"Pearson correlation\"),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".4g\")\n",
    "            if crits_christoph_vs_current_run_corr[c].dtype == float\n",
    "            else c\n",
    "            for c in crits_christoph_vs_current_run_corr.columns.tolist()\n",
    "        ],\n",
    "    )\n",
    "    .mark_circle(size=50, opacity=1)\n",
    "    .properties(\n",
    "        width=alt.Step(11),\n",
    "        height=150,\n",
    "        title=(\n",
    "            \"Correlation in read counts aligned to mitochondrial genomes of different \"\n",
    "            \"species in Crits-Christoph et al and current study\"\n",
    "        )\n",
    "    )\n",
    "    .add_params(aligned_reads_current_slider, aligned_reads_crits_christoph_slider)\n",
    "    .transform_filter(\n",
    "        alt.datum[\"aligned_reads_current_study\"] >= aligned_reads_current_slider\n",
    "    )\n",
    "    .transform_filter(\n",
    "        alt.datum[\"aligned_reads_Crits_Christoph\"] >= aligned_reads_crits_christoph_slider\n",
    "    )\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_run_corr_chart"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "074cc324-6e0f-4822-b38e-e7e87adfe11f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
