{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bdf42685-7bac-45b8-b061-0851c35f3a59",
   "metadata": {},
   "source": [
    "# Make plots"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30fe09f9-1157-4594-a37d-c60db2190d6c",
   "metadata": {},
   "source": [
    "Import Python modules:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "814bf235-7c93-47f7-adbc-ba242b931670",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T15:00:10.980009Z",
     "iopub.status.busy": "2023-04-08T15:00:10.979578Z",
     "iopub.status.idle": "2023-04-08T15:00:12.040445Z",
     "shell.execute_reply": "2023-04-08T15:00:12.039335Z",
     "shell.execute_reply.started": "2023-04-08T15:00:10.979982Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "import itertools\n",
    "import os\n",
    "\n",
    "import altair as alt\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "_ = alt.data_transformers.disable_max_rows()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "897e563c-8a14-48e7-958c-4c69e57bfd69",
   "metadata": {},
   "source": [
    "Get data and variable from `snakemake`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e4be094-39bd-4972-b49d-76e3b400d3ab",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "sars2_aligned_by_run = pd.read_csv(snakemake.input.sars2_aligned_by_run)\n",
    "\n",
    "sars2_aligned_by_metagenomic_sample = pd.read_csv(snakemake.input.sars2_aligned_by_metagenomic_sample)\n",
    "\n",
    "mito_composition_by_metagenomic_run = pd.read_csv(snakemake.input.mito_composition_by_metagenomic_run)\n",
    "\n",
    "mito_composition_by_metagenomic_sample = pd.read_csv(snakemake.input.mito_composition_by_metagenomic_sample)\n",
    "\n",
    "crits_christoph_read_counts = pd.read_csv(snakemake.input.crits_christoph_read_counts)\n",
    "\n",
    "ngdc_to_crits_christoph = pd.read_csv(snakemake.input.ngdc_to_crits_christoph)\n",
    "\n",
    "plotsdir = snakemake.output.plotsdir"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f887eb07-373c-42b4-9717-5b7702b84f0c",
   "metadata": {},
   "source": [
    "## Compare mitochondrial DNA composition to Crits-Christoph et al\n",
    "First get Crits-Christoph read counts in tidy format, assigning NGDC run accessions and summing counts for runs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b2a9fb08-f0fd-4539-9e7a-59f1ae2a9c44",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T15:00:12.200157Z",
     "iopub.status.busy": "2023-04-08T15:00:12.199945Z",
     "iopub.status.idle": "2023-04-08T15:00:12.237752Z",
     "shell.execute_reply": "2023-04-08T15:00:12.237135Z",
     "shell.execute_reply.started": "2023-04-08T15:00:12.200138Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Run accession</th>\n",
       "      <th>species</th>\n",
       "      <th>aligned reads Crits-Christoph</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Arctonyx collaris</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Canis lupus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Capra aegagrus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1-29-8</td>\n",
       "      <td>CRR710770</td>\n",
       "      <td>Erinaceus amurensis</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2923</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Rattus tiomanicus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2924</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Rhizomys pruinosus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2925</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Sus scrofa</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2926</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2927</th>\n",
       "      <td>wws-1#-0</td>\n",
       "      <td>CRR711161</td>\n",
       "      <td>Vulpes zerda</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2928 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample Run accession                   species  \\\n",
       "0       1-29-8     CRR710770         Arctonyx collaris   \n",
       "1       1-29-8     CRR710770                Bos taurus   \n",
       "2       1-29-8     CRR710770               Canis lupus   \n",
       "3       1-29-8     CRR710770            Capra aegagrus   \n",
       "4       1-29-8     CRR710770       Erinaceus amurensis   \n",
       "...        ...           ...                       ...   \n",
       "2923  wws-1#-0     CRR711161         Rattus tiomanicus   \n",
       "2924  wws-1#-0     CRR711161        Rhizomys pruinosus   \n",
       "2925  wws-1#-0     CRR711161                Sus scrofa   \n",
       "2926  wws-1#-0     CRR711161  Urocyon cinereoargenteus   \n",
       "2927  wws-1#-0     CRR711161              Vulpes zerda   \n",
       "\n",
       "      aligned reads Crits-Christoph  \n",
       "0                                 0  \n",
       "1                                 0  \n",
       "2                                 0  \n",
       "3                                 0  \n",
       "4                                 0  \n",
       "...                             ...  \n",
       "2923                              0  \n",
       "2924                              0  \n",
       "2925                             32  \n",
       "2926                              0  \n",
       "2927                              0  \n",
       "\n",
       "[2928 rows x 4 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crits_christoph_read_counts_tidy = (\n",
    "    crits_christoph_read_counts\n",
    "    .drop(columns=[\"Location\", \"Sample_name\"])\n",
    "    .merge(\n",
    "        ngdc_to_crits_christoph\n",
    "        .assign(Filename=lambda x: x[\"fastq Crits-Christoph\"].map(lambda s: s.split(\".\")[0]))\n",
    "        .rename(columns={\"Run accession NGDC\": \"Run accession\"})\n",
    "        [[\"Filename\", \"Run accession\"]]\n",
    "        .drop_duplicates(),\n",
    "        validate=\"one_to_one\",\n",
    "    )\n",
    "    .merge(\n",
    "        sars2_aligned_by_run[[\"Run accession\", \"sample\"]].drop_duplicates(),\n",
    "        validate=\"many_to_one\",\n",
    "        how=\"left\",\n",
    "    )\n",
    "    .melt(\n",
    "        id_vars=[\"sample\", \"Run accession\", \"Filename\"],\n",
    "        var_name=\"species\",\n",
    "        value_name=\"aligned reads Crits-Christoph\",\n",
    "    )\n",
    "    .groupby([\"sample\", \"Run accession\", \"species\"], as_index=False)\n",
    "    .aggregate({\"aligned reads Crits-Christoph\": \"sum\"})\n",
    ")\n",
    "\n",
    "crits_christoph_read_counts_tidy"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81141895-958a-4cc7-b1d0-02e33f8c0064",
   "metadata": {},
   "source": [
    "Now merge current read counts with those from Crits-Christoph for all species:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "116340c5-ebe7-4da3-8472-d1acf24694ef",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T15:00:12.241513Z",
     "iopub.status.busy": "2023-04-08T15:00:12.241277Z",
     "iopub.status.idle": "2023-04-08T15:00:12.266798Z",
     "shell.execute_reply": "2023-04-08T15:00:12.265777Z",
     "shell.execute_reply.started": "2023-04-08T15:00:12.241492Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_species = crits_christoph_read_counts_tidy[\"species\"].unique().tolist()\n",
    "\n",
    "# get current study data in form to compare, harmozing species names\n",
    "mito_composition_to_compare = (\n",
    "    mito_composition_by_metagenomic_run\n",
    "    [[\"Run accession\", \"species\", \"common_name\", \"alignment_count_type\", \"aligned_reads\"]]\n",
    "    .rename(columns={\"aligned_reads\": \"aligned reads current study\"})\n",
    ")\n",
    "\n",
    "# expand mito composition by run to include all Crits-Christoph species\n",
    "assert set(crits_christoph_species).issubset(mito_composition_to_compare[\"species\"])\n",
    "\n",
    "\n",
    "crits_christoph_vs_current = (\n",
    "    crits_christoph_read_counts_tidy\n",
    "    .merge(\n",
    "        mito_composition_to_compare,\n",
    "        validate=\"one_to_many\",\n",
    "        on=[\"Run accession\", \"species\"],\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f2893ad-9059-4a99-b026-971364fc4177",
   "metadata": {},
   "source": [
    "Now get the correlations by species and plot them:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fad10471-0e9e-47e4-b488-a89c2dc09d0a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T15:00:12.267974Z",
     "iopub.status.busy": "2023-04-08T15:00:12.267659Z",
     "iopub.status.idle": "2023-04-08T15:00:12.308066Z",
     "shell.execute_reply": "2023-04-08T15:00:12.307194Z",
     "shell.execute_reply.started": "2023-04-08T15:00:12.267955Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>species</th>\n",
       "      <th>common_name</th>\n",
       "      <th>alignment_count_type</th>\n",
       "      <th>Pearson correlation</th>\n",
       "      <th>aligned_reads_Crits_Christoph</th>\n",
       "      <th>aligned_reads_current_study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Arctonyx collaris</td>\n",
       "      <td>hog badger</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0.924400</td>\n",
       "      <td>188</td>\n",
       "      <td>330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Arctonyx collaris</td>\n",
       "      <td>hog badger</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0.955924</td>\n",
       "      <td>188</td>\n",
       "      <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>cow</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0.999974</td>\n",
       "      <td>61278</td>\n",
       "      <td>32632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>cow</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0.999613</td>\n",
       "      <td>61278</td>\n",
       "      <td>22702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Canis lupus</td>\n",
       "      <td>dog</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0.998951</td>\n",
       "      <td>7941</td>\n",
       "      <td>6386</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             species common_name       alignment_count_type  \\\n",
       "0  Arctonyx collaris  hog badger                  n_primary   \n",
       "1  Arctonyx collaris  hog badger  n_primary_and_unique_best   \n",
       "2         Bos taurus         cow                  n_primary   \n",
       "3         Bos taurus         cow  n_primary_and_unique_best   \n",
       "4        Canis lupus         dog                  n_primary   \n",
       "\n",
       "   Pearson correlation  aligned_reads_Crits_Christoph  \\\n",
       "0             0.924400                            188   \n",
       "1             0.955924                            188   \n",
       "2             0.999974                          61278   \n",
       "3             0.999613                          61278   \n",
       "4             0.998951                           7941   \n",
       "\n",
       "   aligned_reads_current_study  \n",
       "0                          330  \n",
       "1                          221  \n",
       "2                        32632  \n",
       "3                        22702  \n",
       "4                         6386  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crits_christoph_vs_current_species_corr = (\n",
    "    crits_christoph_vs_current\n",
    "    .groupby([\"species\", \"common_name\", \"alignment_count_type\"])\n",
    "    [[\"aligned reads Crits-Christoph\", \"aligned reads current study\"]]\n",
    "    .corr(method=\"pearson\")\n",
    "    .reset_index()\n",
    "    .query(\"level_3 == 'aligned reads Crits-Christoph'\")\n",
    "    .rename(columns={\"aligned reads current study\": \"Pearson correlation\"})\n",
    "    .drop(columns=[\"level_3\", \"aligned reads Crits-Christoph\"])\n",
    "    .query(\"`Pearson correlation`.notnull()\")\n",
    "    .merge(\n",
    "        crits_christoph_vs_current\n",
    "        .groupby([\"species\", \"alignment_count_type\"], as_index=False)\n",
    "        .aggregate(\n",
    "            aligned_reads_Crits_Christoph=pd.NamedAgg(\"aligned reads Crits-Christoph\", \"sum\"),\n",
    "            aligned_reads_current_study=pd.NamedAgg(\"aligned reads current study\", \"sum\"),\n",
    "        )\n",
    "    )\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_species_corr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "14913bb6-f68f-4c02-9fe9-fb824f2a28ec",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T15:00:12.309307Z",
     "iopub.status.busy": "2023-04-08T15:00:12.308959Z",
     "iopub.status.idle": "2023-04-08T15:00:12.428777Z",
     "shell.execute_reply": "2023-04-08T15:00:12.427879Z",
     "shell.execute_reply.started": "2023-04-08T15:00:12.309286Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<div id=\"altair-viz-e07f8ee5402149a5bd1bb2e3c2217ca6\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-e07f8ee5402149a5bd1bb2e3c2217ca6\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-e07f8ee5402149a5bd1bb2e3c2217ca6\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm/vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm/vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm/vega-lite@5.6.1?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm/vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"5.6.1\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 300, \"continuousHeight\": 300}}, \"data\": {\"name\": \"data-7a2dd60b9beddf8f77ccee6a528d98aa\"}, \"mark\": {\"type\": \"circle\", \"opacity\": 1, \"size\": 50}, \"encoding\": {\"tooltip\": [{\"field\": \"species\", \"type\": \"nominal\"}, {\"field\": \"common_name\", \"type\": \"nominal\"}, {\"field\": \"alignment_count_type\", \"type\": \"nominal\"}, {\"field\": \"Pearson correlation\", \"format\": \".3g\", \"type\": \"quantitative\"}, {\"field\": \"aligned_reads_Crits_Christoph\", \"type\": \"quantitative\"}, {\"field\": \"aligned_reads_current_study\", \"type\": \"quantitative\"}], \"x\": {\"field\": \"Pearson correlation\", \"type\": \"quantitative\"}, \"y\": {\"field\": \"common_name\", \"sort\": {\"field\": \"Pearson correlation\", \"order\": \"descending\"}, \"title\": null, \"type\": \"nominal\"}}, \"height\": {\"step\": 13}, \"params\": [{\"name\": \"param_1\", \"select\": {\"type\": \"point\", \"fields\": [\"alignment_count_type\"]}, \"bind\": {\"input\": \"select\", \"options\": [\"n_primary\", \"n_primary_and_unique_best\"], \"name\": \"alignment count type\"}, \"value\": \"n_primary\"}], \"title\": {\"text\": [\"correlation across runs of reads aligning to each\", \"mitochondrial genome in this study vs Crits-Christoph et al\"], \"align\": \"left\", \"anchor\": \"start\", \"fontSize\": 12}, \"transform\": [{\"filter\": {\"param\": \"param_1\"}}], \"width\": 185, \"$schema\": \"https://vega.github.io/schema/vega-lite/v5.6.1.json\", \"datasets\": {\"data-7a2dd60b9beddf8f77ccee6a528d98aa\": [{\"species\": \"Arctonyx collaris\", \"common_name\": \"hog badger\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9244000242867562, \"aligned_reads_Crits_Christoph\": 188, \"aligned_reads_current_study\": 330}, {\"species\": \"Arctonyx collaris\", \"common_name\": \"hog badger\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9559244712657082, \"aligned_reads_Crits_Christoph\": 188, \"aligned_reads_current_study\": 221}, {\"species\": \"Bos taurus\", \"common_name\": \"cow\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9999741751637752, \"aligned_reads_Crits_Christoph\": 61278, \"aligned_reads_current_study\": 32632}, {\"species\": \"Bos taurus\", \"common_name\": \"cow\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9996131993905085, \"aligned_reads_Crits_Christoph\": 61278, \"aligned_reads_current_study\": 22702}, {\"species\": \"Canis lupus\", \"common_name\": \"dog\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9989513451557378, \"aligned_reads_Crits_Christoph\": 7941, \"aligned_reads_current_study\": 6386}, {\"species\": \"Canis lupus\", \"common_name\": \"dog\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9975974877845469, \"aligned_reads_Crits_Christoph\": 7941, \"aligned_reads_current_study\": 5763}, {\"species\": \"Capra aegagrus\", \"common_name\": \"goat\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9816095353951505, \"aligned_reads_Crits_Christoph\": 2710, \"aligned_reads_current_study\": 1425}, {\"species\": \"Capra aegagrus\", \"common_name\": \"goat\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9669124764007945, \"aligned_reads_Crits_Christoph\": 2710, \"aligned_reads_current_study\": 966}, {\"species\": \"Erinaceus amurensis\", \"common_name\": \"Amur hedgehog\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9999305631860063, \"aligned_reads_Crits_Christoph\": 7035, \"aligned_reads_current_study\": 4287}, {\"species\": \"Erinaceus amurensis\", \"common_name\": \"Amur hedgehog\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9998038358522283, \"aligned_reads_Crits_Christoph\": 7035, \"aligned_reads_current_study\": 3923}, {\"species\": \"Erinaceus europaeus\", \"common_name\": \"European hedgehog\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9984493059910062, \"aligned_reads_Crits_Christoph\": 1236, \"aligned_reads_current_study\": 1009}, {\"species\": \"Erinaceus europaeus\", \"common_name\": \"European hedgehog\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9973810148150326, \"aligned_reads_Crits_Christoph\": 1236, \"aligned_reads_current_study\": 623}, {\"species\": \"Felis margarita\", \"common_name\": \"sand cat\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9679660655789116, \"aligned_reads_Crits_Christoph\": 914, \"aligned_reads_current_study\": 208}, {\"species\": \"Felis margarita\", \"common_name\": \"sand cat\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9811663630271902, \"aligned_reads_Crits_Christoph\": 914, \"aligned_reads_current_study\": 135}, {\"species\": \"Hemiechinus auritus\", \"common_name\": \"long-eared hedgehog\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9907592890753047, \"aligned_reads_Crits_Christoph\": 151, \"aligned_reads_current_study\": 175}, {\"species\": \"Hemiechinus auritus\", \"common_name\": \"long-eared hedgehog\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9905329129452352, \"aligned_reads_Crits_Christoph\": 151, \"aligned_reads_current_study\": 122}, {\"species\": \"Homo sapiens\", \"common_name\": \"human\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9991895703368543, \"aligned_reads_Crits_Christoph\": 91786, \"aligned_reads_current_study\": 84703}, {\"species\": \"Homo sapiens\", \"common_name\": \"human\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9989978340242176, \"aligned_reads_Crits_Christoph\": 91786, \"aligned_reads_current_study\": 75729}, {\"species\": \"Hystrix brachyura\", \"common_name\": \"Malayan porcupine\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9999905654844058, \"aligned_reads_Crits_Christoph\": 6022, \"aligned_reads_current_study\": 3780}, {\"species\": \"Hystrix brachyura\", \"common_name\": \"Malayan porcupine\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9999916170744853, \"aligned_reads_Crits_Christoph\": 6022, \"aligned_reads_current_study\": 3700}, {\"species\": \"Marmota himalayana\", \"common_name\": \"Himalayan marmot\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9998469251359242, \"aligned_reads_Crits_Christoph\": 796, \"aligned_reads_current_study\": 638}, {\"species\": \"Marmota himalayana\", \"common_name\": \"Himalayan marmot\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9999657952055414, \"aligned_reads_Crits_Christoph\": 796, \"aligned_reads_current_study\": 602}, {\"species\": \"Muntiacus putaoensis\", \"common_name\": \"leaf deer\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.899519189114785, \"aligned_reads_Crits_Christoph\": 89, \"aligned_reads_current_study\": 110}, {\"species\": \"Muntiacus putaoensis\", \"common_name\": \"leaf deer\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9547809100194581, \"aligned_reads_Crits_Christoph\": 89, \"aligned_reads_current_study\": 75}, {\"species\": \"Mustela sibirica\", \"common_name\": \"Siberian weasel\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.917549682172403, \"aligned_reads_Crits_Christoph\": 216, \"aligned_reads_current_study\": 202}, {\"species\": \"Mustela sibirica\", \"common_name\": \"Siberian weasel\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.8553648166074136, \"aligned_reads_Crits_Christoph\": 216, \"aligned_reads_current_study\": 129}, {\"species\": \"Nyctereutes procyonoides\", \"common_name\": \"raccoon dog\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9999996414657636, \"aligned_reads_Crits_Christoph\": 17086, \"aligned_reads_current_study\": 13732}, {\"species\": \"Nyctereutes procyonoides\", \"common_name\": \"raccoon dog\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.999999462409767, \"aligned_reads_Crits_Christoph\": 17086, \"aligned_reads_current_study\": 13162}, {\"species\": \"Oryctolagus cuniculus\", \"common_name\": \"rabbit\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9971361279770634, \"aligned_reads_Crits_Christoph\": 1018, \"aligned_reads_current_study\": 961}, {\"species\": \"Oryctolagus cuniculus\", \"common_name\": \"rabbit\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9965272401465893, \"aligned_reads_Crits_Christoph\": 1018, \"aligned_reads_current_study\": 859}, {\"species\": \"Otocyon megalotis\", \"common_name\": \"bat-eared fox\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9993045779042994, \"aligned_reads_Crits_Christoph\": 255, \"aligned_reads_current_study\": 211}, {\"species\": \"Otocyon megalotis\", \"common_name\": \"bat-eared fox\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9912241393850777, \"aligned_reads_Crits_Christoph\": 255, \"aligned_reads_current_study\": 43}, {\"species\": \"Ovis orientalis\", \"common_name\": \"sheep\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9913638816189271, \"aligned_reads_Crits_Christoph\": 28263, \"aligned_reads_current_study\": 16183}, {\"species\": \"Ovis orientalis\", \"common_name\": \"sheep\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9682242134306027, \"aligned_reads_Crits_Christoph\": 28263, \"aligned_reads_current_study\": 11245}, {\"species\": \"Paguma larvata\", \"common_name\": \"masked palm civet\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.6015277263214948, \"aligned_reads_Crits_Christoph\": 536, \"aligned_reads_current_study\": 3223}, {\"species\": \"Paguma larvata\", \"common_name\": \"masked palm civet\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.6586732526464866, \"aligned_reads_Crits_Christoph\": 536, \"aligned_reads_current_study\": 2483}, {\"species\": \"Rattus norvegicus\", \"common_name\": \"brown rat\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9989227740399469, \"aligned_reads_Crits_Christoph\": 10141, \"aligned_reads_current_study\": 1361}, {\"species\": \"Rattus norvegicus\", \"common_name\": \"brown rat\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.998855223179743, \"aligned_reads_Crits_Christoph\": 10141, \"aligned_reads_current_study\": 1281}, {\"species\": \"Rattus tiomanicus\", \"common_name\": \"Mayalan field rat\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.8909760047373336, \"aligned_reads_Crits_Christoph\": 394, \"aligned_reads_current_study\": 320}, {\"species\": \"Rattus tiomanicus\", \"common_name\": \"Mayalan field rat\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.8805028031566878, \"aligned_reads_Crits_Christoph\": 394, \"aligned_reads_current_study\": 258}, {\"species\": \"Rhizomys pruinosus\", \"common_name\": \"hoary bamboo rat\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9984446707141166, \"aligned_reads_Crits_Christoph\": 1263, \"aligned_reads_current_study\": 944}, {\"species\": \"Rhizomys pruinosus\", \"common_name\": \"hoary bamboo rat\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.998123971118571, \"aligned_reads_Crits_Christoph\": 1263, \"aligned_reads_current_study\": 881}, {\"species\": \"Sus scrofa\", \"common_name\": \"pig\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9987803055250261, \"aligned_reads_Crits_Christoph\": 9821, \"aligned_reads_current_study\": 5348}, {\"species\": \"Sus scrofa\", \"common_name\": \"pig\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.985240548783991, \"aligned_reads_Crits_Christoph\": 9821, \"aligned_reads_current_study\": 3597}, {\"species\": \"Urocyon cinereoargenteus\", \"common_name\": \"gray fox\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.3361016708480335, \"aligned_reads_Crits_Christoph\": 552, \"aligned_reads_current_study\": 41302}, {\"species\": \"Urocyon cinereoargenteus\", \"common_name\": \"gray fox\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.2022872469711302, \"aligned_reads_Crits_Christoph\": 552, \"aligned_reads_current_study\": 15731}, {\"species\": \"Vulpes zerda\", \"common_name\": \"fennec fox\", \"alignment_count_type\": \"n_primary\", \"Pearson correlation\": 0.9981498425275961, \"aligned_reads_Crits_Christoph\": 165, \"aligned_reads_current_study\": 169}, {\"species\": \"Vulpes zerda\", \"common_name\": \"fennec fox\", \"alignment_count_type\": \"n_primary_and_unique_best\", \"Pearson correlation\": 0.9957380145710041, \"aligned_reads_Crits_Christoph\": 165, \"aligned_reads_current_study\": 53}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alignment_count_type_selection = alt.selection_point(\n",
    "    fields=[\"alignment_count_type\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=sars2_aligned_by_run[\"alignment_count_type\"].unique(),\n",
    "        name=\"alignment count type\",\n",
    "    ),\n",
    "    value=\"n_primary\",\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_species_corr_chart = (\n",
    "    alt.Chart(crits_christoph_vs_current_species_corr)\n",
    "    .encode(\n",
    "        x=\"Pearson correlation\",\n",
    "        y=alt.Y(\n",
    "            \"common_name\",\n",
    "            sort=alt.SortField(\"Pearson correlation\", order=\"descending\"),\n",
    "            title=None,\n",
    "        ),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".3g\")\n",
    "            if crits_christoph_vs_current_species_corr[c].dtype == float\n",
    "            else c\n",
    "            for c in crits_christoph_vs_current_species_corr.columns.tolist()\n",
    "        ],\n",
    "    )\n",
    "    .mark_circle(size=50, opacity=1)\n",
    "    .properties(\n",
    "        height=alt.Step(13),\n",
    "        width=185,\n",
    "        title=alt.TitleParams(\n",
    "            [\"correlation across runs of reads aligning to each\", \"mitochondrial genome in this study vs Crits-Christoph et al\"],\n",
    "            align=\"left\",\n",
    "            anchor=\"start\",\n",
    "            fontSize=12,\n",
    "        ),\n",
    "    )\n",
    "    .add_params(alignment_count_type_selection)\n",
    "    .transform_filter(alignment_count_type_selection)\n",
    ")\n",
    "    \n",
    "    \n",
    "crits_christoph_vs_current_species_corr_chart"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "481be2ab-cf9d-4668-8437-2145bc1df369",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T15:00:12.430378Z",
     "iopub.status.busy": "2023-04-08T15:00:12.430041Z",
     "iopub.status.idle": "2023-04-08T15:00:12.445176Z",
     "shell.execute_reply": "2023-04-08T15:00:12.444213Z",
     "shell.execute_reply.started": "2023-04-08T15:00:12.430359Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Run accession</th>\n",
       "      <th>species</th>\n",
       "      <th>aligned reads Crits-Christoph</th>\n",
       "      <th>common_name</th>\n",
       "      <th>alignment_count_type</th>\n",
       "      <th>aligned reads current study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2925</th>\n",
       "      <td>CSSQ-1-3</td>\n",
       "      <td>CRR710782</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1677</th>\n",
       "      <td>A63</td>\n",
       "      <td>CRR710850</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1629</th>\n",
       "      <td>A63</td>\n",
       "      <td>CRR710849</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>573</th>\n",
       "      <td>A15</td>\n",
       "      <td>CRR710821</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3789</th>\n",
       "      <td>EWS-3#-2</td>\n",
       "      <td>CRR711158</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4412</th>\n",
       "      <td>Q61</td>\n",
       "      <td>CRR710762</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>552</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>3288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2780</th>\n",
       "      <td>B5</td>\n",
       "      <td>CRR710873</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>3598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2732</th>\n",
       "      <td>B5</td>\n",
       "      <td>CRR710872</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>3619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2876</th>\n",
       "      <td>B5</td>\n",
       "      <td>CRR710875</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>3649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2828</th>\n",
       "      <td>B5</td>\n",
       "      <td>CRR710874</td>\n",
       "      <td>Urocyon cinereoargenteus</td>\n",
       "      <td>0</td>\n",
       "      <td>gray fox</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>3658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>218 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample Run accession                   species  \\\n",
       "2925  CSSQ-1-3     CRR710782  Urocyon cinereoargenteus   \n",
       "1677       A63     CRR710850  Urocyon cinereoargenteus   \n",
       "1629       A63     CRR710849  Urocyon cinereoargenteus   \n",
       "573        A15     CRR710821  Urocyon cinereoargenteus   \n",
       "3789  EWS-3#-2     CRR711158  Urocyon cinereoargenteus   \n",
       "...        ...           ...                       ...   \n",
       "4412       Q61     CRR710762  Urocyon cinereoargenteus   \n",
       "2780        B5     CRR710873  Urocyon cinereoargenteus   \n",
       "2732        B5     CRR710872  Urocyon cinereoargenteus   \n",
       "2876        B5     CRR710875  Urocyon cinereoargenteus   \n",
       "2828        B5     CRR710874  Urocyon cinereoargenteus   \n",
       "\n",
       "      aligned reads Crits-Christoph common_name       alignment_count_type  \\\n",
       "2925                              0    gray fox  n_primary_and_unique_best   \n",
       "1677                              0    gray fox  n_primary_and_unique_best   \n",
       "1629                              0    gray fox  n_primary_and_unique_best   \n",
       "573                               0    gray fox  n_primary_and_unique_best   \n",
       "3789                              0    gray fox  n_primary_and_unique_best   \n",
       "...                             ...         ...                        ...   \n",
       "4412                            552    gray fox                  n_primary   \n",
       "2780                              0    gray fox                  n_primary   \n",
       "2732                              0    gray fox                  n_primary   \n",
       "2876                              0    gray fox                  n_primary   \n",
       "2828                              0    gray fox                  n_primary   \n",
       "\n",
       "      aligned reads current study  \n",
       "2925                            0  \n",
       "1677                            0  \n",
       "1629                            0  \n",
       "573                             0  \n",
       "3789                            0  \n",
       "...                           ...  \n",
       "4412                         3288  \n",
       "2780                         3598  \n",
       "2732                         3619  \n",
       "2876                         3649  \n",
       "2828                         3658  \n",
       "\n",
       "[218 rows x 7 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "    crits_christoph_vs_current\n",
    "    .query(\"common_name == 'gray fox'\")\n",
    "    .sort_values(\"aligned reads current study\")\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08f3611f-4831-4925-a23b-c6fc817cd43d",
   "metadata": {},
   "source": [
    "Now get the correlations by run:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "197c548e-b73a-46e8-ae3b-c3401ed36d6c",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-08T15:00:12.446581Z",
     "iopub.status.busy": "2023-04-08T15:00:12.446179Z",
     "iopub.status.idle": "2023-04-08T15:00:12.573443Z",
     "shell.execute_reply": "2023-04-08T15:00:12.572407Z",
     "shell.execute_reply.started": "2023-04-08T15:00:12.446557Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run accession</th>\n",
       "      <th>sample</th>\n",
       "      <th>alignment_count_type</th>\n",
       "      <th>correlation</th>\n",
       "      <th>aligned_reads_Crits_Christoph</th>\n",
       "      <th>aligned_reads_current_study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>CRR710854</td>\n",
       "      <td>A87</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0.046917</td>\n",
       "      <td>787</td>\n",
       "      <td>1547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>CRR710852</td>\n",
       "      <td>A87</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0.048627</td>\n",
       "      <td>742</td>\n",
       "      <td>1462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>CRR710853</td>\n",
       "      <td>A87</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0.050413</td>\n",
       "      <td>734</td>\n",
       "      <td>1487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>CRR710855</td>\n",
       "      <td>A87</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0.076438</td>\n",
       "      <td>854</td>\n",
       "      <td>1498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>CRR710813</td>\n",
       "      <td>A101</td>\n",
       "      <td>n_primary</td>\n",
       "      <td>0.141779</td>\n",
       "      <td>1187</td>\n",
       "      <td>2221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>CRR710900</td>\n",
       "      <td>F46</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0.999984</td>\n",
       "      <td>1138</td>\n",
       "      <td>1489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>CRR710902</td>\n",
       "      <td>F46</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0.999991</td>\n",
       "      <td>1175</td>\n",
       "      <td>1455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>CRR710903</td>\n",
       "      <td>F46</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1172</td>\n",
       "      <td>1448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>CRR710901</td>\n",
       "      <td>F46</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1130</td>\n",
       "      <td>1355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>CRR710857</td>\n",
       "      <td>A88</td>\n",
       "      <td>n_primary_and_unique_best</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>62</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>202 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Run accession sample       alignment_count_type  correlation  \\\n",
       "96      CRR710854    A87                  n_primary     0.046917   \n",
       "92      CRR710852    A87                  n_primary     0.048627   \n",
       "94      CRR710853    A87                  n_primary     0.050413   \n",
       "98      CRR710855    A87                  n_primary     0.076438   \n",
       "34      CRR710813   A101                  n_primary     0.141779   \n",
       "..            ...    ...                        ...          ...   \n",
       "177     CRR710900    F46  n_primary_and_unique_best     0.999984   \n",
       "181     CRR710902    F46  n_primary_and_unique_best     0.999991   \n",
       "183     CRR710903    F46  n_primary_and_unique_best     0.999995   \n",
       "179     CRR710901    F46  n_primary_and_unique_best     0.999995   \n",
       "103     CRR710857    A88  n_primary_and_unique_best     1.000000   \n",
       "\n",
       "     aligned_reads_Crits_Christoph  aligned_reads_current_study  \n",
       "96                             787                         1547  \n",
       "92                             742                         1462  \n",
       "94                             734                         1487  \n",
       "98                             854                         1498  \n",
       "34                            1187                         2221  \n",
       "..                             ...                          ...  \n",
       "177                           1138                         1489  \n",
       "181                           1175                         1455  \n",
       "183                           1172                         1448  \n",
       "179                           1130                         1355  \n",
       "103                             62                           18  \n",
       "\n",
       "[202 rows x 6 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crits_christoph_vs_current_corr = (\n",
    "    crits_christoph_vs_current\n",
    "#    .query(\"common_name not in ['gray fox', 'masked palm civet']\")\n",
    "    .groupby([\"Run accession\", \"sample\", \"alignment_count_type\"])\n",
    "    [[\"aligned reads Crits-Christoph\", \"aligned reads current study\"]]\n",
    "    .corr(method=\"pearson\")\n",
    "    .reset_index()\n",
    "    .query(\"level_3 == 'aligned reads Crits-Christoph'\")\n",
    "    .rename(columns={\"aligned reads current study\": \"correlation\"})\n",
    "    .drop(columns=[\"level_3\", \"aligned reads Crits-Christoph\"])\n",
    "    .query(\"correlation.notnull()\")\n",
    "    .merge(\n",
    "        crits_christoph_vs_current\n",
    "        .groupby([\"Run accession\", \"alignment_count_type\"], as_index=False)\n",
    "        .aggregate(\n",
    "            aligned_reads_Crits_Christoph=pd.NamedAgg(\"aligned reads Crits-Christoph\", \"sum\"),\n",
    "            aligned_reads_current_study=pd.NamedAgg(\"aligned reads current study\", \"sum\"),\n",
    "        )\n",
    "    )\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_corr.sort_values(\"correlation\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "060ef350-67e0-4690-b2b6-65af8245a757",
   "metadata": {},
   "source": [
    "The correlations by species:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8aed3b3-6113-40ab-83df-eec514e825b1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d4f8ed5-c52d-4c7f-880b-c297821b268b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
