{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bdf42685-7bac-45b8-b061-0851c35f3a59",
   "metadata": {},
   "source": [
    "# Make plots"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30fe09f9-1157-4594-a37d-c60db2190d6c",
   "metadata": {},
   "source": [
    "Import Python modules:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "814bf235-7c93-47f7-adbc-ba242b931670",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import itertools\n",
    "import os\n",
    "import warnings\n",
    "\n",
    "import altair as alt\n",
    "\n",
    "import numpy\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "import scipy.stats\n",
    "\n",
    "_ = alt.data_transformers.disable_max_rows()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "897e563c-8a14-48e7-958c-4c69e57bfd69",
   "metadata": {},
   "source": [
    "Get data and variable from `snakemake`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e4be094-39bd-4972-b49d-76e3b400d3ab",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "metagenomic_descriptions = snakemake.params.metagenomic_descriptions\n",
    "\n",
    "sars2_aligned_by_run = pd.read_csv(snakemake.input.sars2_aligned_by_run)\n",
    "\n",
    "sars2_aligned_by_sample = (\n",
    "    pd.read_csv(snakemake.input.sars2_aligned_by_sample)\n",
    "    .rename(\n",
    "        columns={\"percent_preprocessed_reads_aligning_to_SARS2\": \"% reads aligning to SARS2\"},\n",
    "    )\n",
    "    # correct spelling of \"wing\" in original metadata\n",
    "    .assign(**{\"Isolation source\": lambda x: x[\"Isolation source\"].str.replace(\"ine \", \"ing \")})\n",
    "    # only keep metagenomic samples\n",
    "    .query(\"description in @metagenomic_descriptions\")\n",
    ")\n",
    "\n",
    "mito_composition_by_run = pd.read_csv(snakemake.input.mito_composition_by_run)\n",
    "\n",
    "mito_composition_by_sample = (\n",
    "    pd.read_csv(snakemake.input.mito_composition_by_sample)\n",
    "    .assign(**{\"Isolation source\": lambda x: x[\"Isolation source\"].str.replace(\"ine\", \"ing \")})\n",
    "    .query(\"description in @metagenomic_descriptions\")\n",
    ")\n",
    "\n",
    "contig_counts = pd.read_csv(snakemake.input.contig_counts).assign(\n",
    "    species=lambda x: x[\"species\"].str.replace(\"_\", \" \")\n",
    ")\n",
    "\n",
    "crits_christoph_plotted_species = snakemake.params.crits_christoph_plotted_species\n",
    "\n",
    "crits_christoph_read_counts = pd.read_csv(snakemake.input.crits_christoph_read_counts)\n",
    "\n",
    "ngdc_to_crits_christoph = pd.read_csv(snakemake.input.ngdc_to_crits_christoph)\n",
    "\n",
    "output_plots = snakemake.output\n",
    "\n",
    "susceptible_table_params = snakemake.params.susceptible_table\n",
    "\n",
    "susceptible_csv = snakemake.output.susceptible_csv\n",
    "susceptible_tex = snakemake.output.susceptible_tex\n",
    "susceptible_mammal_csv = snakemake.output.susceptible_mammal_csv\n",
    "susceptible_mammal_tex = snakemake.output.susceptible_mammal_tex\n",
    "raccoon_dog_long_csv = snakemake.output.raccoon_dog_long"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f887eb07-373c-42b4-9717-5b7702b84f0c",
   "metadata": {},
   "source": [
    "## Compare mitochondrial DNA composition to Crits-Christoph et al\n",
    "First get Crits-Christoph read counts in tidy format, assigning NGDC run accessions and summing counts for runs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2a9fb08-f0fd-4539-9e7a-59f1ae2a9c44",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_read_counts_tidy = (\n",
    "    crits_christoph_read_counts\n",
    "    .drop(columns=[\"Location\", \"Sample_name\"])\n",
    "    .merge(\n",
    "        ngdc_to_crits_christoph\n",
    "        .assign(Filename=lambda x: x[\"fastq Crits-Christoph\"].map(lambda s: s.split(\".\")[0]))\n",
    "        .rename(columns={\"Run accession NGDC\": \"Run accession\"})\n",
    "        [[\"Filename\", \"Run accession\"]]\n",
    "        .drop_duplicates(),\n",
    "        validate=\"one_to_one\",\n",
    "    )\n",
    "    .merge(\n",
    "        sars2_aligned_by_run[[\"Run accession\", \"sample\"]].drop_duplicates(),\n",
    "        validate=\"many_to_one\",\n",
    "        how=\"left\",\n",
    "    )\n",
    "    .melt(\n",
    "        id_vars=[\"sample\", \"Run accession\", \"Filename\"],\n",
    "        var_name=\"species\",\n",
    "        value_name=\"aligned reads Crits-Christoph\",\n",
    "    )\n",
    "    .groupby([\"sample\", \"Run accession\", \"species\"], as_index=False)\n",
    "    .aggregate({\"aligned reads Crits-Christoph\": \"sum\"})\n",
    ")\n",
    "\n",
    "crits_christoph_read_counts_tidy"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81141895-958a-4cc7-b1d0-02e33f8c0064",
   "metadata": {},
   "source": [
    "Now merge current read counts with those from Crits-Christoph for all species:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "116340c5-ebe7-4da3-8472-d1acf24694ef",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_species = crits_christoph_read_counts_tidy[\"species\"].unique().tolist()\n",
    "\n",
    "mito_composition_to_compare = (\n",
    "    mito_composition_by_run\n",
    "    [[\"Run accession\", \"species\", \"common_name\", \"aligned_reads\", \"covered_bases\"]]\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"aligned_reads\": \"aligned reads current study\",\n",
    "            \"covered_bases\": \"covered bases current study\",\n",
    "        }\n",
    "    )\n",
    ")\n",
    "\n",
    "assert set(crits_christoph_species).issubset(mito_composition_to_compare[\"species\"])\n",
    "\n",
    "crits_christoph_vs_current = (\n",
    "    crits_christoph_read_counts_tidy\n",
    "    .merge(\n",
    "        mito_composition_to_compare,\n",
    "        validate=\"one_to_one\",\n",
    "        on=[\"Run accession\", \"species\"],\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f2893ad-9059-4a99-b026-971364fc4177",
   "metadata": {},
   "source": [
    "Now get the correlations by species and plot them:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fad10471-0e9e-47e4-b488-a89c2dc09d0a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_vs_current_species_corr = (\n",
    "    crits_christoph_vs_current\n",
    "    .groupby([\"species\", \"common_name\"])\n",
    "    [[\"aligned reads Crits-Christoph\", \"aligned reads current study\"]]\n",
    "    .corr(method=\"pearson\")\n",
    "    .reset_index()\n",
    "    .query(\"level_2 == 'aligned reads Crits-Christoph'\")\n",
    "    .rename(columns={\"aligned reads current study\": \"correlation\"})\n",
    "    .drop(columns=[\"level_2\", \"aligned reads Crits-Christoph\"])\n",
    "    .query(\"correlation.notnull()\")\n",
    "    .merge(\n",
    "        crits_christoph_vs_current\n",
    "        .groupby(\"species\", as_index=False)\n",
    "        .aggregate(\n",
    "            aligned_reads_Crits_Christoph=pd.NamedAgg(\"aligned reads Crits-Christoph\", \"sum\"),\n",
    "            aligned_reads_current_study=pd.NamedAgg(\"aligned reads current study\", \"sum\"),\n",
    "        )\n",
    "    )\n",
    "    .sort_values(\"correlation\")\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_species_corr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14913bb6-f68f-4c02-9fe9-fb824f2a28ec",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_vs_current_species_corr_chart = (\n",
    "    alt.Chart(crits_christoph_vs_current_species_corr)\n",
    "    .encode(\n",
    "        x=\"correlation\",\n",
    "        y=alt.Y(\n",
    "            \"common_name\",\n",
    "            sort=alt.SortField(\"correlation\", order=\"descending\"),\n",
    "            title=None,\n",
    "        ),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".3g\")\n",
    "            if crits_christoph_vs_current_species_corr[c].dtype == float\n",
    "            else c\n",
    "            for c in crits_christoph_vs_current_species_corr.columns.tolist()\n",
    "        ],\n",
    "    )\n",
    "    .mark_circle(size=50, opacity=1)\n",
    "    .properties(\n",
    "        height=alt.Step(13),\n",
    "        width=185,\n",
    "    )\n",
    ") \n",
    "    \n",
    "crits_christoph_vs_current_species_corr_chart"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08f3611f-4831-4925-a23b-c6fc817cd43d",
   "metadata": {},
   "source": [
    "Now get the correlations by run:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "197c548e-b73a-46e8-ae3b-c3401ed36d6c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_vs_current_run_corr = (\n",
    "    crits_christoph_vs_current\n",
    "    .groupby([\"Run accession\", \"sample\"])\n",
    "    [[\"aligned reads Crits-Christoph\", \"aligned reads current study\"]]\n",
    "    .corr(method=\"pearson\")\n",
    "    .reset_index()\n",
    "    .query(\"level_2 == 'aligned reads Crits-Christoph'\")\n",
    "    .rename(columns={\"aligned reads current study\": \"correlation\"})\n",
    "    .drop(columns=[\"level_2\", \"aligned reads Crits-Christoph\"])\n",
    "    .query(\"correlation.notnull()\")\n",
    "    .merge(\n",
    "        crits_christoph_vs_current\n",
    "        .groupby(\"Run accession\", as_index=False)\n",
    "        .aggregate(\n",
    "            aligned_reads_Crits_Christoph=pd.NamedAgg(\"aligned reads Crits-Christoph\", \"sum\"),\n",
    "            aligned_reads_current_study=pd.NamedAgg(\"aligned reads current study\", \"sum\"),\n",
    "        )\n",
    "    )\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_run_corr.sort_values(\"correlation\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "060ef350-67e0-4690-b2b6-65af8245a757",
   "metadata": {},
   "source": [
    "Plot the correlations by run:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8aed3b3-6113-40ab-83df-eec514e825b1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "aligned_reads_current_slider = alt.param(\n",
    "    bind=alt.binding_range(\n",
    "        name=\"minimum aligned reads in current study\",\n",
    "        min=0,\n",
    "        max=1000,\n",
    "    ),\n",
    "    value=50,\n",
    ")\n",
    "\n",
    "aligned_reads_crits_christoph_slider = alt.param(\n",
    "    bind=alt.binding_range(\n",
    "        name=\"minimum aligned reads in Crits-Christoph et al\",\n",
    "        min=0,\n",
    "        max=1000,\n",
    "    ),\n",
    "    value=50,\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_run_corr_chart = (\n",
    "    alt.Chart(crits_christoph_vs_current_run_corr)\n",
    "    .encode(\n",
    "        x=alt.X(\n",
    "            \"Run accession\",\n",
    "            sort=alt.SortField(\"correlation\", order=\"descending\"),\n",
    "        ),\n",
    "        y=alt.Y(\"correlation\", title=\"Pearson correlation\"),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".4g\")\n",
    "            if crits_christoph_vs_current_run_corr[c].dtype == float\n",
    "            else c\n",
    "            for c in crits_christoph_vs_current_run_corr.columns.tolist()\n",
    "        ],\n",
    "    )\n",
    "    .mark_circle(size=50, opacity=1)\n",
    "    .properties(\n",
    "        width=alt.Step(11),\n",
    "        height=150,\n",
    "        title=(\n",
    "            \"Correlation in read counts aligned to mitochondrial genomes of different \"\n",
    "            \"species in Crits-Christoph et al and current study\"\n",
    "        )\n",
    "    )\n",
    "    .add_params(aligned_reads_current_slider, aligned_reads_crits_christoph_slider)\n",
    "    .transform_filter(\n",
    "        alt.datum[\"aligned_reads_current_study\"] >= aligned_reads_current_slider\n",
    "    )\n",
    "    .transform_filter(\n",
    "        alt.datum[\"aligned_reads_Crits_Christoph\"] >= aligned_reads_crits_christoph_slider\n",
    "    )\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_run_corr_chart.save(\n",
    "    output_plots[\"crits_christoph_vs_current_run_corr\"]\n",
    ")\n",
    "\n",
    "crits_christoph_vs_current_run_corr_chart"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64a1950a-5692-4675-b6c3-74c37966d2af",
   "metadata": {},
   "source": [
    "## Plot mitochondrial composition for samples\n",
    "First, for each sample we get just the species that are sufficiently abundant in either the full species set or the Crits-Christoph species set, and call other species \"other\":"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db686fdc-cbac-4b7f-9796-98e4213959b1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "assert set(crits_christoph_species).issubset(mito_composition_by_sample[\"species\"])\n",
    "\n",
    "group_as_other_cutoff = 5  # group as \"other\" if < this %\n",
    "max_species_per_sample = 10  # group as other lower percents so we never get more than this many\n",
    "\n",
    "# require at least this many reads aligned to mitochondrial genomes\n",
    "min_mito_reads = 100\n",
    "\n",
    "composition_df = (\n",
    "    mito_composition_by_sample\n",
    "    .assign(\n",
    "        cc_species=lambda x: x[\"species\"].isin(crits_christoph_species),\n",
    "        cc_plotted=lambda x: x[\"common_name\"].isin(crits_christoph_plotted_species),\n",
    "        total_mito=lambda x: x.groupby(\"sample\")[\"aligned_reads\"].transform(\"sum\"),\n",
    "        percent=lambda x: 100 * x[\"aligned_reads\"] / x[\"total_mito\"],\n",
    "        aligned_reads_cc=lambda x: x[\"aligned_reads\"].where(x[\"cc_species\"], 0),\n",
    "        total_mito_cc=lambda x: x.groupby(\"sample\")[\"aligned_reads_cc\"].transform(\"sum\"),\n",
    "        percent_cc=lambda x: 100 * x[\"aligned_reads_cc\"] / x[\"total_mito_cc\"],\n",
    "        rank=lambda x: x.groupby(\"sample\")[\"aligned_reads\"].transform(\"rank\", method=\"first\", ascending=False),\n",
    "        other=lambda x: (\n",
    "            (x[[\"percent\", \"percent_cc\"]].max(axis=1) < group_as_other_cutoff)\n",
    "            | (x[\"rank\"] >= max_species_per_sample)\n",
    "        ),\n",
    "        species=lambda x: x[\"species\"].where(~x[\"other\"], \"other\"),\n",
    "        common_name=lambda x: x[\"common_name\"].where(~x[\"other\"], \"other\"),\n",
    "    )\n",
    "    .query(\"total_mito > @min_mito_reads\")\n",
    "    .groupby([\"sample\", \"species\", \"common_name\"], as_index=False)\n",
    "    [[\"aligned_reads\", \"aligned_reads_cc\", \"percent\", \"percent_cc\"]]\n",
    "    .aggregate(\"sum\")\n",
    ")\n",
    "\n",
    "composition_df = (\n",
    "    composition_df\n",
    "    .rename(columns={\"aligned_reads\": \"all\", \"aligned_reads_cc\": \"crits_christoph\"})\n",
    "    .melt(\n",
    "        id_vars=[\"sample\", \"species\", \"common_name\"],\n",
    "        value_vars=[\"all\", \"crits_christoph\"],\n",
    "        value_name=\"mitochondria_aligned_reads\",\n",
    "        var_name=\"species_set\",\n",
    "    )\n",
    "    .merge(\n",
    "        composition_df\n",
    "        .rename(columns={\"percent\": \"all\", \"percent_cc\": \"crits_christoph\"})\n",
    "        .melt(\n",
    "            id_vars=[\"sample\", \"species\", \"common_name\"],\n",
    "            value_vars=[\"all\", \"crits_christoph\"],\n",
    "            value_name=\"percent mitochondria reads\",\n",
    "            var_name=\"species_set\",\n",
    "        )\n",
    "    )\n",
    "    .assign(\n",
    "        species_set=lambda x: x[\"species_set\"].map(\n",
    "            {\n",
    "                \"all\": \"chordates\",\n",
    "                \"crits_christoph\": \"mammals\",\n",
    "            }\n",
    "        ),\n",
    "    )\n",
    "    .merge(\n",
    "        (\n",
    "            sars2_aligned_by_sample\n",
    "            [[\"sample\", \"preprocessed_reads\", \"SARS2_aligned_reads\"]]\n",
    "            .rename(\n",
    "                columns={\n",
    "                    \"preprocessed_reads\": \"total pre-processed reads\",\n",
    "                    \"SARS2_aligned_reads\": \"reads aligning to SARS2\",\n",
    "                }\n",
    "            )\n",
    "        ),\n",
    "        how=\"left\",\n",
    "        validate=\"many_to_one\",\n",
    "    )\n",
    "    .query(\"mitochondria_aligned_reads > 0\")\n",
    ")\n",
    "\n",
    "composition_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f406c4b-a8aa-4a13-8ec0-174515b29557",
   "metadata": {},
   "source": [
    "Now get composition from the actual Crits-Christoph data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e36cde66-b0e2-4df8-b28d-dcb101649aff",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_composition_df = (\n",
    "    crits_christoph_read_counts_tidy\n",
    "    .drop(columns=\"Run accession\")\n",
    "    .rename(columns={\"aligned reads Crits-Christoph\": \"mitochondria_aligned_reads\"})\n",
    "    .merge(\n",
    "        mito_composition_by_sample\n",
    "        [[\"species\", \"common_name\",]]\n",
    "        .drop_duplicates(),\n",
    "        on=\"species\",\n",
    "        validate=\"many_to_one\",\n",
    "    )\n",
    ")\n",
    "\n",
    "crits_christoph_composition_df = (\n",
    "    pd.concat(\n",
    "        [\n",
    "            crits_christoph_composition_df.assign(species_set=\"Crits-Christoph Table S3\"),\n",
    "            (\n",
    "                crits_christoph_composition_df\n",
    "                .query(\"common_name in @crits_christoph_plotted_species\")\n",
    "                .assign(species_set=\"Crits-Christoph Figure 1\")\n",
    "            )\n",
    "        ]\n",
    "    )\n",
    "    .assign(\n",
    "        percent=lambda x: (\n",
    "            100 * x[\"mitochondria_aligned_reads\"]\n",
    "            / x.groupby([\"sample\", \"species_set\"])[\"mitochondria_aligned_reads\"].transform(\"sum\")\n",
    "        ),\n",
    "        rank=lambda x: x.groupby([\"sample\", \"species_set\"])[\"mitochondria_aligned_reads\"].transform(\"rank\", method=\"first\", ascending=False),\n",
    "        other=lambda x: (\n",
    "            (x[[\"percent\"]].max(axis=1) < group_as_other_cutoff)\n",
    "            | (x[\"rank\"] >= max_species_per_sample)\n",
    "        ),\n",
    "        species=lambda x: x[\"species\"].where(~x[\"other\"], \"other\"),\n",
    "        common_name=lambda x: x[\"common_name\"].where(~x[\"other\"], \"other\"),\n",
    "    )\n",
    "    .groupby([\"species\", \"common_name\", \"sample\", \"species_set\"], as_index=False)\n",
    "    .aggregate({\"percent\": \"sum\", \"mitochondria_aligned_reads\": \"sum\"})\n",
    "    .rename(columns={\"percent\": \"percent mitochondria reads\"})\n",
    "    .merge(\n",
    "        composition_df\n",
    "        [[\"sample\", \"total pre-processed reads\", \"reads aligning to SARS2\"]]\n",
    "        .drop_duplicates(),\n",
    "        on=\"sample\",\n",
    "        validate=\"many_to_one\",\n",
    "    )\n",
    ")\n",
    "\n",
    "crits_christoph_composition_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aeed0b25-f5bc-41f1-8348-731c87b35730",
   "metadata": {},
   "source": [
    "Now make pie charts:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd37ae1e-e419-413a-b494-2446a86109fc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "composition_to_plot = pd.concat([composition_df, crits_christoph_composition_df])\n",
    "\n",
    "sample_selection = alt.selection_point(\n",
    "    fields=[\"sample\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=composition_to_plot[\"sample\"].unique(),\n",
    "        name=\"sample\",\n",
    "    ),\n",
    "    value=\"Q61\",\n",
    ")\n",
    "\n",
    "species_set_selection = alt.selection_point(\n",
    "    fields=[\"species_set\"],\n",
    "    bind=\"legend\",\n",
    "    toggle=\"true\",\n",
    ")\n",
    "\n",
    "# https://gist.github.com/leblancfg/b145a966108be05b4a387789c4f9f474\n",
    "tab10_colors = [\n",
    "    \"#5778a4\",\n",
    "    \"#e49444\",\n",
    "    \"#d1615d\",\n",
    "    \"#85b6b2\",\n",
    "    \"#6a9f58\",\n",
    "    \"#e7ca60\",\n",
    "    \"#a87c9f\",\n",
    "    \"#f1a2a9\",\n",
    "    \"#967662\",\n",
    "    \"#b8b0ac\",\n",
    "]\n",
    "\n",
    "mito_composition_chart = (\n",
    "    alt.Chart(composition_to_plot)\n",
    "    .encode(\n",
    "        theta=\"percent mitochondria reads\",\n",
    "        color=alt.Color(\n",
    "            \"common_name\",\n",
    "            title=\"species (common name)\",\n",
    "            scale=alt.Scale(range=tab10_colors),\n",
    "            legend=alt.Legend(\n",
    "                orient=\"left\", columns=1, symbolSize=125, symbolType=\"square\", labelLimit=0,\n",
    "            ),\n",
    "        ),\n",
    "        opacity=alt.Opacity(\n",
    "            \"species_set\",\n",
    "            title=\"species sets to show (click to select/deselect)\",\n",
    "            scale=alt.Scale(\n",
    "                domain=composition_to_plot[\"species_set\"].unique(),\n",
    "                range=[1] * composition_to_plot[\"species_set\"].nunique(),\n",
    "            ),\n",
    "            legend=alt.Legend(orient=\"bottom\", titleLimit=0, labelLimit=0),\n",
    "        ),\n",
    "        column=alt.Column(\n",
    "            \"species_set\",\n",
    "            title=None,\n",
    "            sort=composition_to_plot[\"species_set\"].unique(),\n",
    "            header=alt.Header(labelFontSize=11, labelFontWeight=\"bold\"),\n",
    "        ),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".3g\") if composition_to_plot[c].dtype == float else c\n",
    "            for c in composition_to_plot.columns\n",
    "        ],\n",
    "    )\n",
    "    .mark_arc()\n",
    "    .add_params(sample_selection, species_set_selection)\n",
    "    .transform_filter(sample_selection)\n",
    "    .transform_filter(species_set_selection)\n",
    "    .properties(height=160, width=160)\n",
    ")\n",
    "\n",
    "mito_composition_chart.save(output_plots[\"mito_composition\"])\n",
    "\n",
    "mito_composition_chart"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e64cd53c-f9cf-454d-83be-339650c3fb29",
   "metadata": {},
   "source": [
    "## Tabulate samples with a susceptible non-human species\n",
    "Tabulate samples where a susceptible species is at >20%:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31be74f1-4261-4fbc-8a3e-24f335108e4e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "top_chordates_annotation = (\n",
    "    composition_df\n",
    "    .query(\"species_set == 'chordates'\")\n",
    "    .assign(\n",
    "        top_chordates=lambda x: (\n",
    "            x[\"common_name\"]\n",
    "            + \" (\" + x[\"percent mitochondria reads\"].round().astype(int).astype(str) + \"%)\"\n",
    "        )\n",
    "    )\n",
    "    .sort_values(\"percent mitochondria reads\", ascending=False)\n",
    "    .groupby(\"sample\", as_index=False)\n",
    "    .aggregate({\"top_chordates\": lambda s: \", \".join(s)})\n",
    ")\n",
    "\n",
    "top_mammals_annotation = (\n",
    "    composition_df\n",
    "    .query(\"species_set == 'mammals'\")\n",
    "    .assign(\n",
    "        top_mammals=lambda x: (\n",
    "            x[\"common_name\"]\n",
    "            + \" (\" + x[\"percent mitochondria reads\"].round().astype(int).astype(str) + \"%)\"\n",
    "        )\n",
    "    )\n",
    "    .sort_values(\"percent mitochondria reads\", ascending=False)\n",
    "    .groupby(\"sample\", as_index=False)\n",
    "    .aggregate({\"top_mammals\": lambda s: \", \".join(s)})\n",
    ")\n",
    "\n",
    "top_species_annotation = top_chordates_annotation.merge(\n",
    "    top_mammals_annotation, how=\"outer\", validate=\"one_to_one\",\n",
    ")\n",
    "\n",
    "top_species_annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e17e0f5-0690-49e6-b238-09cb3e54717b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "susceptible = susceptible_table_params[\"species\"]\n",
    "\n",
    "susceptible_min_percent = susceptible_table_params[\"min_percent\"]\n",
    "\n",
    "assert set(susceptible).issubset(composition_to_plot[\"common_name\"])\n",
    "    \n",
    "susceptible_table = (\n",
    "    composition_to_plot\n",
    "    .query('species_set == \"chordates\"')\n",
    "    .query(\"common_name in @susceptible\")\n",
    "    .sort_values(\"reads aligning to SARS2\", ascending=False)\n",
    "    .groupby([\"sample\", \"common_name\"], as_index=False, sort=False)\n",
    "    .first()\n",
    "    [[\"common_name\", \"sample\", \"percent mitochondria reads\", \"reads aligning to SARS2\", \"total pre-processed reads\", ]]\n",
    "    .assign(common_name=lambda x: pd.Categorical(x[\"common_name\"], susceptible, ordered=True))\n",
    "    .sort_values([\"common_name\", \"percent mitochondria reads\"], ascending=[True, False])\n",
    "    .query(\"`percent mitochondria reads` >= @susceptible_min_percent\")\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"common_name\": \"species\",\n",
    "            \"percent mitochondria reads\": \"% chordate mitochondrial reads from species\",\n",
    "        }\n",
    "    )\n",
    "    .reset_index(drop=True)\n",
    ")\n",
    "\n",
    "susceptible_table.to_csv(susceptible_csv, index=False, float_format=\"%.3g\")\n",
    "\n",
    "susceptible_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4c01750-92c4-4cff-959a-0a4600dad44e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "susceptible_mammal_table = (\n",
    "    composition_to_plot\n",
    "    .query('species_set == \"mammals\"')\n",
    "    .query(\"common_name in @susceptible\")\n",
    "    .sort_values(\"reads aligning to SARS2\", ascending=False)\n",
    "    .groupby([\"sample\", \"common_name\"], as_index=False, sort=False)\n",
    "    .first()\n",
    "    [[\"common_name\", \"sample\", \"percent mitochondria reads\", \"reads aligning to SARS2\", \"total pre-processed reads\", ]]\n",
    "    .assign(common_name=lambda x: pd.Categorical(x[\"common_name\"], susceptible, ordered=True))\n",
    "    .sort_values([\"common_name\", \"percent mitochondria reads\"], ascending=[True, False])\n",
    "    .query(\"`percent mitochondria reads` >= @susceptible_min_percent\")\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"common_name\": \"species\",\n",
    "            \"percent mitochondria reads\": \"% mammal mitochondrial reads from species\",\n",
    "        }\n",
    "    )\n",
    "    .reset_index(drop=True)\n",
    ")\n",
    "\n",
    "susceptible_mammal_table.to_csv(susceptible_mammal_csv, index=False, float_format=\"%.3g\")\n",
    "\n",
    "susceptible_mammal_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e952812-3bc4-4594-a793-3c3a6d5757de",
   "metadata": {},
   "source": [
    "Write LaTex version of table:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9cff4e9-8190-4538-9590-b177194112ef",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "latex_tab = (\n",
    "    susceptible_table\n",
    "    .set_index([\"species\", \"sample\"])\n",
    "    .to_latex(\n",
    "        sparsify=True,\n",
    "        multirow=True,\n",
    "        formatters={\n",
    "            \"total pre-processed reads\": (\n",
    "                lambda x: f\"{x:.2g}\"\n",
    "            ),\n",
    "            \"% chordate mitochondrial reads from species\": (\n",
    "                lambda x: f\"{x:.2g}%\"\n",
    "            ),\n",
    "        },\n",
    "    )\n",
    ")\n",
    "\n",
    "with open(susceptible_tex, \"w\") as f:\n",
    "    f.write(latex_tab)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "140d1ed1-1b7d-41d5-bf0d-ff076aee2807",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "latex_tab = (\n",
    "    susceptible_mammal_table\n",
    "    .set_index([\"species\", \"sample\"])\n",
    "    .to_latex(\n",
    "        sparsify=True,\n",
    "        multirow=True,\n",
    "        formatters={\n",
    "            \"total pre-processed reads\": (\n",
    "                lambda x: f\"{x:.2g}\"\n",
    "            ),\n",
    "            \"% mammal mitochondrial reads from species\": (\n",
    "                lambda x: f\"{x:.2g}%\"\n",
    "            ),\n",
    "        },\n",
    "    )\n",
    ")\n",
    "\n",
    "with open(susceptible_mammal_tex, \"w\") as f:\n",
    "    f.write(latex_tab)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8433c025-ea73-4537-8437-aade7891ff4a",
   "metadata": {},
   "source": [
    "Get long table ordered by raccoon dog percent:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d51567e0-1778-4ee8-9495-8c825c50ad97",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mito_composition_by_sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d61c215-5db6-468d-81be-0364c4b3eafd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "crits_christoph_species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee0a6b42-ac0c-4fc5-85da-f5690ee2d712",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "raccoon_dog_long = (\n",
    "    mito_composition_by_sample\n",
    "    .assign(\n",
    "        percent_chordate_reads=lambda x: x[\"aligned_reads\"] / x.groupby(\"sample\")[\"aligned_reads\"].transform(\"sum\") * 100,\n",
    "        mammal_reads=lambda x: x[\"aligned_reads\"].where(x[\"species\"].isin(crits_christoph_species), 0),\n",
    "        percent_mammal_reads=lambda x: x[\"aligned_reads\"] / x.groupby(\"sample\")[\"mammal_reads\"].transform(\"sum\") * 100,\n",
    "    )\n",
    "    .query(\"common_name == 'raccoon dog'\")\n",
    "    [[\"sample\", \"percent_chordate_reads\", \"percent_mammal_reads\", \"SARS2_aligned_reads\", \"preprocessed_reads\", \"Collection date\"]]\n",
    "    .sort_values([\"percent_chordate_reads\", \"percent_mammal_reads\", \"SARS2_aligned_reads\"], ascending=False)\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"percent_chordate_reads\": \"% chordate mitochondria from raccoon dog\",\n",
    "            \"percent_mammal_reads\": \"% mammal mitochondria from raccoon dog\",\n",
    "            \"SARS2_aligned_reads\": \"reads aligning to SARS2\",\n",
    "            \"preprocessed_reads\": \"total pre-processed reads\",\n",
    "        }\n",
    "    )\n",
    "    .merge(top_species_annotation, validate=\"one_to_one\", how=\"left\")\n",
    ")\n",
    "\n",
    "display(raccoon_dog_long)\n",
    "\n",
    "raccoon_dog_long.to_csv(raccoon_dog_long_csv, index=False, float_format=\"%.3g\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4499585c-f314-4beb-ab0c-cdef85fa428e",
   "metadata": {},
   "source": [
    "## Genomic contig composition\n",
    "Plot this for samples for which it was calculated:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59e29e4f-db2c-469c-a987-d73088ced0c2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# currently code assumes each accession a unique sample, check this is true\n",
    "dup_samples = (\n",
    "    contig_counts\n",
    "    .groupby(\"sample\")\n",
    "    .aggregate(\n",
    "        n_accessions=pd.NamedAgg(\"accession\", \"nunique\"),\n",
    "        accessions=pd.NamedAgg(\"accession\", \"unique\"),\n",
    "    )\n",
    "    .query(\"n_accessions > 1\")\n",
    ")\n",
    "assert len(dup_samples) == 0, dup_samples\n",
    "\n",
    "species = [\"chicken\", \"dog\", \"raccoon dog\", \"duck\"]\n",
    "colors = tab10_colors[: 2] + tab10_colors[3: 5]\n",
    "# must be true as long as we hardcode species to get desired colors\n",
    "assert set(contig_counts[\"species\"]).issubset(species)\n",
    "\n",
    "sample_selection = alt.selection_point(\n",
    "    fields=[\"sample\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=contig_counts[\"sample\"].unique(),\n",
    "        name=\"sample\",\n",
    "    ),\n",
    "    value=\"Q61\",\n",
    ")\n",
    "\n",
    "contig_composition_chart = (\n",
    "    alt.Chart(contig_counts)\n",
    "    .encode(\n",
    "        theta=\"aligned_contigs\",\n",
    "        color=alt.Color(\n",
    "            \"species\",\n",
    "            title=\"species\",\n",
    "            scale=alt.Scale(domain=species, range=colors),\n",
    "            legend=alt.Legend(\n",
    "                orient=\"right\", columns=1, symbolSize=125, symbolType=\"square\", labelLimit=0,\n",
    "            ),\n",
    "        ),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".3g\") if contig_counts[c].dtype == float else c\n",
    "            for c in contig_counts.columns\n",
    "        ],\n",
    "    )\n",
    "    .mark_arc()\n",
    "    .add_params(sample_selection)\n",
    "    .transform_filter(sample_selection)\n",
    "    .properties(height=175, width=175)\n",
    ")\n",
    "\n",
    "contig_composition_chart.save(output_plots[\"genomic_contig_composition\"])\n",
    "\n",
    "contig_composition_chart"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "828353b5-317d-4d70-8219-9669b878bcd1",
   "metadata": {},
   "source": [
    "## Distribution of amount of SARS-CoV-2 per sample\n",
    "First add annotations of top species to number of SARS2 aligned reads per sample:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "acfec5b8-411e-4ca0-a2bd-a7219ba8ea77",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "sars2_aligned_df = sars2_aligned_by_sample.merge(top_species_annotation, how=\"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e63110d-1d40-4cf0-b279-289b1c389873",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "date_selection = alt.selection_point(\n",
    "    fields=[\"Collection date\"],\n",
    "    bind=\"legend\",\n",
    "    toggle=\"true\",\n",
    ")\n",
    "\n",
    "source_selection = alt.selection_point(\n",
    "    fields=[\"Isolation source\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=[None, *sars2_aligned_df[\"Isolation source\"].unique()],\n",
    "        labels=[\"all\", *sars2_aligned_df[\"Isolation source\"].unique()],\n",
    "        name=\"sample isolation source\",\n",
    "    ),\n",
    ")\n",
    "\n",
    "sars2_aligned_chart_base = (\n",
    "    alt.Chart(sars2_aligned_df.query(\"preprocessed_reads > 0\"))\n",
    "    .encode(\n",
    "        color=alt.Color(\n",
    "            \"Collection date:O\",\n",
    "            title=\"Collection date (click to select/deselect)\",\n",
    "            scale=alt.Scale(\n",
    "                domain=sorted(sars2_aligned_df[\"Collection date\"].unique()),\n",
    "                scheme=\"viridis\",\n",
    "                reverse=True,\n",
    "            ),\n",
    "            legend=alt.Legend(orient=\"bottom\", columns=7, titleLimit=0),\n",
    "        ),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".3~g\")\n",
    "            if sars2_aligned_df[c].dtype in [float, int]\n",
    "            else c\n",
    "            for c in sars2_aligned_df.columns\n",
    "        ],\n",
    "    )\n",
    "    .mark_point(size=60, opacity=1, filled=True)\n",
    "    .add_params(date_selection, source_selection)\n",
    "    .transform_filter(source_selection)\n",
    "    .transform_filter(date_selection)\n",
    ")\n",
    "\n",
    "sars2_aligned_chart = (\n",
    "    sars2_aligned_chart_base\n",
    "    .encode(\n",
    "        x=alt.X(\n",
    "            \"sample\",\n",
    "            sort=alt.SortField(\"% reads aligning to SARS2\", order=\"descending\"),\n",
    "        ),\n",
    "        y=alt.Y(\n",
    "            \"% reads aligning to SARS2\",\n",
    "            title=\"% reads aligning to SARS2 (symlog scale)\",\n",
    "            scale=alt.Scale(type=\"symlog\", constant=0.00001),\n",
    "        ),\n",
    "    )\n",
    "    .properties(width=alt.Step(11), height=220)\n",
    ")\n",
    "\n",
    "sars2_aligned_chart.save(output_plots[\"sars2_aligned\"])\n",
    "\n",
    "sars2_aligned_chart"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "57b6d69d-2255-40ea-8910-6a3bfebe2e64",
   "metadata": {},
   "source": [
    "Make a vertical version of the same chart:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f83fd818-ca63-4894-bc20-14c61c7d01ac",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "sars2_aligned_chart_vertical = (\n",
    "    sars2_aligned_chart_base\n",
    "    # some transforms to facet into two columns\n",
    "    .transform_window(\n",
    "        rank=\"row_number()\",\n",
    "        sort=[\n",
    "            alt.SortField(\"% reads aligning to SARS2\"),\n",
    "        ],\n",
    "    )\n",
    "    .transform_joinaggregate(max_rank=\"max(rank)\")\n",
    "    .transform_calculate(\n",
    "        facet=alt.expr.if_(alt.datum[\"rank\"] / alt.datum[\"max_rank\"] > 0.5, 1, 2),\n",
    "    )\n",
    "    .encode(\n",
    "        y=alt.Y(\n",
    "            \"sample\",\n",
    "            title=None,\n",
    "            sort=alt.SortField(\"rank\", order=\"descending\"),\n",
    "        ),\n",
    "        x=alt.X(\n",
    "            \"% reads aligning to SARS2\",\n",
    "            title=\"% reads aligning to SARS2 (symlog scale)\",\n",
    "            scale=alt.Scale(type=\"symlog\", constant=0.00001),\n",
    "        ),\n",
    "        column=alt.Column(\"facet:N\", title=None, header=alt.Header(labels=False)),\n",
    "    )\n",
    "    .properties(height=alt.Step(9), width=200)\n",
    "    .configure_legend(orient=\"right\")\n",
    "    .resolve_scale(y=\"independent\")\n",
    ")\n",
    "\n",
    "sars2_aligned_chart_vertical.save(output_plots[\"sars2_aligned_vertical\"])\n",
    "\n",
    "sars2_aligned_chart_vertical"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4dcc522-d75d-41cc-b0c9-692e30b9e00e",
   "metadata": {},
   "source": [
    "## Per species correlations with SARS2 reads\n",
    "Make plots that correlation percent of reads aligned to SARS2 with percent aligned to each species.\n",
    "To enable log scaled plotting, when a percent is zero we set it to half the lowest non-zero percent observed across all samples / species."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b673a88-5535-4370-b338-49b5265f5e17",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "per_species_corr_df = (\n",
    "    mito_composition_by_sample\n",
    "    .drop(columns=[\"reference_id\", \"species\"])\n",
    "    .assign(\n",
    "        percent_SARS2=lambda x: 100 * x[\"SARS2_aligned_reads\"] / x[\"preprocessed_reads\"],\n",
    "        percent_species=lambda x: 100 * x[\"aligned_reads\"] / x[\"preprocessed_reads\"],\n",
    "        total_mitochondrial_reads=lambda x: x.groupby(\"sample\")[\"aligned_reads\"].transform(\"sum\"),\n",
    "        percent_of_mitochondrial_reads=lambda x: x[\"aligned_reads\"] / x[\"total_mitochondrial_reads\"] * 100,\n",
    "        axis_scale=\"linear scale with exact %\",\n",
    "    )\n",
    ")\n",
    "\n",
    "per_species_corr_df_log = (\n",
    "    per_species_corr_df\n",
    "    .assign(\n",
    "        percent_SARS2=lambda x: numpy.log10(\n",
    "            x[\"percent_SARS2\"].clip(lower=x.query(\"percent_SARS2 > 0\")[\"percent_SARS2\"].min() / 2)\n",
    "        ),\n",
    "        percent_species=lambda x: numpy.log10(\n",
    "            x[\"percent_species\"].clip(lower=x.query(\"percent_species > 0\")[\"percent_species\"].min() / 2)\n",
    "        ),\n",
    "        percent_of_mitochondrial_reads=lambda x: numpy.log10(\n",
    "            x[\"percent_of_mitochondrial_reads\"].clip(\n",
    "                lower=x.query(\"percent_of_mitochondrial_reads > 0\")[\"percent_of_mitochondrial_reads\"].min() / 2\n",
    "            )\n",
    "        ),\n",
    "        axis_scale=\"log10 of % (zero plotted as minimum non-zero value)\",\n",
    "    )\n",
    ")\n",
    "\n",
    "per_species_corr_df = (\n",
    "    pd.concat([per_species_corr_df, per_species_corr_df_log], ignore_index=True)\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"aligned_reads\": \"species_mitochondrial_reads\",\n",
    "            \"percent_SARS2\": \"% SARS2 reads\",\n",
    "            \"common_name\": \"species (common name)\",\n",
    "            \"percent_species\": \"% of all preprocessed reads\",\n",
    "            \"percent_of_mitochondrial_reads\": \"% of all mitochondrial reads\",\n",
    "        }\n",
    "    )\n",
    "    .merge(top_species_annotation, how=\"left\")\n",
    ")\n",
    "\n",
    "# melt on how mitochondrial read percent computed\n",
    "value_vars = [\"% of all preprocessed reads\", \"% of all mitochondrial reads\"]\n",
    "per_species_corr_df = (\n",
    "    per_species_corr_df\n",
    "    .melt(\n",
    "        id_vars=[c for c in per_species_corr_df.columns if c not in value_vars],\n",
    "        value_vars=value_vars,\n",
    "        value_name=\"% species mitochondria reads\",\n",
    "        var_name=\"% species mitochondria reads computed as\",\n",
    "    )\n",
    ")\n",
    "\n",
    "per_species_corr_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8945ed3-b095-448a-938c-9701d12a2200",
   "metadata": {},
   "source": [
    "First make base chart:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "088f9845-c8c5-47ca-a5bb-09ca925e95d7",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "has_sars2 = alt.param(\n",
    "    bind=alt.binding_radio(\n",
    "        options=[\"yes\", \"no\"], name=\"only show samples with SARS-CoV-2\",\n",
    "    ),\n",
    "    value=\"no\",\n",
    ")\n",
    "\n",
    "source_selection = alt.selection_point(\n",
    "    fields=[\"Isolation source\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=[None, *per_species_corr_df[\"Isolation source\"].unique()],\n",
    "        labels=[\"all\", *per_species_corr_df[\"Isolation source\"].unique()],\n",
    "        name=\"sample isolation source\",\n",
    "    ),\n",
    ")\n",
    "\n",
    "axis_scale_selection = alt.selection_point(\n",
    "    fields=[\"axis_scale\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=per_species_corr_df[\"axis_scale\"].unique(),\n",
    "        name=\"axis scale\",\n",
    "    ),\n",
    "    value=\"log10 of % (zero plotted as minimum non-zero value)\",\n",
    ")\n",
    "\n",
    "species_selection = alt.selection_point(\n",
    "    fields=[\"species (common name)\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=sorted(per_species_corr_df[\"species (common name)\"].unique()),\n",
    "        name=\"species (common name)\",\n",
    "    ),\n",
    "    value=\"raccoon dog\",\n",
    ")\n",
    "\n",
    "mito_percent_selection = alt.selection_point(\n",
    "    fields=[\"% species mitochondria reads computed as\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=per_species_corr_df[\"% species mitochondria reads computed as\"].unique(),\n",
    "        name=\"% species mitochondria reads computed as\",\n",
    "    ),\n",
    "    value= \"% of all preprocessed reads\",\n",
    ")\n",
    "\n",
    "min_mito_reads = 200\n",
    "\n",
    "mitochondrial_reads_slider = alt.param(\n",
    "    value=min_mito_reads,\n",
    "    bind=alt.binding_range(\n",
    "        min=0, max=1000, name=\"minimum total mitochondrial reads to show sample\",\n",
    "    ),\n",
    ")\n",
    "\n",
    "per_species_corr_chart_base = (\n",
    "    alt.Chart(per_species_corr_df)\n",
    "    .add_params(\n",
    "        date_selection,\n",
    "        axis_scale_selection,\n",
    "        mito_percent_selection,\n",
    "        source_selection,\n",
    "        mitochondrial_reads_slider,\n",
    "        has_sars2,\n",
    "    )\n",
    "    .transform_filter(mito_percent_selection)\n",
    "    .transform_filter(source_selection)\n",
    "    .transform_filter(date_selection)\n",
    "    .transform_filter(alt.datum[\"total_mitochondrial_reads\"] >= mitochondrial_reads_slider)\n",
    "    .transform_filter(axis_scale_selection)\n",
    "    .transform_filter(\n",
    "        (alt.datum[\"SARS2_aligned_reads\"] > 0) | (has_sars2 == \"no\")\n",
    "    )\n",
    ")\n",
    "    \n",
    "per_species_corr_chart_scatter_base = (\n",
    "    per_species_corr_chart_base\n",
    "    .encode(\n",
    "        y=alt.Y(\"% SARS2 reads\", scale=alt.Scale(zero=False)),\n",
    "        x=alt.X(\"% species mitochondria reads\", scale=alt.Scale(zero=False)),\n",
    "        color=alt.Color(\n",
    "            \"Collection date:O\",\n",
    "            title=\"Collection date (click to select/deselect)\",\n",
    "            scale=alt.Scale(\n",
    "                range=[\"black\"] * per_species_corr_df[\"Collection date\"].nunique(),\n",
    "                domain=sorted(per_species_corr_df[\"Collection date\"].unique()),\n",
    "            ),\n",
    "            legend=alt.Legend(orient=\"bottom\", columns=7, titleLimit=0),\n",
    "        ),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".3~g\")\n",
    "            if per_species_corr_df[c].dtype in [float, int]\n",
    "            else c\n",
    "            for c in per_species_corr_df.columns\n",
    "            if c not in {\n",
    "                \"axis_scale\",\n",
    "                \"total_mitochondrial_reads\",\n",
    "                \"% species mitochondria reads computed as\",\n",
    "            }\n",
    "        ],\n",
    "    )\n",
    "    .mark_circle(opacity=0.5, size=50)\n",
    ")\n",
    "\n",
    "# correlation coefficient\n",
    "params_r = (\n",
    "    per_species_corr_chart_base\n",
    "    .transform_regression(\n",
    "        \"% species mitochondria reads\", \"% SARS2 reads\", params=True,\n",
    "    )\n",
    "    .transform_calculate(\n",
    "        r=alt.expr.if_(\n",
    "            alt.datum[\"coef\"][1] >= 0,\n",
    "            alt.expr.sqrt(alt.datum[\"rSquared\"]),\n",
    "            -alt.expr.sqrt(alt.datum[\"rSquared\"]),\n",
    "        ),\n",
    "        label='\"r = \" + format(datum.r, \".2f\")',\n",
    "    )\n",
    "    .mark_text(align=\"left\", color=\"orange\", fontWeight=500, fontSize=12)\n",
    "    .encode(\n",
    "        x=alt.value(6),\n",
    "        y=alt.value(8),\n",
    "        text=alt.Text(\"label:N\"),\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b48b0e4-cd17-47c0-a298-17348d38d35a",
   "metadata": {},
   "source": [
    "Now use base to make faceted chart:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30b34c8e-aaa1-42d0-a6bf-7347912e008b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "per_species_corr_chart_faceted = (\n",
    "    (per_species_corr_chart_scatter_base + params_r)\n",
    "    .properties(width=150, height=150)\n",
    "    .facet(\n",
    "        alt.Facet(\n",
    "            \"species (common name)\",\n",
    "            title=None,\n",
    "            header=alt.Header(labelPadding=3, labelFontSize=12, labelFontWeight=500),\n",
    "        ),\n",
    "        columns=7,\n",
    "        spacing=12,\n",
    "    )\n",
    "    .configure(padding=25)\n",
    "    .configure_axis(grid=False)\n",
    ")\n",
    "\n",
    "per_species_corr_chart_faceted.save(\n",
    "    output_plots[\"per_species_corr_faceted\"]\n",
    ")\n",
    "\n",
    "per_species_corr_chart_faceted"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46f207ac-7ef3-4e76-ad48-67673719afc5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-09T14:11:36.861688Z",
     "iopub.status.busy": "2023-04-09T14:11:36.861051Z",
     "iopub.status.idle": "2023-04-09T14:11:36.869092Z",
     "shell.execute_reply": "2023-04-09T14:11:36.867969Z",
     "shell.execute_reply.started": "2023-04-09T14:11:36.861645Z"
    },
    "tags": []
   },
   "source": [
    "And use base to make single-species selection chart:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "425c809c-5af5-4171-979b-ae6342319331",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "per_species_corr_chart_single = (\n",
    "    (per_species_corr_chart_scatter_base + params_r)\n",
    "    .properties(width=200, height=200)\n",
    "    .add_params(species_selection)\n",
    "    .transform_filter(species_selection)\n",
    "    .configure(padding=25)\n",
    "    .configure_axis(grid=False)\n",
    ")\n",
    "\n",
    "per_species_corr_chart_single.save(\n",
    "    output_plots[\"per_species_corr_single\"]\n",
    ")\n",
    "\n",
    "per_species_corr_chart_single"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45b5ec10-2c0e-4397-af65-936924e1c825",
   "metadata": {},
   "source": [
    "## Overall correlations of SARS-CoV-2 and reads from species\n",
    "We filter samples on total mitochondrial reads first."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d2447d3-3ba8-4ac3-9763-abc506ad66be",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "warnings.simplefilter(\"ignore\")\n",
    "\n",
    "overall_corr_dfs = []\n",
    "for date, source, sars2 in itertools.product(\n",
    "    [\"all\", *per_species_corr_df[\"Collection date\"].unique()],\n",
    "    [\"all\", *per_species_corr_df[\"Isolation source\"].unique()],\n",
    "    [True, False],\n",
    "):\n",
    "    df = per_species_corr_df.query(\"total_mitochondrial_reads >= @min_mito_reads\")\n",
    "    if date != \"all\":\n",
    "        df = df.query(\"`Collection date` == @date\")\n",
    "    if source != \"all\":\n",
    "        df = df.query(\"`Isolation source` == @source\")\n",
    "    if sars2:\n",
    "        df = df.query(\"SARS2_aligned_reads > 0\").assign(has_sars2=\"yes\")\n",
    "    else:\n",
    "        df = df.assign(has_sars2=\"no\")\n",
    "    if (len(df) == 0):\n",
    "        continue\n",
    "    overall_corr_dfs.append(\n",
    "        df\n",
    "        .groupby(\n",
    "            [\n",
    "                \"species (common name)\",\n",
    "                \"axis_scale\",\n",
    "                \"% species mitochondria reads computed as\",\n",
    "                \"has_sars2\",\n",
    "            ],\n",
    "            as_index=False,\n",
    "        )\n",
    "        .apply(\n",
    "            lambda x: pd.Series(\n",
    "                {\n",
    "                    \"Pearson correlation\": x[\"% SARS2 reads\"].corr(\n",
    "                        x[\"% species mitochondria reads\"], method=\"pearson\",\n",
    "                    ),\n",
    "                    \"Spearman correlation\": x[\"% SARS2 reads\"].corr(\n",
    "                        x[\"% species mitochondria reads\"], method=\"spearman\",\n",
    "                    ),\n",
    "                }\n",
    "            )\n",
    "        )\n",
    "        .assign(**{\"Collection date\": date, \"Isolation source\": source})\n",
    "        .query(\"`Pearson correlation`.notnull() and `Spearman correlation`.notnull()\")\n",
    "    )\n",
    "\n",
    "overall_corr_df = pd.concat(overall_corr_dfs, ignore_index=True)\n",
    "\n",
    "overall_corr_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f24c699-c95e-4e2f-b455-314687961e3a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "source_selection = alt.selection_point(\n",
    "    fields=[\"Isolation source\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=overall_corr_df[\"Isolation source\"].unique(),\n",
    "        name=\"sample isolation source\",\n",
    "    ),\n",
    "    value=\"all\",\n",
    ")\n",
    "\n",
    "date_selection = alt.selection_point(\n",
    "    fields=[\"Collection date\"],\n",
    "    bind=alt.binding_select(\n",
    "        options=sorted(overall_corr_df[\"Collection date\"].unique()),\n",
    "        name=\"sample collection date\",\n",
    "    ),\n",
    "    value=\"all\",\n",
    ")\n",
    "\n",
    "species_selection = alt.selection_point(\n",
    "    fields=[\"species (common name)\"],\n",
    "    bind=alt.binding_radio(\n",
    "        options=[None, *sorted(overall_corr_df[\"species (common name)\"].unique())],\n",
    "        labels=[\"none\", *sorted(overall_corr_df[\"species (common name)\"].unique())],\n",
    "        name=\"species to highlight\",\n",
    "\n",
    "    ),\n",
    "    empty=False,\n",
    ")\n",
    "\n",
    "overall_corr_chart = (\n",
    "    alt.Chart(overall_corr_df)\n",
    "    .encode(\n",
    "        x=alt.X(\"Spearman correlation\", scale=alt.Scale(domain=(-1, 1))),\n",
    "        y=alt.Y(\"Pearson correlation\", scale=alt.Scale(domain=(-1, 1))),\n",
    "        tooltip=[\n",
    "            alt.Tooltip(c, format=\".3f\") if overall_corr_df[c].dtype == float else c\n",
    "            for c in overall_corr_df.columns.tolist()\n",
    "            if c not in [\n",
    "                \"axis_scale\",\n",
    "                \"Collection date\",\n",
    "                \"Isolation source\",\n",
    "                \"% species mitochondria reads computed as\",\n",
    "            ]\n",
    "        ],\n",
    "        size=alt.condition(species_selection, alt.value(100), alt.value(50)),\n",
    "        strokeWidth=alt.condition(species_selection, alt.value(2), alt.value(0)),\n",
    "        opacity=alt.condition(species_selection, alt.value(1), alt.value(0.5)), \n",
    "    )\n",
    "    .mark_circle(color=\"black\", size=50, stroke=\"red\")\n",
    "    .properties(width=200, height=200)\n",
    "    .configure_axis(grid=False)\n",
    "    .add_params(\n",
    "        species_selection,\n",
    "        source_selection,\n",
    "        axis_scale_selection,\n",
    "        has_sars2,\n",
    "        date_selection,\n",
    "        mito_percent_selection,\n",
    "    )\n",
    "    .transform_filter(source_selection)\n",
    "    .transform_filter(mito_percent_selection)\n",
    "    .transform_filter(axis_scale_selection)\n",
    "    .transform_filter(date_selection)\n",
    "    .transform_filter(alt.datum[\"has_sars2\"] == has_sars2)\n",
    ")\n",
    "\n",
    "overall_corr_chart.save(output_plots[\"overall_corr\"])\n",
    "\n",
    "overall_corr_chart"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "047c8533-8377-4b1a-a7ca-d541f37b2785",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
